// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/theme/tokens/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.tokens = void 0;
var _aspectRatios = require("./aspect-ratios");
var _animations = require("./animations");
var _blurs = require("./blurs");
var _borders = require("./borders");
var _colors = require("./colors");
var _durations = require("./durations");
var _easings = require("./easings");
var _fonts = require("./fonts");
var _fontSizes = require("./font-sizes");
var _fontWeights = require("./font-weights");
var _letterSpacings = require("./letter-spacings");
var _lineHeights = require("./line-heights");
var _radii = require("./radii");
var _spacing = require("./spacing");
var _sizes = require("./sizes");
var _zIndex = require("./z-index");
var _cursor = require("./cursor");
var tokens = exports.tokens = {
  aspectRatios: _aspectRatios.aspectRatios,
  animations: _animations.animations,
  blurs: _blurs.blurs,
  borders: _borders.borders,
  colors: _colors.colors,
  durations: _durations.durations,
  easings: _easings.easings,
  fonts: _fonts.fonts,
  fontSizes: _fontSizes.fontSizes,
  fontWeights: _fontWeights.fontWeights,
  letterSpacings: _letterSpacings.letterSpacings,
  lineHeights: _lineHeights.lineHeights,
  radii: _radii.radii,
  spacing: _spacing.spacing,
  sizes: _sizes.sizes,
  zIndex: _zIndex.zIndex,
  cursor: _cursor.cursor
};
