// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Image/Image.stories.js
import React from "react";
import { Image } from "./Image";
import { Stack, HStack } from "../Stack";
import { Box } from "../Box";

export default {
  title: "Media/Image",
  component: Image,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Image src="https://via.placeholder.com/300x200" alt="Placeholder image" />
);

export const WithFallback = () => (
  <Image
    src="invalid-url.jpg"
    fallbackSrc="https://via.placeholder.com/300x200?text=Fallback"
    alt="Image with fallback"
  />
);

export const Sizes = () => (
  <HStack gap="4" align="start">
    <Image src="https://via.placeholder.com/100" alt="Small" boxSize="100px" />
    <Image src="https://via.placeholder.com/150" alt="Medium" boxSize="150px" />
    <Image src="https://via.placeholder.com/200" alt="Large" boxSize="200px" />
  </HStack>
);

export const BorderRadius = () => (
  <HStack gap="4">
    <Image src="https://via.placeholder.com/150" alt="Rounded" boxSize="150px" borderRadius="md" />
    <Image
      src="https://via.placeholder.com/150"
      alt="Full rounded"
      boxSize="150px"
      borderRadius="full"
    />
  </HStack>
);

export const ObjectFit = () => (
  <HStack gap="4">
    <Box w="200px" h="150px" bg="bg.muted">
      <Image
        src="https://via.placeholder.com/400x300"
        alt="Cover"
        objectFit="cover"
        w="100%"
        h="100%"
      />
    </Box>
    <Box w="200px" h="150px" bg="bg.muted">
      <Image
        src="https://via.placeholder.com/400x300"
        alt="Contain"
        objectFit="contain"
        w="100%"
        h="100%"
      />
    </Box>
  </HStack>
);
