// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Box/Box.stories.js
import React from "react";
import { Box } from "./Box";

export default {
  title: "Layout/Box",
  component: Box,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Box background="tomato" width="100%" padding="4" color="white">
    This is the Box
  </Box>
);

export const WithBorder = () => (
  <Box p="4" borderWidth="1px" borderColor="border" borderRadius="md">
    Box with border
  </Box>
);

export const WithShadow = () => (
  <Box bg="bg" shadow="md" borderRadius="md" p="4">
    Box with shadow
  </Box>
);

export const AsProp = () => (
  <Box as="section" color="fg.muted" p="4" bg="bg.muted" borderRadius="md">
    This is a Box rendered as a section element
  </Box>
);

export const WithHover = () => (
  <Box
    bg="primary.500"
    w="100%"
    p="4"
    color="white"
    _hover={{ bg: "primary.600" }}
    cursor="pointer"
    borderRadius="md"
  >
    Hover over me
  </Box>
);

export const ResponsiveWidth = () => (
  <Box
    bg="primary.500"
    w={{ base: "100%", md: "50%", lg: "25%" }}
    p="4"
    color="white"
    borderRadius="md"
  >
    Responsive width
  </Box>
);
