// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Field/Field.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Field = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } // Field wrapper props for form inputs with labels/hints/errors
// Type exports for Chakra Field sub-components
/**
 * Field component - wraps form inputs with Field.Root when label, helperText, or errorText are provided
 * Used internally by Input, Textarea, and Select components
 */
var FieldRoot = /*#__PURE__*/_react["default"].forwardRef(function (_ref, ref) {
  var label = _ref.label,
    helperText = _ref.helperText,
    errorText = _ref.errorText,
    invalid = _ref.invalid,
    required = _ref.required,
    disabled = _ref.disabled,
    readOnly = _ref.readOnly,
    children = _ref.children;
  if (label || helperText || errorText) {
    return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_react2.Field.Root, {
      ref: ref,
      invalid: invalid,
      required: required,
      disabled: disabled,
      readOnly: readOnly,
      children: [label && /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Field.Label, {
        children: label
      }), children, helperText && !invalid && /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Field.HelperText, {
        children: helperText
      }), errorText && invalid && /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Field.ErrorText, {
        children: errorText
      })]
    });
  }
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_jsxRuntime.Fragment, {
    children: children
  });
});
FieldRoot.displayName = "Field";

// Export Chakra Field sub-components for direct use
var Root = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Field.Root, _objectSpread({
    ref: ref
  }, props));
});
Root.displayName = "Field.Root";
var Label = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Field.Label, _objectSpread({
    ref: ref
  }, props));
});
Label.displayName = "Field.Label";
var HelperText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Field.HelperText, _objectSpread({
    ref: ref
  }, props));
});
HelperText.displayName = "Field.HelperText";
var ErrorText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Field.ErrorText, _objectSpread({
    ref: ref
  }, props));
});
ErrorText.displayName = "Field.ErrorText";
var RequiredIndicator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Field.RequiredIndicator, _objectSpread({
    ref: ref
  }, props));
});
RequiredIndicator.displayName = "Field.RequiredIndicator";

// Combine wrapper with sub-components
var Field = exports.Field = Object.assign(FieldRoot, {
  Root: Root,
  Label: Label,
  HelperText: HelperText,
  ErrorText: ErrorText,
  RequiredIndicator: RequiredIndicator
});
var _default = exports["default"] = Field;
