// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/ColorPicker/ColorPicker.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.ColorPicker = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["children"],
  _excluded2 = ["children"],
  _excluded3 = ["children"],
  _excluded4 = ["children"],
  _excluded5 = ["children"],
  _excluded6 = ["children"],
  _excluded7 = ["children"],
  _excluded8 = ["children"],
  _excluded9 = ["children"],
  _excluded0 = ["children"],
  _excluded1 = ["children"],
  _excluded10 = ["children"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
var Root = /*#__PURE__*/_react["default"].forwardRef(function (_ref, ref) {
  var children = _ref.children,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.Root, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Root.displayName = "ColorPicker";
var Control = /*#__PURE__*/_react["default"].forwardRef(function (_ref2, ref) {
  var children = _ref2.children,
    props = _objectWithoutProperties(_ref2, _excluded2);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.Control, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Control.displayName = "ColorPickerControl";
var Trigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.Trigger, _objectSpread({
    ref: ref
  }, props));
});
Trigger.displayName = "ColorPickerTrigger";
var Content = /*#__PURE__*/_react["default"].forwardRef(function (_ref3, ref) {
  var children = _ref3.children,
    props = _objectWithoutProperties(_ref3, _excluded3);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.Content, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Content.displayName = "ColorPickerContent";
var Positioner = /*#__PURE__*/_react["default"].forwardRef(function (_ref4, ref) {
  var children = _ref4.children,
    props = _objectWithoutProperties(_ref4, _excluded4);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.Positioner, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Positioner.displayName = "ColorPickerPositioner";
var Area = /*#__PURE__*/_react["default"].forwardRef(function (_ref5, ref) {
  var children = _ref5.children,
    props = _objectWithoutProperties(_ref5, _excluded5);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.Area, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Area.displayName = "ColorPickerArea";
var AreaBackground = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.AreaBackground, _objectSpread({
    ref: ref
  }, props));
});
AreaBackground.displayName = "ColorPickerAreaBackground";
var AreaThumb = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.AreaThumb, _objectSpread({
    ref: ref
  }, props));
});
AreaThumb.displayName = "ColorPickerAreaThumb";
var ChannelSlider = /*#__PURE__*/_react["default"].forwardRef(function (_ref6, ref) {
  var children = _ref6.children,
    props = _objectWithoutProperties(_ref6, _excluded6);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.ChannelSlider, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
ChannelSlider.displayName = "ColorPickerChannelSlider";
var ChannelSliderTrack = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.ChannelSliderTrack, _objectSpread({
    ref: ref
  }, props));
});
ChannelSliderTrack.displayName = "ColorPickerChannelSliderTrack";
var ChannelSliderThumb = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.ChannelSliderThumb, _objectSpread({
    ref: ref
  }, props));
});
ChannelSliderThumb.displayName = "ColorPickerChannelSliderThumb";
var ChannelInput = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.ChannelInput, _objectSpread({
    ref: ref
  }, props));
});
ChannelInput.displayName = "ColorPickerChannelInput";
var TransparencyGrid = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.TransparencyGrid, _objectSpread({
    ref: ref
  }, props));
});
TransparencyGrid.displayName = "ColorPickerTransparencyGrid";
var SwatchGroup = /*#__PURE__*/_react["default"].forwardRef(function (_ref7, ref) {
  var children = _ref7.children,
    props = _objectWithoutProperties(_ref7, _excluded7);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.SwatchGroup, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
SwatchGroup.displayName = "ColorPickerSwatchGroup";
var SwatchTrigger = /*#__PURE__*/_react["default"].forwardRef(function (_ref8, ref) {
  var children = _ref8.children,
    props = _objectWithoutProperties(_ref8, _excluded8);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.SwatchTrigger, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
SwatchTrigger.displayName = "ColorPickerSwatchTrigger";
var Swatch = /*#__PURE__*/_react["default"].forwardRef(function (_ref9, ref) {
  var children = _ref9.children,
    props = _objectWithoutProperties(_ref9, _excluded9);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.Swatch, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Swatch.displayName = "ColorPickerSwatch";
var SwatchIndicator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.SwatchIndicator, _objectSpread({
    ref: ref
  }, props));
});
SwatchIndicator.displayName = "ColorPickerSwatchIndicator";
var EyeDropperTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.EyeDropperTrigger, _objectSpread({
    ref: ref
  }, props));
});
EyeDropperTrigger.displayName = "ColorPickerEyeDropperTrigger";
var Input = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.ChannelInput, _objectSpread({
    ref: ref,
    channel: "hex"
  }, props));
});
Input.displayName = "ColorPickerInput";
var Label = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.Label, _objectSpread({
    ref: ref
  }, props));
});
Label.displayName = "ColorPickerLabel";
var ValueText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.ValueText, _objectSpread({
    ref: ref
  }, props));
});
ValueText.displayName = "ColorPickerValueText";
var ValueSwatch = /*#__PURE__*/_react["default"].forwardRef(function (_ref0, ref) {
  var children = _ref0.children,
    props = _objectWithoutProperties(_ref0, _excluded0);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.ValueSwatch, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
ValueSwatch.displayName = "ColorPickerValueSwatch";
var View = /*#__PURE__*/_react["default"].forwardRef(function (_ref1, ref) {
  var children = _ref1.children,
    props = _objectWithoutProperties(_ref1, _excluded1);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.View, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
View.displayName = "ColorPickerView";
var FormatSelect = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.FormatSelect, _objectSpread({
    ref: ref
  }, props));
});
FormatSelect.displayName = "ColorPickerFormatSelect";
var HiddenInput = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.HiddenInput, _objectSpread({
    ref: ref
  }, props));
});
HiddenInput.displayName = "ColorPickerHiddenInput";
var ChannelSliderLabel = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.ChannelSliderLabel, _objectSpread({
    ref: ref
  }, props));
});
ChannelSliderLabel.displayName = "ColorPickerChannelSliderLabel";
var ChannelSliderValueText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.ChannelSliderValueText, _objectSpread({
    ref: ref
  }, props));
});
ChannelSliderValueText.displayName = "ColorPickerChannelSliderValueText";
var FormatTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.FormatTrigger, _objectSpread({
    ref: ref
  }, props));
});
FormatTrigger.displayName = "ColorPickerFormatTrigger";
var Sliders = /*#__PURE__*/_react["default"].forwardRef(function (_ref10, ref) {
  var children = _ref10.children,
    props = _objectWithoutProperties(_ref10, _excluded10);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.Sliders, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Sliders.displayName = "ColorPickerSliders";
var ChannelText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.ChannelText, _objectSpread({
    ref: ref
  }, props));
});
ChannelText.displayName = "ColorPickerChannelText";
var EyeDropper = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ColorPicker.EyeDropper, _objectSpread({
    ref: ref
  }, props));
});
EyeDropper.displayName = "ColorPickerEyeDropper";

// Attach sub-components as properties
var ColorPicker = exports.ColorPicker = Object.assign(Root, {
  Control: Control,
  Trigger: Trigger,
  Content: Content,
  Positioner: Positioner,
  Area: Area,
  AreaBackground: AreaBackground,
  AreaThumb: AreaThumb,
  ChannelSlider: ChannelSlider,
  ChannelSliderTrack: ChannelSliderTrack,
  ChannelSliderThumb: ChannelSliderThumb,
  ChannelInput: ChannelInput,
  TransparencyGrid: TransparencyGrid,
  SwatchGroup: SwatchGroup,
  SwatchTrigger: SwatchTrigger,
  Swatch: Swatch,
  SwatchIndicator: SwatchIndicator,
  EyeDropperTrigger: EyeDropperTrigger,
  Input: Input,
  Label: Label,
  ValueText: ValueText,
  ValueSwatch: ValueSwatch,
  View: View,
  FormatSelect: FormatSelect,
  HiddenInput: HiddenInput,
  ChannelSliderLabel: ChannelSliderLabel,
  ChannelSliderValueText: ChannelSliderValueText,
  FormatTrigger: FormatTrigger,
  Sliders: Sliders,
  ChannelText: ChannelText,
  EyeDropper: EyeDropper
});
var _default = exports["default"] = ColorPicker;
