// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/EmptyState/EmptyState.stories.js
import React from "react";

import { EmptyState } from "./EmptyState";
import { Button } from "../Button";
import { Stack } from "../Stack";

export default {
  title: "Data Display/EmptyState",
  component: EmptyState,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <EmptyState>
    <EmptyState.Content>
      <EmptyState.Indicator />
      <EmptyState.Title>No results found</EmptyState.Title>
      <EmptyState.Description>
        Try adjusting your search or filter to find what you're looking for.
      </EmptyState.Description>
    </EmptyState.Content>
  </EmptyState>
);

export const WithAction = () => (
  <EmptyState>
    <EmptyState.Content>
      <EmptyState.Indicator />
      <EmptyState.Title>No projects yet</EmptyState.Title>
      <EmptyState.Description>Get started by creating your first project.</EmptyState.Description>
      <Button colorPalette="primary">Create Project</Button>
    </EmptyState.Content>
  </EmptyState>
);

export const NoData = () => (
  <EmptyState>
    <EmptyState.Content>
      <EmptyState.Indicator />
      <EmptyState.Title>No data available</EmptyState.Title>
      <EmptyState.Description>There is no data to display at this time.</EmptyState.Description>
    </EmptyState.Content>
  </EmptyState>
);

export const SearchEmpty = () => (
  <EmptyState>
    <EmptyState.Content>
      <EmptyState.Indicator />
      <EmptyState.Title>No search results</EmptyState.Title>
      <EmptyState.Description>
        We couldn't find any results for your search. Try using different keywords.
      </EmptyState.Description>
      <Button variant="outline">Clear Search</Button>
    </EmptyState.Content>
  </EmptyState>
);
