// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/ActionBar/ActionBar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.ActionBar = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var Root = function Root(props) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ActionBar.Root, _objectSpread({}, props));
};
Root.displayName = "ActionBar";
var Content = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ActionBar.Content, _objectSpread({
    ref: ref
  }, props));
});
Content.displayName = "ActionBarContent";
var Positioner = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ActionBar.Positioner, _objectSpread({
    ref: ref
  }, props));
});
Positioner.displayName = "ActionBarPositioner";
var Separator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ActionBar.Separator, _objectSpread({
    ref: ref
  }, props));
});
Separator.displayName = "ActionBarSeparator";
var SelectionTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ActionBar.SelectionTrigger, _objectSpread({
    ref: ref
  }, props));
});
SelectionTrigger.displayName = "ActionBarSelectionTrigger";
var CloseTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ActionBar.CloseTrigger, _objectSpread({
    ref: ref
  }, props));
});
CloseTrigger.displayName = "ActionBarCloseTrigger";

// Attach sub-components as properties
var ActionBar = exports.ActionBar = Object.assign(Root, {
  Content: Content,
  Positioner: Positioner,
  Separator: Separator,
  SelectionTrigger: SelectionTrigger,
  CloseTrigger: CloseTrigger
});
var _default = exports["default"] = ActionBar;
