// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Carousel/Carousel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.CarouselProgressText = exports.CarouselPrevTrigger = exports.CarouselNextTrigger = exports.CarouselItemGroup = exports.CarouselItem = exports.CarouselIndicators = exports.CarouselIndicatorGroup = exports.CarouselIndicator = exports.CarouselControl = exports.CarouselAutoplayTrigger = exports.Carousel = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var Root = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Carousel.Root, _objectSpread({
    ref: ref
  }, props));
});
Root.displayName = "Carousel";
var ItemGroup = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Carousel.ItemGroup, _objectSpread({
    ref: ref
  }, props));
});
ItemGroup.displayName = "CarouselItemGroup";
var Item = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Carousel.Item, _objectSpread({
    ref: ref
  }, props));
});
Item.displayName = "CarouselItem";
var Control = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Carousel.Control, _objectSpread({
    ref: ref
  }, props));
});
Control.displayName = "CarouselControl";
var PrevTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Carousel.PrevTrigger, _objectSpread({
    ref: ref
  }, props));
});
PrevTrigger.displayName = "CarouselPrevTrigger";
var NextTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Carousel.NextTrigger, _objectSpread({
    ref: ref
  }, props));
});
NextTrigger.displayName = "CarouselNextTrigger";
var Indicator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Carousel.Indicator, _objectSpread({
    ref: ref
  }, props));
});
Indicator.displayName = "CarouselIndicator";
var IndicatorGroup = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Carousel.IndicatorGroup, _objectSpread({
    ref: ref
  }, props));
});
IndicatorGroup.displayName = "CarouselIndicatorGroup";
var AutoplayTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Carousel.AutoplayTrigger, _objectSpread({
    ref: ref
  }, props));
});
AutoplayTrigger.displayName = "CarouselAutoplayTrigger";
var ProgressText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Carousel.ProgressText, _objectSpread({
    ref: ref
  }, props));
});
ProgressText.displayName = "CarouselProgressText";

// Attach sub-components as properties
var Carousel = exports.Carousel = Object.assign(Root, {
  ItemGroup: ItemGroup,
  Item: Item,
  Control: Control,
  PrevTrigger: PrevTrigger,
  NextTrigger: NextTrigger,
  Indicator: Indicator,
  IndicatorGroup: IndicatorGroup,
  AutoplayTrigger: AutoplayTrigger,
  ProgressText: ProgressText,
  // Deprecated: Use IndicatorGroup instead
  Indicators: IndicatorGroup
});

// Named exports for backward compatibility
var CarouselItemGroup = exports.CarouselItemGroup = ItemGroup;
var CarouselItem = exports.CarouselItem = Item;
var CarouselControl = exports.CarouselControl = Control;
var CarouselPrevTrigger = exports.CarouselPrevTrigger = PrevTrigger;
var CarouselNextTrigger = exports.CarouselNextTrigger = NextTrigger;
var CarouselIndicator = exports.CarouselIndicator = Indicator;
var CarouselIndicatorGroup = exports.CarouselIndicatorGroup = IndicatorGroup;
var CarouselAutoplayTrigger = exports.CarouselAutoplayTrigger = AutoplayTrigger;
var CarouselProgressText = exports.CarouselProgressText = ProgressText;
// Deprecated: Use CarouselIndicatorGroup instead
var CarouselIndicators = exports.CarouselIndicators = IndicatorGroup;
var _default = exports["default"] = Carousel;
