// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Timeline/Timeline.stories.js
import React from "react";

import { Timeline } from "./Timeline";
import { Stack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Data Display/Timeline",
  component: Timeline,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Timeline>
    <Timeline.Item>
      <Timeline.Connector>
        <Timeline.Separator />
        <Timeline.Indicator />
      </Timeline.Connector>
      <Timeline.Content>
        <Timeline.Title>Step 1</Timeline.Title>
        <Timeline.Description>First step description</Timeline.Description>
      </Timeline.Content>
    </Timeline.Item>
    <Timeline.Item>
      <Timeline.Connector>
        <Timeline.Separator />
        <Timeline.Indicator />
      </Timeline.Connector>
      <Timeline.Content>
        <Timeline.Title>Step 2</Timeline.Title>
        <Timeline.Description>Second step description</Timeline.Description>
      </Timeline.Content>
    </Timeline.Item>
    <Timeline.Item>
      <Timeline.Connector>
        <Timeline.Separator />
        <Timeline.Indicator />
      </Timeline.Connector>
      <Timeline.Content>
        <Timeline.Title>Step 3</Timeline.Title>
        <Timeline.Description>Third step description</Timeline.Description>
      </Timeline.Content>
    </Timeline.Item>
  </Timeline>
);

export const OrderHistory = () => (
  <Timeline>
    <Timeline.Item>
      <Timeline.Connector>
        <Timeline.Separator />
        <Timeline.Indicator bg="green.500" />
      </Timeline.Connector>
      <Timeline.Content>
        <Timeline.Title>Order Delivered</Timeline.Title>
        <Timeline.Description>Dec 25, 2024 at 2:30 PM</Timeline.Description>
      </Timeline.Content>
    </Timeline.Item>
    <Timeline.Item>
      <Timeline.Connector>
        <Timeline.Separator />
        <Timeline.Indicator bg="blue.500" />
      </Timeline.Connector>
      <Timeline.Content>
        <Timeline.Title>Out for Delivery</Timeline.Title>
        <Timeline.Description>Dec 25, 2024 at 9:00 AM</Timeline.Description>
      </Timeline.Content>
    </Timeline.Item>
    <Timeline.Item>
      <Timeline.Connector>
        <Timeline.Separator />
        <Timeline.Indicator bg="blue.500" />
      </Timeline.Connector>
      <Timeline.Content>
        <Timeline.Title>Shipped</Timeline.Title>
        <Timeline.Description>Dec 23, 2024 at 4:15 PM</Timeline.Description>
      </Timeline.Content>
    </Timeline.Item>
    <Timeline.Item>
      <Timeline.Connector>
        <Timeline.Separator />
        <Timeline.Indicator bg="gray.500" />
      </Timeline.Connector>
      <Timeline.Content>
        <Timeline.Title>Order Placed</Timeline.Title>
        <Timeline.Description>Dec 22, 2024 at 10:00 AM</Timeline.Description>
      </Timeline.Content>
    </Timeline.Item>
  </Timeline>
);

export const Sizes = () => (
  <Stack gap="8">
    <Timeline size="sm">
      <Timeline.Item>
        <Timeline.Connector>
          <Timeline.Separator />
          <Timeline.Indicator />
        </Timeline.Connector>
        <Timeline.Content>
          <Timeline.Title>Small size</Timeline.Title>
        </Timeline.Content>
      </Timeline.Item>
    </Timeline>
    <Timeline size="md">
      <Timeline.Item>
        <Timeline.Connector>
          <Timeline.Separator />
          <Timeline.Indicator />
        </Timeline.Connector>
        <Timeline.Content>
          <Timeline.Title>Medium size</Timeline.Title>
        </Timeline.Content>
      </Timeline.Item>
    </Timeline>
    <Timeline size="lg">
      <Timeline.Item>
        <Timeline.Connector>
          <Timeline.Separator />
          <Timeline.Indicator />
        </Timeline.Connector>
        <Timeline.Content>
          <Timeline.Title>Large size</Timeline.Title>
        </Timeline.Content>
      </Timeline.Item>
    </Timeline>
  </Stack>
);
