// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/RadioCard/RadioCard.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.RadioCardLabel = exports.RadioCardItemText = exports.RadioCardItemIndicator = exports.RadioCardItemHiddenInput = exports.RadioCardItemDescription = exports.RadioCardItemControl = exports.RadioCardItemContent = exports.RadioCardItemAddon = exports.RadioCardItem = exports.RadioCard = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var Root = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.RadioCard.Root, _objectSpread({
    ref: ref
  }, props));
});
Root.displayName = "RadioCard";
var Item = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.RadioCard.Item, _objectSpread({
    ref: ref
  }, props));
});
Item.displayName = "RadioCardItem";
var ItemControl = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.RadioCard.ItemControl, _objectSpread({
    ref: ref
  }, props));
});
ItemControl.displayName = "RadioCardItemControl";
var ItemText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.RadioCard.ItemText, _objectSpread({
    ref: ref
  }, props));
});
ItemText.displayName = "RadioCardItemText";
var ItemDescription = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.RadioCard.ItemDescription, _objectSpread({
    ref: ref
  }, props));
});
ItemDescription.displayName = "RadioCardItemDescription";
var ItemIndicator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.RadioCard.ItemIndicator, _objectSpread({
    ref: ref
  }, props));
});
ItemIndicator.displayName = "RadioCardItemIndicator";
var ItemHiddenInput = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.RadioCard.ItemHiddenInput, _objectSpread({
    ref: ref
  }, props));
});
ItemHiddenInput.displayName = "RadioCardItemHiddenInput";
var ItemContent = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.RadioCard.ItemContent, _objectSpread({
    ref: ref
  }, props));
});
ItemContent.displayName = "RadioCardItemContent";
var ItemAddon = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.RadioCard.ItemAddon, _objectSpread({
    ref: ref
  }, props));
});
ItemAddon.displayName = "RadioCardItemAddon";
var Label = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.RadioCard.Label, _objectSpread({
    ref: ref
  }, props));
});
Label.displayName = "RadioCardLabel";

// Attach sub-components as properties
var RadioCard = exports.RadioCard = Object.assign(Root, {
  Item: Item,
  ItemControl: ItemControl,
  ItemText: ItemText,
  ItemDescription: ItemDescription,
  ItemIndicator: ItemIndicator,
  ItemHiddenInput: ItemHiddenInput,
  ItemContent: ItemContent,
  ItemAddon: ItemAddon,
  Label: Label
});

// Backward compatible named exports
var RadioCardItem = exports.RadioCardItem = Item;
var RadioCardItemControl = exports.RadioCardItemControl = ItemControl;
var RadioCardItemText = exports.RadioCardItemText = ItemText;
var RadioCardItemDescription = exports.RadioCardItemDescription = ItemDescription;
var RadioCardItemIndicator = exports.RadioCardItemIndicator = ItemIndicator;
var RadioCardItemHiddenInput = exports.RadioCardItemHiddenInput = ItemHiddenInput;
var RadioCardItemContent = exports.RadioCardItemContent = ItemContent;
var RadioCardItemAddon = exports.RadioCardItemAddon = ItemAddon;
var RadioCardLabel = exports.RadioCardLabel = Label;
var _default = exports["default"] = RadioCard;
