// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Code/Code.stories.js
import React from "react";

import { Code } from "./Code";
import { Stack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Typography/Code",
  component: Code,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => <Code>console.log("hello")</Code>;

export const Inline = () => (
  <Text>
    Use the <Code>useState</Code> hook to manage state in React.
  </Text>
);

export const Variants = () => (
  <Stack gap="4">
    <Code variant="plain">Plain variant</Code>
    <Code variant="outline">Outline variant</Code>
    <Code variant="solid">Solid variant</Code>
    <Code variant="surface">Surface variant</Code>
    <Code variant="subtle">Subtle variant</Code>
  </Stack>
);

export const Sizes = () => (
  <Stack gap="4">
    <Code size="xs">size xs</Code>
    <Code size="sm">size sm</Code>
    <Code size="md">size md</Code>
    <Code size="lg">size lg</Code>
  </Stack>
);

export const ColorPalettes = () => (
  <Stack gap="4">
    <Code colorPalette="gray">Gray palette</Code>
    <Code colorPalette="primary">Primary palette</Code>
    <Code colorPalette="red">Red palette</Code>
    <Code colorPalette="green">Green palette</Code>
    <Code colorPalette="blue">Blue palette</Code>
  </Stack>
);
