// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Wrap/Wrap.stories.js
import React from "react";
import { Wrap } from "./Wrap";
import { Box } from "../Box";
import { Badge } from "../Badge";
import { Center } from "../Center";

export default {
  title: "Layout/Wrap",
  component: Wrap,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Wrap>
    <Badge>Badge 1</Badge>
    <Badge>Badge 2</Badge>
    <Badge>Badge 3</Badge>
    <Badge>Badge 4</Badge>
    <Badge>Badge 5</Badge>
  </Wrap>
);

export const WithGap = () => (
  <Wrap gap="5">
    {Array.from({ length: 10 }).map((_, index) => (
      <Box key={index} h="12" w="12" bg="primary.500" borderRadius="md" />
    ))}
  </Wrap>
);

export const WithJustify = () => (
  <Wrap gap="6" justify="center">
    {Array.from({ length: 5 }).map((_, index) => (
      <Wrap.Item key={index}>
        <Center w="180px" h="80px" bg="primary.500" color="white" borderRadius="md">
          Box {index + 1}
        </Center>
      </Wrap.Item>
    ))}
  </Wrap>
);

export const WithAlign = () => (
  <Wrap gap="6" align="center">
    {Array.from({ length: 5 }).map((_, index) => (
      <Wrap.Item key={index}>
        <Center
          w="180px"
          h={`${40 + index * 20}px`}
          bg="primary.500"
          color="white"
          borderRadius="md"
        >
          Box {index + 1}
        </Center>
      </Wrap.Item>
    ))}
  </Wrap>
);

export const ResponsiveGap = () => (
  <Wrap gap={{ base: "2", md: "4", lg: "6" }}>
    {Array.from({ length: 12 }).map((_, index) => (
      <Box key={index} h="12" w="12" bg="primary.500" borderRadius="md" />
    ))}
  </Wrap>
);

export const RowAndColumnGap = () => (
  <Wrap rowGap="6" columnGap="2">
    {Array.from({ length: 10 }).map((_, index) => (
      <Box key={index} w="12" h="12" bg="primary.500" borderRadius="md" />
    ))}
  </Wrap>
);
