// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/FileUpload/FileUpload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.FileUploadTrigger = exports.FileUploadLabel = exports.FileUploadItemSizeText = exports.FileUploadItemPreview = exports.FileUploadItemName = exports.FileUploadItemGroup = exports.FileUploadItemDeleteTrigger = exports.FileUploadItem = exports.FileUploadDropzone = exports.FileUpload = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var Root = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.FileUpload.Root, _objectSpread({
    ref: ref
  }, props));
});
Root.displayName = "FileUpload";
var Dropzone = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.FileUpload.Dropzone, _objectSpread({
    ref: ref
  }, props));
});
Dropzone.displayName = "FileUploadDropzone";
var Trigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.FileUpload.Trigger, _objectSpread({
    ref: ref
  }, props));
});
Trigger.displayName = "FileUploadTrigger";
var Label = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.FileUpload.Label, _objectSpread({
    ref: ref
  }, props));
});
Label.displayName = "FileUploadLabel";
var ItemGroup = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.FileUpload.ItemGroup, _objectSpread({
    ref: ref
  }, props));
});
ItemGroup.displayName = "FileUploadItemGroup";
var Item = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.FileUpload.Item, _objectSpread({
    ref: ref
  }, props));
});
Item.displayName = "FileUploadItem";
var ItemPreview = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.FileUpload.ItemPreview, _objectSpread({
    ref: ref
  }, props));
});
ItemPreview.displayName = "FileUploadItemPreview";
var ItemName = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.FileUpload.ItemName, _objectSpread({
    ref: ref
  }, props));
});
ItemName.displayName = "FileUploadItemName";
var ItemSizeText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.FileUpload.ItemSizeText, _objectSpread({
    ref: ref
  }, props));
});
ItemSizeText.displayName = "FileUploadItemSizeText";
var ItemDeleteTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.FileUpload.ItemDeleteTrigger, _objectSpread({
    ref: ref
  }, props));
});
ItemDeleteTrigger.displayName = "FileUploadItemDeleteTrigger";

// Attach sub-components as properties
var FileUpload = exports.FileUpload = Object.assign(Root, {
  Dropzone: Dropzone,
  Trigger: Trigger,
  Label: Label,
  ItemGroup: ItemGroup,
  Item: Item,
  ItemPreview: ItemPreview,
  ItemName: ItemName,
  ItemSizeText: ItemSizeText,
  ItemDeleteTrigger: ItemDeleteTrigger
});

// Backward compatible named exports
var FileUploadDropzone = exports.FileUploadDropzone = Dropzone;
var FileUploadTrigger = exports.FileUploadTrigger = Trigger;
var FileUploadLabel = exports.FileUploadLabel = Label;
var FileUploadItemGroup = exports.FileUploadItemGroup = ItemGroup;
var FileUploadItem = exports.FileUploadItem = Item;
var FileUploadItemPreview = exports.FileUploadItemPreview = ItemPreview;
var FileUploadItemName = exports.FileUploadItemName = ItemName;
var FileUploadItemSizeText = exports.FileUploadItemSizeText = ItemSizeText;
var FileUploadItemDeleteTrigger = exports.FileUploadItemDeleteTrigger = ItemDeleteTrigger;
var _default = exports["default"] = FileUpload;
