// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/FileUpload/FileUpload.stories.js
import React from "react";
import { FileUpload as ChakraFileUpload, useFileUploadContext } from "@chakra-ui/react";

import { FileUpload } from "./FileUpload";
import { Stack, HStack } from "../Stack";
import { Button } from "../Button";
import { Text } from "../Text";
import { Box } from "../Box";

export default {
  title: "Forms/FileUpload",
  component: FileUpload,
  parameters: {
    layout: "padded",
  },
};

// Helper component to render file list
const FileList = ({ showSize = false, clearable = true }) => {
  const fileUpload = useFileUploadContext();
  const files = fileUpload.acceptedFiles;

  if (files.length === 0) return null;

  return (
    <ChakraFileUpload.ItemGroup>
      {files.map((file) => (
        <ChakraFileUpload.Item key={file.name} file={file}>
          <ChakraFileUpload.ItemPreview />
          <ChakraFileUpload.ItemName />
          {showSize && <ChakraFileUpload.ItemSizeText />}
          {clearable && (
            <ChakraFileUpload.ItemDeleteTrigger asChild>
              <Button size="xs" variant="ghost">
                ×
              </Button>
            </ChakraFileUpload.ItemDeleteTrigger>
          )}
        </ChakraFileUpload.Item>
      ))}
    </ChakraFileUpload.ItemGroup>
  );
};

export const Basic = () => (
  <FileUpload maxW="300px">
    <ChakraFileUpload.HiddenInput />
    <FileUpload.Label>Upload file</FileUpload.Label>
    <FileUpload.Trigger asChild>
      <Button variant="outline">Choose file</Button>
    </FileUpload.Trigger>
    <FileList />
  </FileUpload>
);

export const WithDropzone = () => (
  <FileUpload maxW="400px">
    <ChakraFileUpload.HiddenInput />
    <FileUpload.Dropzone>
      <Box textAlign="center">
        <Text fontWeight="medium">Drag and drop files here</Text>
        <Text color="fg.muted" fontSize="sm" mt="1">
          or
        </Text>
        <FileUpload.Trigger asChild>
          <Button size="sm" variant="outline" mt="2">
            Browse files
          </Button>
        </FileUpload.Trigger>
      </Box>
    </FileUpload.Dropzone>
    <FileList showSize />
  </FileUpload>
);

export const MultipleFiles = () => (
  <FileUpload maxFiles={5} maxW="400px">
    <ChakraFileUpload.HiddenInput />
    <FileUpload.Label>Upload files (max 5)</FileUpload.Label>
    <FileUpload.Dropzone>
      <Box textAlign="center">
        <Text>Drag and drop files here</Text>
        <FileUpload.Trigger asChild>
          <Button size="sm" variant="outline" mt="2">
            Browse files
          </Button>
        </FileUpload.Trigger>
      </Box>
    </FileUpload.Dropzone>
    <FileList showSize />
  </FileUpload>
);

export const AcceptedTypes = () => (
  <FileUpload accept="image/*" maxW="400px">
    <ChakraFileUpload.HiddenInput />
    <FileUpload.Label>Upload images only</FileUpload.Label>
    <FileUpload.Dropzone>
      <Box textAlign="center">
        <Text>Drag and drop images here</Text>
        <Text color="fg.muted" fontSize="sm">
          .png, .jpg, .gif accepted
        </Text>
        <FileUpload.Trigger asChild>
          <Button size="sm" variant="outline" mt="2">
            Browse images
          </Button>
        </FileUpload.Trigger>
      </Box>
    </FileUpload.Dropzone>
    <FileList showSize />
  </FileUpload>
);
