// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/FileUpload/index.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "FileUpload", {
  enumerable: true,
  get: function get() {
    return _FileUpload.FileUpload;
  }
});
Object.defineProperty(exports, "FileUploadDropzone", {
  enumerable: true,
  get: function get() {
    return _FileUpload.FileUploadDropzone;
  }
});
Object.defineProperty(exports, "FileUploadItem", {
  enumerable: true,
  get: function get() {
    return _FileUpload.FileUploadItem;
  }
});
Object.defineProperty(exports, "FileUploadItemDeleteTrigger", {
  enumerable: true,
  get: function get() {
    return _FileUpload.FileUploadItemDeleteTrigger;
  }
});
Object.defineProperty(exports, "FileUploadItemGroup", {
  enumerable: true,
  get: function get() {
    return _FileUpload.FileUploadItemGroup;
  }
});
Object.defineProperty(exports, "FileUploadItemName", {
  enumerable: true,
  get: function get() {
    return _FileUpload.FileUploadItemName;
  }
});
Object.defineProperty(exports, "FileUploadItemPreview", {
  enumerable: true,
  get: function get() {
    return _FileUpload.FileUploadItemPreview;
  }
});
Object.defineProperty(exports, "FileUploadItemSizeText", {
  enumerable: true,
  get: function get() {
    return _FileUpload.FileUploadItemSizeText;
  }
});
Object.defineProperty(exports, "FileUploadLabel", {
  enumerable: true,
  get: function get() {
    return _FileUpload.FileUploadLabel;
  }
});
Object.defineProperty(exports, "FileUploadTrigger", {
  enumerable: true,
  get: function get() {
    return _FileUpload.FileUploadTrigger;
  }
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _FileUpload["default"];
  }
});
var _FileUpload = _interopRequireWildcard(require("./FileUpload"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, "default": e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
