// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Steps/Steps.stories.js
import React from "react";

import { Steps } from "./Steps";
import { Button } from "../Button";
import { Box } from "../Box";
import { Stack, HStack } from "../Stack";

export default {
  title: "Navigation/Steps",
  component: Steps,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Steps defaultStep={1} count={3}>
    <Steps.List>
      <Steps.Item index={0}>
        <Steps.Trigger>
          <Steps.Indicator />
          <Box>
            <Steps.Title>Step 1</Steps.Title>
            <Steps.Description>Account</Steps.Description>
          </Box>
        </Steps.Trigger>
        <Steps.Separator />
      </Steps.Item>
      <Steps.Item index={1}>
        <Steps.Trigger>
          <Steps.Indicator />
          <Box>
            <Steps.Title>Step 2</Steps.Title>
            <Steps.Description>Details</Steps.Description>
          </Box>
        </Steps.Trigger>
        <Steps.Separator />
      </Steps.Item>
      <Steps.Item index={2}>
        <Steps.Trigger>
          <Steps.Indicator />
          <Box>
            <Steps.Title>Step 3</Steps.Title>
            <Steps.Description>Confirm</Steps.Description>
          </Box>
        </Steps.Trigger>
      </Steps.Item>
    </Steps.List>
    <Steps.Content index={0}>
      <Box p="4" bg="bg.muted" borderRadius="md">
        Step 1: Create your account
      </Box>
    </Steps.Content>
    <Steps.Content index={1}>
      <Box p="4" bg="bg.muted" borderRadius="md">
        Step 2: Fill in your details
      </Box>
    </Steps.Content>
    <Steps.Content index={2}>
      <Box p="4" bg="bg.muted" borderRadius="md">
        Step 3: Confirm your information
      </Box>
    </Steps.Content>
    <Steps.CompletedContent>
      <Box p="4" bg="green.100" borderRadius="md" color="green.800">
        All steps completed!
      </Box>
    </Steps.CompletedContent>
    <HStack mt="4">
      <Steps.PrevTrigger asChild>
        <Button variant="outline" size="sm">
          Previous
        </Button>
      </Steps.PrevTrigger>
      <Steps.NextTrigger asChild>
        <Button variant="solid" size="sm">
          Next
        </Button>
      </Steps.NextTrigger>
    </HStack>
  </Steps>
);

export const Vertical = () => (
  <Steps orientation="vertical" defaultStep={1} count={3}>
    <Steps.List>
      <Steps.Item index={0}>
        <Steps.Trigger>
          <Steps.Indicator />
          <Box>
            <Steps.Title>Order Placed</Steps.Title>
            <Steps.Description>Your order has been received</Steps.Description>
          </Box>
        </Steps.Trigger>
        <Steps.Separator />
      </Steps.Item>
      <Steps.Item index={1}>
        <Steps.Trigger>
          <Steps.Indicator />
          <Box>
            <Steps.Title>Processing</Steps.Title>
            <Steps.Description>Your order is being prepared</Steps.Description>
          </Box>
        </Steps.Trigger>
        <Steps.Separator />
      </Steps.Item>
      <Steps.Item index={2}>
        <Steps.Trigger>
          <Steps.Indicator />
          <Box>
            <Steps.Title>Shipped</Steps.Title>
            <Steps.Description>Your order is on its way</Steps.Description>
          </Box>
        </Steps.Trigger>
      </Steps.Item>
    </Steps.List>
  </Steps>
);

export const Sizes = () => (
  <Stack gap="8">
    <Steps size="sm" defaultStep={1} count={3}>
      <Steps.List>
        <Steps.Item index={0}>
          <Steps.Trigger>
            <Steps.Indicator />
            <Steps.Title>Small</Steps.Title>
          </Steps.Trigger>
          <Steps.Separator />
        </Steps.Item>
        <Steps.Item index={1}>
          <Steps.Trigger>
            <Steps.Indicator />
            <Steps.Title>Step 2</Steps.Title>
          </Steps.Trigger>
          <Steps.Separator />
        </Steps.Item>
        <Steps.Item index={2}>
          <Steps.Trigger>
            <Steps.Indicator />
            <Steps.Title>Step 3</Steps.Title>
          </Steps.Trigger>
        </Steps.Item>
      </Steps.List>
    </Steps>
    <Steps size="md" defaultStep={1} count={3}>
      <Steps.List>
        <Steps.Item index={0}>
          <Steps.Trigger>
            <Steps.Indicator />
            <Steps.Title>Medium</Steps.Title>
          </Steps.Trigger>
          <Steps.Separator />
        </Steps.Item>
        <Steps.Item index={1}>
          <Steps.Trigger>
            <Steps.Indicator />
            <Steps.Title>Step 2</Steps.Title>
          </Steps.Trigger>
          <Steps.Separator />
        </Steps.Item>
        <Steps.Item index={2}>
          <Steps.Trigger>
            <Steps.Indicator />
            <Steps.Title>Step 3</Steps.Title>
          </Steps.Trigger>
        </Steps.Item>
      </Steps.List>
    </Steps>
    <Steps size="lg" defaultStep={1} count={3}>
      <Steps.List>
        <Steps.Item index={0}>
          <Steps.Trigger>
            <Steps.Indicator />
            <Steps.Title>Large</Steps.Title>
          </Steps.Trigger>
          <Steps.Separator />
        </Steps.Item>
        <Steps.Item index={1}>
          <Steps.Trigger>
            <Steps.Indicator />
            <Steps.Title>Step 2</Steps.Title>
          </Steps.Trigger>
          <Steps.Separator />
        </Steps.Item>
        <Steps.Item index={2}>
          <Steps.Trigger>
            <Steps.Indicator />
            <Steps.Title>Step 3</Steps.Title>
          </Steps.Trigger>
        </Steps.Item>
      </Steps.List>
    </Steps>
  </Stack>
);
