// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Steps/Steps.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.StepsTrigger = exports.StepsTitle = exports.StepsSeparator = exports.StepsPrevTrigger = exports.StepsNextTrigger = exports.StepsList = exports.StepsItem = exports.StepsIndicator = exports.StepsDescription = exports.StepsContent = exports.StepsCompletedContent = exports.Steps = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var Root = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Steps.Root, _objectSpread({
    ref: ref
  }, props));
});
Root.displayName = "Steps";
var List = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Steps.List, _objectSpread({
    ref: ref
  }, props));
});
List.displayName = "StepsList";
var Item = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Steps.Item, _objectSpread({
    ref: ref
  }, props));
});
Item.displayName = "StepsItem";
var Trigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Steps.Trigger, _objectSpread({
    ref: ref
  }, props));
});
Trigger.displayName = "StepsTrigger";
var Indicator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Steps.Indicator, _objectSpread({
    ref: ref
  }, props));
});
Indicator.displayName = "StepsIndicator";
var Separator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Steps.Separator, _objectSpread({
    ref: ref
  }, props));
});
Separator.displayName = "StepsSeparator";
var Content = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Steps.Content, _objectSpread({
    ref: ref
  }, props));
});
Content.displayName = "StepsContent";
var Title = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Steps.Title, _objectSpread({
    ref: ref
  }, props));
});
Title.displayName = "StepsTitle";
var Description = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Steps.Description, _objectSpread({
    ref: ref
  }, props));
});
Description.displayName = "StepsDescription";
var NextTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Steps.NextTrigger, _objectSpread({
    ref: ref
  }, props));
});
NextTrigger.displayName = "StepsNextTrigger";
var PrevTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Steps.PrevTrigger, _objectSpread({
    ref: ref
  }, props));
});
PrevTrigger.displayName = "StepsPrevTrigger";
var CompletedContent = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Steps.CompletedContent, _objectSpread({
    ref: ref
  }, props));
});
CompletedContent.displayName = "StepsCompletedContent";

// Attach sub-components as properties
var Steps = exports.Steps = Object.assign(Root, {
  List: List,
  Item: Item,
  Trigger: Trigger,
  Indicator: Indicator,
  Separator: Separator,
  Content: Content,
  Title: Title,
  Description: Description,
  NextTrigger: NextTrigger,
  PrevTrigger: PrevTrigger,
  CompletedContent: CompletedContent
});

// Named exports for individual components
var StepsList = exports.StepsList = Steps.List;
var StepsItem = exports.StepsItem = Steps.Item;
var StepsTrigger = exports.StepsTrigger = Steps.Trigger;
var StepsIndicator = exports.StepsIndicator = Steps.Indicator;
var StepsSeparator = exports.StepsSeparator = Steps.Separator;
var StepsContent = exports.StepsContent = Steps.Content;
var StepsTitle = exports.StepsTitle = Steps.Title;
var StepsDescription = exports.StepsDescription = Steps.Description;
var StepsNextTrigger = exports.StepsNextTrigger = Steps.NextTrigger;
var StepsPrevTrigger = exports.StepsPrevTrigger = Steps.PrevTrigger;
var StepsCompletedContent = exports.StepsCompletedContent = Steps.CompletedContent;
var _default = exports["default"] = Steps;
