// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Link/Link.stories.js
import React from "react";

import { Box } from "../Box";
import Link from "./Link";
import { Stack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Navigation/Link",
  component: Link,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "Link component for navigation with support for external links.",
      },
    },
  },
  argTypes: {
    variant: {
      control: "select",
      options: ["plain", "underline"],
      description: "Visual variant of the link",
    },
    colorPalette: {
      control: "select",
      options: ["primary", "gray", "blue", "green", "red"],
      description: "Color palette for the link",
    },
    external: {
      control: "boolean",
      description: "Open link in new tab with security attributes",
    },
  },
};

export const Default = () => <Link href="#">Default Link</Link>;

export const Variants = () => (
  <Stack gap={4}>
    <Link href="#" variant="plain">
      Plain Link
    </Link>
    <Link href="#" variant="underline">
      Underline Link
    </Link>
  </Stack>
);

export const ColorPalettes = () => (
  <Stack gap={4}>
    <Link href="#" colorPalette="primary">
      Primary Link
    </Link>
    <Link href="#" colorPalette="blue">
      Blue Link
    </Link>
    <Link href="#" colorPalette="green">
      Green Link
    </Link>
    <Link href="#" colorPalette="red">
      Red Link
    </Link>
    <Link href="#" colorPalette="gray">
      Gray Link
    </Link>
  </Stack>
);

export const ExternalLink = () => (
  <Stack gap={4}>
    <Link href="https://chakra-ui.com" external>
      External Link (opens in new tab)
    </Link>
    <Link href="https://github.com" external colorPalette="blue">
      GitHub →
    </Link>
  </Stack>
);

export const InlineWithText = () => (
  <Text>
    This is a paragraph with an{" "}
    <Link href="#" colorPalette="blue" variant="underline">
      inline link
    </Link>{" "}
    that you can click. Links can also be{" "}
    <Link href="https://example.com" external colorPalette="blue" variant="underline">
      external
    </Link>{" "}
    and open in a new tab.
  </Text>
);

export const NavigationLinks = () => (
  <Stack direction="row" gap={6}>
    <Link href="#">Home</Link>
    <Link href="#">About</Link>
    <Link href="#">Services</Link>
    <Link href="#">Contact</Link>
  </Stack>
);

export const WithIcon = () => (
  <Stack gap={4}>
    <Link href="#">
      <Box as="span" mr={2}>
        📧
      </Box>
      Email Us
    </Link>
    <Link href="#" external>
      <Box as="span" mr={2}>
        🔗
      </Box>
      External Resource
    </Link>
    <Link href="#">
      <Box as="span" mr={2}>
        📚
      </Box>
      Documentation
    </Link>
  </Stack>
);

export const FooterLinks = () => (
  <Box bg="gray.100" p={6} borderRadius="md">
    <Stack direction="row" gap={6} justify="center">
      <Link href="#" colorPalette="gray">
        Privacy Policy
      </Link>
      <Link href="#" colorPalette="gray">
        Terms of Service
      </Link>
      <Link href="#" colorPalette="gray">
        Cookie Policy
      </Link>
      <Link href="#" colorPalette="gray">
        Contact
      </Link>
    </Stack>
  </Box>
);
