// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Toast/Toast.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.toaster = exports["default"] = exports.Toaster = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } // Create a toaster instance
var toaster = exports.toaster = (0, _react2.createToaster)({
  placement: "bottom-end",
  pauseOnPageIdle: true
});

// Toaster component to render toasts
var Toaster = exports.Toaster = function Toaster(props) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Portal, {
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Toaster, _objectSpread(_objectSpread({
      toaster: toaster,
      insetInline: {
        mdDown: "4"
      }
    }, props), {}, {
      children: function children(toast) {
        return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_react2.Toast.Root, {
          width: {
            md: "sm"
          },
          children: [toast.type === "loading" ? /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Spinner, {
            size: "sm",
            color: "blue.solid"
          }) : /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Toast.Indicator, {}), /*#__PURE__*/(0, _jsxRuntime.jsxs)(_react2.Stack, {
            gap: "1",
            flex: "1",
            maxWidth: "100%",
            children: [toast.title && /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Toast.Title, {
              children: toast.title
            }), toast.description && /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Toast.Description, {
              children: toast.description
            })]
          }), toast.action && /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Toast.ActionTrigger, {
            children: toast.action.label
          }), toast.closable && /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Toast.CloseTrigger, {})]
        });
      }
    }))
  });
};
Toaster.displayName = "Toaster";
var _default = exports["default"] = Toaster;
