// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Toast/Toast.stories.js
import React from "react";

import { Toaster, toaster } from "./Toast";
import { Stack } from "../Stack";
import { Button } from "../Button";

export default {
  title: "Feedback/Toast",
  component: Toaster,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <>
    <Button
      onClick={() =>
        toaster.create({
          title: "Toast notification",
          description: "This is a basic toast message",
        })
      }
    >
      Show Toast
    </Button>
    <Toaster />
  </>
);

export const Types = () => (
  <>
    <Stack direction="row" gap="4">
      <Button
        onClick={() =>
          toaster.create({
            title: "Info",
            description: "This is an info toast",
            type: "info",
          })
        }
      >
        Info
      </Button>
      <Button
        onClick={() =>
          toaster.create({
            title: "Success",
            description: "This is a success toast",
            type: "success",
          })
        }
      >
        Success
      </Button>
      <Button
        onClick={() =>
          toaster.create({
            title: "Warning",
            description: "This is a warning toast",
            type: "warning",
          })
        }
      >
        Warning
      </Button>
      <Button
        onClick={() =>
          toaster.create({
            title: "Error",
            description: "This is an error toast",
            type: "error",
          })
        }
      >
        Error
      </Button>
    </Stack>
    <Toaster />
  </>
);

export const WithAction = () => (
  <>
    <Button
      onClick={() =>
        toaster.create({
          title: "Update available",
          description: "A new version is available",
          action: {
            label: "Update",
            onClick: () => console.log("Update clicked"),
          },
        })
      }
    >
      Show Toast with Action
    </Button>
    <Toaster />
  </>
);

export const Positions = () => (
  <>
    <Stack direction="row" gap="4" wrap="wrap">
      <Button
        onClick={() =>
          toaster.create({
            title: "Top Start",
            placement: "top-start",
          })
        }
      >
        Top Start
      </Button>
      <Button
        onClick={() =>
          toaster.create({
            title: "Top",
            placement: "top",
          })
        }
      >
        Top
      </Button>
      <Button
        onClick={() =>
          toaster.create({
            title: "Top End",
            placement: "top-end",
          })
        }
      >
        Top End
      </Button>
      <Button
        onClick={() =>
          toaster.create({
            title: "Bottom Start",
            placement: "bottom-start",
          })
        }
      >
        Bottom Start
      </Button>
      <Button
        onClick={() =>
          toaster.create({
            title: "Bottom",
            placement: "bottom",
          })
        }
      >
        Bottom
      </Button>
      <Button
        onClick={() =>
          toaster.create({
            title: "Bottom End",
            placement: "bottom-end",
          })
        }
      >
        Bottom End
      </Button>
    </Stack>
    <Toaster />
  </>
);

export const Duration = () => (
  <>
    <Stack direction="row" gap="4">
      <Button
        onClick={() =>
          toaster.create({
            title: "Short duration",
            description: "Closes in 2 seconds",
            duration: 2000,
          })
        }
      >
        2 seconds
      </Button>
      <Button
        onClick={() =>
          toaster.create({
            title: "Long duration",
            description: "Closes in 10 seconds",
            duration: 10000,
          })
        }
      >
        10 seconds
      </Button>
      <Button
        onClick={() =>
          toaster.create({
            title: "Persistent",
            description: "Click X to close",
            duration: Infinity,
          })
        }
      >
        Persistent
      </Button>
    </Stack>
    <Toaster />
  </>
);
