// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Center/Center.stories.js
import React from "react";
import { Center } from "./Center";
import { Box } from "../Box";
import { HStack } from "../Stack";

export default {
  title: "Layout/Center",
  component: Center,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Center bg="bg.muted" h="100px" borderRadius="md">
    <Box>This will be centered</Box>
  </Center>
);

export const WithAbsoluteCenter = () => (
  <Box position="relative" h="100px" bg="bg.muted" borderRadius="md">
    <Center.Absolute bg="primary.500" p="4" color="white" borderRadius="md">
      Absolute Center
    </Center.Absolute>
  </Box>
);

export const WithAbsoluteCenterAxis = () => (
  <HStack gap="8">
    <Box position="relative" h="100px" w="200px" bg="bg.muted" borderRadius="md">
      <Center.Absolute axis="horizontal" bg="primary.500" p="2" color="white" borderRadius="md">
        Horizontal
      </Center.Absolute>
    </Box>
    <Box position="relative" h="100px" w="200px" bg="bg.muted" borderRadius="md">
      <Center.Absolute axis="vertical" bg="primary.500" p="2" color="white" borderRadius="md">
        Vertical
      </Center.Absolute>
    </Box>
  </HStack>
);

export const CircleShape = () => (
  <HStack gap="4">
    <Center.Circle size="10" bg="primary.500" color="white">
      1
    </Center.Circle>
    <Center.Circle size="10" bg="primary.600" color="white">
      2
    </Center.Circle>
    <Center.Circle size="10" bg="primary.700" color="white">
      3
    </Center.Circle>
  </HStack>
);

export const SquareShape = () => (
  <HStack gap="4">
    <Center.Square size="10" bg="primary.500" color="white" borderRadius="md">
      1
    </Center.Square>
    <Center.Square size="10" bg="primary.600" color="white" borderRadius="md">
      2
    </Center.Square>
    <Center.Square size="10" bg="primary.700" color="white" borderRadius="md">
      3
    </Center.Square>
  </HStack>
);

export const InlineCenter = () => (
  <Center inline gap="2" p="2" bg="bg.muted" borderRadius="md">
    <Box>Inline centered content</Box>
    <Box color="primary.500">→</Box>
  </Center>
);
