// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/For/For.stories.js
import React from "react";

import { For } from "./For";
import { Box } from "../Box";
import { Stack, HStack } from "../Stack";
import { Badge } from "../Badge";

export default {
  title: "Utility/For",
  component: For,
  parameters: {
    layout: "padded",
  },
};

const items = ["Apple", "Banana", "Cherry", "Date", "Elderberry"];

export const Basic = () => (
  <Stack gap="2">
    <For each={items}>
      {(item, index) => (
        <Box key={index} p="2" bg="bg.muted" borderRadius="md">
          {item}
        </Box>
      )}
    </For>
  </Stack>
);

export const WithBadges = () => (
  <HStack gap="2">
    <For each={["React", "TypeScript", "Chakra UI", "Vite"]}>
      {(tech, index) => (
        <Badge key={index} colorPalette="primary">
          {tech}
        </Badge>
      )}
    </For>
  </HStack>
);

export const WithObjects = () => {
  const users = [
    { id: 1, name: "John", role: "Admin" },
    { id: 2, name: "Jane", role: "Editor" },
    { id: 3, name: "Bob", role: "Viewer" },
  ];

  return (
    <Stack gap="2">
      <For each={users}>
        {(user) => (
          <HStack key={user.id} p="3" bg="bg.muted" borderRadius="md" justify="space-between">
            <Box fontWeight="semibold">{user.name}</Box>
            <Badge>{user.role}</Badge>
          </HStack>
        )}
      </For>
    </Stack>
  );
};

export const WithFallback = () => (
  <For each={[]} fallback={<Box color="fg.muted">No items to display</Box>}>
    {(item) => <Box>{item}</Box>}
  </For>
);
