// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/providers/chakra.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
Object.defineProperty(exports, "osmosSystem", {
  enumerable: true,
  get: function get() {
    return _theme.osmosSystem;
  }
});
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _nextThemes = require("next-themes");
var _theme = require("../theme");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
// React import is used by JSX transform

/**
 * UIProvider wraps the application with Chakra UI and our custom Osmos theme
 *
 * The themeConfig is created using our custom createOsmosTheme() factory, which
 * automatically merges our base configuration with custom tokens and recipes.
 * This gives us a single, complete theme configuration without needing to
 * manually merge multiple configs.
 *
 * Theme features:
 * - Custom tokens (colors, typography, spacing, etc.)
 * - Component recipes with variants and sizes
 * - Semantic tokens for light/dark mode
 * - Complete independence from Chakra's defaultConfig
 *
 * Supports light/dark mode via next-themes
 */

function OsmosProvider(_ref) {
  var children = _ref.children,
    _ref$defaultColorMode = _ref.defaultColorMode,
    defaultColorMode = _ref$defaultColorMode === void 0 ? "light" : _ref$defaultColorMode,
    forcedColorMode = _ref.forcedColorMode;
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.ChakraProvider, {
    value: _theme.osmosSystem,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_nextThemes.ThemeProvider, {
      attribute: "class",
      defaultTheme: defaultColorMode,
      forcedTheme: forcedColorMode,
      enableSystem: true,
      disableTransitionOnChange: true,
      storageKey: "osmos-color-mode",
      children: children
    })
  });
}
var _default = exports["default"] = OsmosProvider;
