// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Radio/Radio.stories.js
import React, { useState } from "react";

import { Box } from "../Box";
import { Radio, RadioGroup } from "./Radio";
import { Stack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Form/Radio",
  component: RadioGroup,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "Radio and RadioGroup components for single selection from a list of options.",
      },
    },
  },
  argTypes: {
    size: {
      control: "select",
      options: ["sm", "md", "lg"],
      description: "Size of the radio buttons",
    },
    colorPalette: {
      control: "select",
      options: ["primary", "gray", "red", "green", "blue", "yellow", "purple", "teal"],
      description: "Color palette for the radio buttons",
    },
    animated: {
      control: "boolean",
      description: "Enable/disable animations",
      defaultValue: true,
    },
  },
};

export const Default = () => (
  <RadioGroup defaultValue="option1">
    <Stack gap={2}>
      <Radio value="option1">Option 1</Radio>
      <Radio value="option2">Option 2</Radio>
      <Radio value="option3">Option 3</Radio>
    </Stack>
  </RadioGroup>
);

export const Sizes = () => (
  <Stack gap={6}>
    <RadioGroup defaultValue="sm" size="sm">
      <Stack direction="row" gap={4}>
        <Radio value="sm">Small</Radio>
        <Radio value="md">Option 2</Radio>
      </Stack>
    </RadioGroup>
    <RadioGroup defaultValue="md" size="md">
      <Stack direction="row" gap={4}>
        <Radio value="md">Medium</Radio>
        <Radio value="other">Option 2</Radio>
      </Stack>
    </RadioGroup>
    <RadioGroup defaultValue="lg" size="lg">
      <Stack direction="row" gap={4}>
        <Radio value="lg">Large</Radio>
        <Radio value="other">Option 2</Radio>
      </Stack>
    </RadioGroup>
  </Stack>
);

export const ColorPalettes = () => (
  <Stack gap={4}>
    <RadioGroup defaultValue="primary" colorPalette="primary">
      <Radio value="primary">Primary</Radio>
    </RadioGroup>
    <RadioGroup defaultValue="green" colorPalette="green">
      <Radio value="green">Green</Radio>
    </RadioGroup>
    <RadioGroup defaultValue="blue" colorPalette="blue">
      <Radio value="blue">Blue</Radio>
    </RadioGroup>
    <RadioGroup defaultValue="red" colorPalette="red">
      <Radio value="red">Red</Radio>
    </RadioGroup>
  </Stack>
);

export const Horizontal = () => (
  <RadioGroup defaultValue="option1">
    <Stack direction="row" gap={4}>
      <Radio value="option1">Option 1</Radio>
      <Radio value="option2">Option 2</Radio>
      <Radio value="option3">Option 3</Radio>
    </Stack>
  </RadioGroup>
);

export const Disabled = () => (
  <RadioGroup defaultValue="option1">
    <Stack gap={2}>
      <Radio value="option1">Enabled option</Radio>
      <Radio value="option2" disabled>
        Disabled option
      </Radio>
      <Radio value="option3">Another enabled</Radio>
    </Stack>
  </RadioGroup>
);

export const Animations = () => {
  const [value, setValue] = useState("option1");

  return (
    <Stack gap={6}>
      <Box>
        <Text fontWeight="bold" mb={2}>
          Pop Animation
        </Text>
        <Text fontSize="sm" color="gray.600" mb={3}>
          Click different options to see the pop animation effect
        </Text>
        <RadioGroup value={value} onValueChange={(e) => setValue(e.value)}>
          <Stack gap={2}>
            <Radio value="option1">Option 1</Radio>
            <Radio value="option2">Option 2</Radio>
            <Radio value="option3">Option 3</Radio>
          </Stack>
        </RadioGroup>
      </Box>

      <Box>
        <Text fontWeight="bold" mb={2}>
          Hover Animation
        </Text>
        <Text fontSize="sm" color="gray.600" mb={3}>
          Hover over the radio buttons to see scale effect
        </Text>
        <RadioGroup defaultValue="hover1">
          <Stack direction="row" gap={4}>
            <Radio value="hover1">Hover me</Radio>
            <Radio value="hover2">Or me</Radio>
          </Stack>
        </RadioGroup>
      </Box>

      <Box>
        <Text fontWeight="bold" mb={2}>
          Different Colors
        </Text>
        <Text fontSize="sm" color="gray.600" mb={3}>
          Animations work with all color palettes
        </Text>
        <Stack gap={3}>
          <RadioGroup defaultValue="primary" colorPalette="primary">
            <Radio value="primary">Primary</Radio>
          </RadioGroup>
          <RadioGroup defaultValue="green" colorPalette="green">
            <Radio value="green">Green</Radio>
          </RadioGroup>
          <RadioGroup defaultValue="purple" colorPalette="purple">
            <Radio value="purple">Purple</Radio>
          </RadioGroup>
        </Stack>
      </Box>

      <Box>
        <Text fontWeight="bold" mb={2}>
          Disabled State
        </Text>
        <RadioGroup defaultValue="enabled">
          <Stack gap={2}>
            <Radio value="enabled">Enabled</Radio>
            <Radio value="disabled" disabled>
              Disabled (no animations)
            </Radio>
          </Stack>
        </RadioGroup>
      </Box>
    </Stack>
  );
};

export const WithoutAnimations = () => (
  <Stack gap={4}>
    <Text fontSize="sm" color="gray.600">
      Set animated=false to disable all animations
    </Text>
    <RadioGroup defaultValue="option1" animated={false}>
      <Stack gap={2}>
        <Radio value="option1">No animations</Radio>
        <Radio value="option2">Still no animations</Radio>
      </Stack>
    </RadioGroup>
  </Stack>
);
