// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Drawer/Drawer.stories.js
import React, { useState } from "react";
import { fn, userEvent, within, expect } from "@storybook/test";

import { Button } from "../Button";
import { Drawer } from "./Drawer";
import { Input } from "../Input";
import { Stack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Overlay/Drawer",
  component: Drawer,
  tags: ["autodocs"],
  parameters: {
    layout: "padded",
    docs: {
      description: {
        component:
          "Drawer component for displaying content in a sliding panel from the edge of the screen.",
      },
    },
  },
  args: {
    onOpenChange: fn(),
    onExitComplete: fn(),
  },
  argTypes: {
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg", "xl", "full"],
      description: "Size of the drawer",
    },
    placement: {
      control: "select",
      options: ["top", "bottom", "start", "end"],
      description: "Placement of the drawer",
    },
    onOpenChange: {
      action: "open changed",
      description: "Callback when drawer open state changes",
    },
    onExitComplete: {
      action: "exit complete",
      description: "Callback when drawer exit animation completes",
    },
  },
};

export const Default = {
  args: {
    size: "md",
    placement: "end",
  },
  render: (args) => (
    <Drawer {...args}>
      <Drawer.Trigger>
        <Button>Open Drawer</Button>
      </Drawer.Trigger>
      <Drawer.Content>
        <Drawer.Header>
          <Drawer.Title>Drawer Title</Drawer.Title>
        </Drawer.Header>
        <Drawer.Body>
          <Text>
            This is the drawer content. Drawers are typically used for navigation, forms, or
            displaying additional information.
          </Text>
        </Drawer.Body>
        <Drawer.Footer>
          <Drawer.ActionTrigger>
            <Button variant="outline">Cancel</Button>
          </Drawer.ActionTrigger>
          <Button>Save</Button>
        </Drawer.Footer>
        <Drawer.CloseTrigger />
      </Drawer.Content>
    </Drawer>
  ),
  play: async ({ canvasElement, args }) => {
    const canvas = within(canvasElement);
    const openButton = canvas.getByRole("button", { name: /open drawer/i });

    // Click to open drawer
    await userEvent.click(openButton);
    await expect(args.onOpenChange).toHaveBeenCalled();
  },
};

export const Placements = {
  args: {},
  render: (args) => (
    <Stack direction="row" gap={4} wrap="wrap">
      {["start", "end", "top", "bottom"].map((placement) => (
        <Drawer key={placement} placement={placement} {...args}>
          <Drawer.Trigger>
            <Button variant="outline">{placement}</Button>
          </Drawer.Trigger>
          <Drawer.Content>
            <Drawer.Header>
              <Drawer.Title>Placement: {placement}</Drawer.Title>
            </Drawer.Header>
            <Drawer.Body>
              <Text>Drawer from the {placement}.</Text>
            </Drawer.Body>
            <Drawer.CloseTrigger />
          </Drawer.Content>
        </Drawer>
      ))}
    </Stack>
  ),
};

export const Sizes = {
  args: {
    placement: "end",
  },
  render: (args) => (
    <Stack direction="row" gap={4} wrap="wrap">
      {["xs", "sm", "md", "lg", "xl"].map((size) => (
        <Drawer key={size} size={size} {...args}>
          <Drawer.Trigger>
            <Button variant="outline">{size.toUpperCase()}</Button>
          </Drawer.Trigger>
          <Drawer.Content>
            <Drawer.Header>
              <Drawer.Title>Size: {size}</Drawer.Title>
            </Drawer.Header>
            <Drawer.Body>
              <Text>This is a {size} drawer.</Text>
            </Drawer.Body>
            <Drawer.CloseTrigger />
          </Drawer.Content>
        </Drawer>
      ))}
    </Stack>
  ),
};

export const WithForm = {
  args: {
    size: "md",
    placement: "end",
  },
  render: (args) => (
    <Drawer {...args}>
      <Drawer.Trigger>
        <Button>Edit Profile</Button>
      </Drawer.Trigger>
      <Drawer.Content>
        <Drawer.Header>
          <Drawer.Title>Edit Profile</Drawer.Title>
          <Drawer.Description>Update your profile information</Drawer.Description>
        </Drawer.Header>
        <Drawer.Body>
          <Stack gap={4}>
            <Input label="Name" placeholder="Enter your name" defaultValue="John Doe" />
            <Input label="Email" placeholder="Enter email" defaultValue="john@example.com" />
            <Input label="Bio" placeholder="Short bio" />
          </Stack>
        </Drawer.Body>
        <Drawer.Footer>
          <Drawer.ActionTrigger>
            <Button variant="outline">Cancel</Button>
          </Drawer.ActionTrigger>
          <Button>Save Changes</Button>
        </Drawer.Footer>
        <Drawer.CloseTrigger />
      </Drawer.Content>
    </Drawer>
  ),
};

export const NavigationDrawer = {
  args: {
    placement: "start",
    size: "xs",
  },
  render: (args) => (
    <Drawer {...args}>
      <Drawer.Trigger>
        <Button variant="outline">Menu</Button>
      </Drawer.Trigger>
      <Drawer.Content>
        <Drawer.Header>
          <Drawer.Title>Navigation</Drawer.Title>
        </Drawer.Header>
        <Drawer.Body>
          <Stack gap={2}>
            <Button variant="ghost" justifyContent="flex-start">
              Home
            </Button>
            <Button variant="ghost" justifyContent="flex-start">
              Profile
            </Button>
            <Button variant="ghost" justifyContent="flex-start">
              Settings
            </Button>
            <Button variant="ghost" justifyContent="flex-start">
              Analytics
            </Button>
            <Button variant="ghost" justifyContent="flex-start">
              Reports
            </Button>
          </Stack>
        </Drawer.Body>
        <Drawer.CloseTrigger />
      </Drawer.Content>
    </Drawer>
  ),
};

export const Controlled = () => {
  const [open, setOpen] = useState(false);

  return (
    <>
      <Button onClick={() => setOpen(true)}>Open Controlled Drawer</Button>
      <Drawer open={open} onClose={() => setOpen(false)}>
        <Drawer.Content>
          <Drawer.Header>
            <Drawer.Title>Controlled Drawer</Drawer.Title>
          </Drawer.Header>
          <Drawer.Body>
            <Text>This drawer is controlled externally.</Text>
          </Drawer.Body>
          <Drawer.Footer>
            <Button onClick={() => setOpen(false)}>Close</Button>
          </Drawer.Footer>
        </Drawer.Content>
      </Drawer>
    </>
  );
};

export const FullScreen = {
  args: {
    size: "full",
  },
  render: (args) => (
    <Drawer {...args}>
      <Drawer.Trigger>
        <Button>Full Screen Drawer</Button>
      </Drawer.Trigger>
      <Drawer.Content>
        <Drawer.Header>
          <Drawer.Title>Full Screen</Drawer.Title>
        </Drawer.Header>
        <Drawer.Body>
          <Text>This drawer takes up the full screen.</Text>
        </Drawer.Body>
        <Drawer.Footer>
          <Drawer.ActionTrigger>
            <Button>Close</Button>
          </Drawer.ActionTrigger>
        </Drawer.Footer>
        <Drawer.CloseTrigger />
      </Drawer.Content>
    </Drawer>
  ),
};
