// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Alert/Alert.stories.js
import React from "react";

import { Alert } from "./Alert";
import { Stack } from "../Stack";

export default {
  title: "Feedback/Alert",
  component: Alert,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component:
          "Alert component for displaying important messages with different status variants.",
      },
    },
  },
  argTypes: {
    status: {
      control: "select",
      options: ["info", "warning", "success", "error"],
      description: "Status type of the alert",
    },
    variant: {
      control: "select",
      options: ["subtle", "solid", "outline"],
      description: "Visual variant of the alert",
    },
    title: {
      control: "text",
      description: "Title of the alert",
    },
    closable: {
      control: "boolean",
      description: "Show close button",
    },
  },
};

export const Default = {
  args: {
    status: "info",
    title: "Information",
    children: "This is an informational alert message.",
  },
};

export const Statuses = () => (
  <Stack gap={4}>
    <Alert status="info" title="Info">
      This is an info alert for general information.
    </Alert>
    <Alert status="success" title="Success">
      Your action was completed successfully.
    </Alert>
    <Alert status="warning" title="Warning">
      Please review before proceeding.
    </Alert>
    <Alert status="error" title="Error">
      An error occurred. Please try again.
    </Alert>
  </Stack>
);

export const Variants = () => (
  <Stack gap={4}>
    <Alert status="info" variant="subtle" title="Subtle">
      Subtle variant alert.
    </Alert>
    <Alert status="info" variant="solid" title="Solid">
      Solid variant alert.
    </Alert>
    <Alert status="info" variant="outline" title="Outline">
      Outline variant alert.
    </Alert>
  </Stack>
);

export const WithoutTitle = {
  args: {
    status: "success",
    children: "Operation completed successfully!",
  },
};

export const Closable = {
  args: {
    status: "warning",
    title: "Warning",
    children: "This alert can be dismissed.",
    closable: true,
  },
};

export const AllStatusesAllVariants = () => (
  <Stack gap={6}>
    {["subtle", "solid", "outline"].map((variant) => (
      <Stack key={variant} gap={2}>
        <strong>{variant}</strong>
        {["info", "success", "warning", "error"].map((status) => (
          <Alert key={`${variant}-${status}`} status={status} variant={variant} title={status}>
            This is a {status} alert with {variant} variant.
          </Alert>
        ))}
      </Stack>
    ))}
  </Stack>
);
