// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Clipboard/Clipboard.stories.js
import React from "react";
import { Clipboard } from "./Clipboard";
import { Button } from "../Button";
import { HStack } from "../Stack";

export default {
  title: "Utility/Clipboard",
  component: Clipboard,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Clipboard value="npm install @chakra-ui/react">
    <Clipboard.Label>Command</Clipboard.Label>
    <Clipboard.Control>
      <Clipboard.Input />
      <Clipboard.Trigger asChild>
        <Button variant="outline" size="sm">
          <Clipboard.Indicator copied="Copied!">Copy</Clipboard.Indicator>
        </Button>
      </Clipboard.Trigger>
    </Clipboard.Control>
  </Clipboard>
);

export const CopyButton = () => (
  <Clipboard value="https://chakra-ui.com">
    <Clipboard.Trigger asChild>
      <Button variant="solid">
        <Clipboard.Indicator copied="Copied!">Copy Link</Clipboard.Indicator>
      </Button>
    </Clipboard.Trigger>
  </Clipboard>
);

export const WithTimeout = () => (
  <Clipboard value="Copied text" timeout={1000}>
    <Clipboard.Trigger asChild>
      <Button variant="outline">
        <Clipboard.Indicator copied="Done!">Copy</Clipboard.Indicator>
      </Button>
    </Clipboard.Trigger>
  </Clipboard>
);

export const ShareLink = () => (
  <Clipboard value="https://example.com/share/abc123">
    <Clipboard.Label>Share this link</Clipboard.Label>
    <HStack>
      <Clipboard.Input readOnly flex="1" />
      <Clipboard.Trigger asChild>
        <Button colorPalette="primary">
          <Clipboard.Indicator copied="Copied!">Copy</Clipboard.Indicator>
        </Button>
      </Clipboard.Trigger>
    </HStack>
  </Clipboard>
);
