// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Slider/Slider.stories.js
import React from "react";

import { Slider } from "./Slider";
import { Stack, HStack } from "../Stack";

export default {
  title: "Forms/Slider",
  component: Slider,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Slider defaultValue={[50]} maxW="300px">
    <Slider.Control>
      <Slider.Track>
        <Slider.Range />
      </Slider.Track>
      <Slider.Thumb index={0} />
    </Slider.Control>
  </Slider>
);

export const WithLabel = () => (
  <Slider defaultValue={[50]} maxW="300px">
    <HStack justify="space-between">
      <Slider.Label>Volume</Slider.Label>
      <Slider.ValueText />
    </HStack>
    <Slider.Control>
      <Slider.Track>
        <Slider.Range />
      </Slider.Track>
      <Slider.Thumb index={0} />
    </Slider.Control>
  </Slider>
);

export const WithMarkers = () => (
  <Slider defaultValue={[50]} maxW="300px">
    <Slider.Control>
      <Slider.Track>
        <Slider.Range />
      </Slider.Track>
      <Slider.Thumb index={0} />
    </Slider.Control>
    <Slider.MarkerGroup>
      <Slider.Marker value={0}>0%</Slider.Marker>
      <Slider.Marker value={50}>50%</Slider.Marker>
      <Slider.Marker value={100}>100%</Slider.Marker>
    </Slider.MarkerGroup>
  </Slider>
);

export const Sizes = () => (
  <Stack gap="8" maxW="300px">
    <Slider defaultValue={[50]} size="xs">
      <Slider.Control>
        <Slider.Track>
          <Slider.Range />
        </Slider.Track>
        <Slider.Thumb index={0} />
      </Slider.Control>
    </Slider>
    <Slider defaultValue={[50]} size="sm">
      <Slider.Control>
        <Slider.Track>
          <Slider.Range />
        </Slider.Track>
        <Slider.Thumb index={0} />
      </Slider.Control>
    </Slider>
    <Slider defaultValue={[50]} size="md">
      <Slider.Control>
        <Slider.Track>
          <Slider.Range />
        </Slider.Track>
        <Slider.Thumb index={0} />
      </Slider.Control>
    </Slider>
    <Slider defaultValue={[50]} size="lg">
      <Slider.Control>
        <Slider.Track>
          <Slider.Range />
        </Slider.Track>
        <Slider.Thumb index={0} />
      </Slider.Control>
    </Slider>
  </Stack>
);

export const RangeSlider = () => (
  <Slider defaultValue={[25, 75]} maxW="300px">
    <HStack justify="space-between">
      <Slider.Label>Price Range</Slider.Label>
      <Slider.ValueText />
    </HStack>
    <Slider.Control>
      <Slider.Track>
        <Slider.Range />
      </Slider.Track>
      <Slider.Thumb index={0} />
      <Slider.Thumb index={1} />
    </Slider.Control>
  </Slider>
);

export const ColorPalettes = () => (
  <Stack gap="6" maxW="300px">
    <Slider defaultValue={[50]} colorPalette="primary">
      <Slider.Control>
        <Slider.Track>
          <Slider.Range />
        </Slider.Track>
        <Slider.Thumb index={0} />
      </Slider.Control>
    </Slider>
    <Slider defaultValue={[50]} colorPalette="green">
      <Slider.Control>
        <Slider.Track>
          <Slider.Range />
        </Slider.Track>
        <Slider.Thumb index={0} />
      </Slider.Control>
    </Slider>
    <Slider defaultValue={[50]} colorPalette="red">
      <Slider.Control>
        <Slider.Track>
          <Slider.Range />
        </Slider.Track>
        <Slider.Thumb index={0} />
      </Slider.Control>
    </Slider>
  </Stack>
);
