// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Select/Select.stories.js
import React from "react";

import { Select } from "./Select";
import { Stack } from "../Stack";

export default {
  title: "Form/Select",
  component: Select,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "Select component for choosing from a list of options with portal support.",
      },
    },
  },
  argTypes: {
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg"],
      description: "Size of the select",
    },
    variant: {
      control: "select",
      options: ["outline", "filled"],
      description: "Visual variant",
    },
    placeholder: {
      control: "text",
      description: "Placeholder text",
    },
    disabled: {
      control: "boolean",
      description: "Disable the select",
    },
    label: {
      control: "text",
      description: "Label for the select",
    },
  },
};

const frameworks = [
  { label: "React", value: "react" },
  { label: "Vue", value: "vue" },
  { label: "Angular", value: "angular" },
  { label: "Svelte", value: "svelte" },
];

export const Default = {
  args: {
    placeholder: "Select a framework",
    options: frameworks,
  },
};

export const WithLabel = {
  args: {
    label: "Framework",
    placeholder: "Select a framework",
    options: frameworks,
  },
};

export const WithHelperText = {
  args: {
    label: "Framework",
    placeholder: "Select a framework",
    options: frameworks,
    helperText: "Choose your preferred frontend framework",
  },
};

export const WithError = {
  args: {
    label: "Framework",
    placeholder: "Select a framework",
    options: frameworks,
    invalid: true,
    errorText: "Please select a framework",
  },
};

export const Sizes = () => (
  <Stack gap={4}>
    <Select size="xs" placeholder="Extra small" options={frameworks} />
    <Select size="sm" placeholder="Small" options={frameworks} />
    <Select size="md" placeholder="Medium (default)" options={frameworks} />
    <Select size="lg" placeholder="Large" options={frameworks} />
  </Stack>
);

export const Disabled = {
  args: {
    placeholder: "Disabled select",
    options: frameworks,
    disabled: true,
  },
};

export const Clearable = {
  args: {
    placeholder: "Select (clearable)",
    options: frameworks,
    clearable: true,
  },
};

export const Multiple = {
  args: {
    placeholder: "Select multiple",
    options: frameworks,
    multiple: true,
  },
};
