// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Select/Select.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.SelectItemGroup = exports.SelectItem = exports.Select = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _Field = require("../Field");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["label", "helperText", "errorText", "invalid", "required", "disabled", "placeholder", "options", "children"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
var Select = exports.Select = /*#__PURE__*/_react["default"].forwardRef(function (_ref, ref) {
  var label = _ref.label,
    helperText = _ref.helperText,
    errorText = _ref.errorText,
    invalid = _ref.invalid,
    required = _ref.required,
    disabled = _ref.disabled,
    _ref$placeholder = _ref.placeholder,
    placeholder = _ref$placeholder === void 0 ? "Select option" : _ref$placeholder,
    _ref$options = _ref.options,
    options = _ref$options === void 0 ? [] : _ref$options,
    children = _ref.children,
    props = _objectWithoutProperties(_ref, _excluded);
  var selectContent = /*#__PURE__*/(0, _jsxRuntime.jsxs)(_react2.Select.Root, _objectSpread(_objectSpread({
    ref: ref,
    disabled: disabled
  }, props), {}, {
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Select.Trigger, {
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Select.ValueText, {
        placeholder: placeholder
      })
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Portal, {
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Select.Positioner, {
        children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Select.Content, {
          children: children || options.map(function (option) {
            return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_react2.Select.Item, {
              item: option,
              children: [option.label, /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Select.ItemIndicator, {})]
            }, option.value);
          })
        })
      })
    })]
  }));
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_Field.Field, {
    label: label,
    helperText: helperText,
    errorText: errorText,
    invalid: invalid,
    required: required,
    disabled: disabled,
    children: selectContent
  });
});
Select.displayName = "Select";

// Export sub-components for advanced usage
var SelectItem = exports.SelectItem = _react2.Select.Item;
var SelectItemGroup = exports.SelectItemGroup = _react2.Select.ItemGroup;
var _default = exports["default"] = Select;
