// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/List/List.stories.js
import React from "react";
import { List } from "./List";
import { Stack } from "../Stack";

export default {
  title: "Data Display/List",
  component: List,
  parameters: {
    layout: "padded",
  },
};

export const Unordered = () => (
  <List>
    <List.Item>First item</List.Item>
    <List.Item>Second item</List.Item>
    <List.Item>Third item</List.Item>
  </List>
);

export const Ordered = () => (
  <List as="ol">
    <List.Item>First step</List.Item>
    <List.Item>Second step</List.Item>
    <List.Item>Third step</List.Item>
  </List>
);

export const WithIndicator = () => (
  <List>
    <List.Item>
      <List.Indicator />
      First item
    </List.Item>
    <List.Item>
      <List.Indicator />
      Second item
    </List.Item>
    <List.Item>
      <List.Indicator />
      Third item
    </List.Item>
  </List>
);

export const Nested = () => (
  <List>
    <List.Item>
      <List.Indicator />
      Parent item 1
      <List>
        <List.Item>
          <List.Indicator />
          Child item 1.1
        </List.Item>
        <List.Item>
          <List.Indicator />
          Child item 1.2
        </List.Item>
      </List>
    </List.Item>
    <List.Item>
      <List.Indicator />
      Parent item 2
    </List.Item>
  </List>
);
