// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/TreeView/TreeView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.TreeView = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var Root = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.Root, _objectSpread({
    ref: ref
  }, props));
});
Root.displayName = "TreeView";
var Tree = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.Tree, _objectSpread({
    ref: ref
  }, props));
});
Tree.displayName = "TreeViewTree";
var Branch = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.Branch, _objectSpread({
    ref: ref
  }, props));
});
Branch.displayName = "TreeViewBranch";
var BranchContent = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.BranchContent, _objectSpread({
    ref: ref
  }, props));
});
BranchContent.displayName = "TreeViewBranchContent";
var BranchControl = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.BranchControl, _objectSpread({
    ref: ref
  }, props));
});
BranchControl.displayName = "TreeViewBranchControl";
var BranchIndentGuide = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.BranchIndentGuide, _objectSpread({
    ref: ref
  }, props));
});
BranchIndentGuide.displayName = "TreeViewBranchIndentGuide";
var BranchIndicator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.BranchIndicator, _objectSpread({
    ref: ref
  }, props));
});
BranchIndicator.displayName = "TreeViewBranchIndicator";
var BranchText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.BranchText, _objectSpread({
    ref: ref
  }, props));
});
BranchText.displayName = "TreeViewBranchText";
var BranchTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.BranchTrigger, _objectSpread({
    ref: ref
  }, props));
});
BranchTrigger.displayName = "TreeViewBranchTrigger";
var Item = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.Item, _objectSpread({
    ref: ref
  }, props));
});
Item.displayName = "TreeViewItem";
var ItemIndicator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.ItemIndicator, _objectSpread({
    ref: ref
  }, props));
});
ItemIndicator.displayName = "TreeViewItemIndicator";
var ItemText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.ItemText, _objectSpread({
    ref: ref
  }, props));
});
ItemText.displayName = "TreeViewItemText";
var Label = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.Label, _objectSpread({
    ref: ref
  }, props));
});
Label.displayName = "TreeViewLabel";
var Node = function Node(props) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.Node, _objectSpread({}, props));
};
Node.displayName = "TreeViewNode";
var NodeCheckbox = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.TreeView.NodeCheckbox, _objectSpread({
    ref: ref
  }, props));
});
NodeCheckbox.displayName = "TreeViewNodeCheckbox";

// Attach sub-components as properties
var TreeView = exports.TreeView = Object.assign(Root, {
  Tree: Tree,
  Branch: Branch,
  BranchContent: BranchContent,
  BranchControl: BranchControl,
  BranchIndentGuide: BranchIndentGuide,
  BranchIndicator: BranchIndicator,
  BranchText: BranchText,
  BranchTrigger: BranchTrigger,
  Item: Item,
  ItemIndicator: ItemIndicator,
  ItemText: ItemText,
  Label: Label,
  Node: Node,
  NodeCheckbox: NodeCheckbox
});
var _default = exports["default"] = TreeView;
