// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Table/Table.stories.js
import React from "react";

import { Badge } from "../Badge";
import { Stack } from "../Stack";
import { Table } from "./Table";

export default {
  title: "Data Display/Table",
  component: Table,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "Table component for displaying tabular data with various styling options.",
      },
    },
  },
  argTypes: {
    size: {
      control: "select",
      options: ["sm", "md", "lg"],
      description: "Size of the table",
    },
    variant: {
      control: "select",
      options: ["line", "outline"],
      description: "Visual variant",
    },
    striped: {
      control: "boolean",
      description: "Striped rows",
    },
    interactive: {
      control: "boolean",
      description: "Hover effect on rows",
    },
  },
};

const sampleData = [
  { id: 1, name: "Laptop", category: "Electronics", price: "$999.99", status: "In Stock" },
  { id: 2, name: "Coffee Maker", category: "Appliances", price: "$49.99", status: "Low Stock" },
  { id: 3, name: "Desk Chair", category: "Furniture", price: "$150.00", status: "In Stock" },
  { id: 4, name: "Smartphone", category: "Electronics", price: "$799.99", status: "Out of Stock" },
  { id: 5, name: "Headphones", category: "Accessories", price: "$199.99", status: "In Stock" },
];

export const Default = () => (
  <Table>
    <Table.Header>
      <Table.Row>
        <Table.ColumnHeader>Product</Table.ColumnHeader>
        <Table.ColumnHeader>Category</Table.ColumnHeader>
        <Table.ColumnHeader textAlign="end">Price</Table.ColumnHeader>
      </Table.Row>
    </Table.Header>
    <Table.Body>
      {sampleData.map((item) => (
        <Table.Row key={item.id}>
          <Table.Cell>{item.name}</Table.Cell>
          <Table.Cell>{item.category}</Table.Cell>
          <Table.Cell textAlign="end">{item.price}</Table.Cell>
        </Table.Row>
      ))}
    </Table.Body>
  </Table>
);

export const Striped = () => (
  <Table striped>
    <Table.Header>
      <Table.Row>
        <Table.ColumnHeader>Product</Table.ColumnHeader>
        <Table.ColumnHeader>Category</Table.ColumnHeader>
        <Table.ColumnHeader textAlign="end">Price</Table.ColumnHeader>
      </Table.Row>
    </Table.Header>
    <Table.Body>
      {sampleData.map((item) => (
        <Table.Row key={item.id}>
          <Table.Cell>{item.name}</Table.Cell>
          <Table.Cell>{item.category}</Table.Cell>
          <Table.Cell textAlign="end">{item.price}</Table.Cell>
        </Table.Row>
      ))}
    </Table.Body>
  </Table>
);

export const Interactive = () => (
  <Table interactive>
    <Table.Header>
      <Table.Row>
        <Table.ColumnHeader>Product</Table.ColumnHeader>
        <Table.ColumnHeader>Category</Table.ColumnHeader>
        <Table.ColumnHeader textAlign="end">Price</Table.ColumnHeader>
      </Table.Row>
    </Table.Header>
    <Table.Body>
      {sampleData.map((item) => (
        <Table.Row key={item.id}>
          <Table.Cell>{item.name}</Table.Cell>
          <Table.Cell>{item.category}</Table.Cell>
          <Table.Cell textAlign="end">{item.price}</Table.Cell>
        </Table.Row>
      ))}
    </Table.Body>
  </Table>
);

export const Sizes = () => (
  <Stack gap={8}>
    {["sm", "md", "lg"].map((size) => (
      <div key={size}>
        <strong>Size: {size}</strong>
        <Table size={size}>
          <Table.Header>
            <Table.Row>
              <Table.ColumnHeader>Product</Table.ColumnHeader>
              <Table.ColumnHeader>Price</Table.ColumnHeader>
            </Table.Row>
          </Table.Header>
          <Table.Body>
            <Table.Row>
              <Table.Cell>Laptop</Table.Cell>
              <Table.Cell>$999</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>Phone</Table.Cell>
              <Table.Cell>$799</Table.Cell>
            </Table.Row>
          </Table.Body>
        </Table>
      </div>
    ))}
  </Stack>
);

export const WithCaption = () => (
  <Table>
    <Table.Caption>Product Inventory - Last updated today</Table.Caption>
    <Table.Header>
      <Table.Row>
        <Table.ColumnHeader>Product</Table.ColumnHeader>
        <Table.ColumnHeader>Category</Table.ColumnHeader>
        <Table.ColumnHeader textAlign="end">Price</Table.ColumnHeader>
      </Table.Row>
    </Table.Header>
    <Table.Body>
      {sampleData.slice(0, 3).map((item) => (
        <Table.Row key={item.id}>
          <Table.Cell>{item.name}</Table.Cell>
          <Table.Cell>{item.category}</Table.Cell>
          <Table.Cell textAlign="end">{item.price}</Table.Cell>
        </Table.Row>
      ))}
    </Table.Body>
  </Table>
);

export const WithStatusBadges = () => (
  <Table>
    <Table.Header>
      <Table.Row>
        <Table.ColumnHeader>Product</Table.ColumnHeader>
        <Table.ColumnHeader>Category</Table.ColumnHeader>
        <Table.ColumnHeader>Status</Table.ColumnHeader>
        <Table.ColumnHeader textAlign="end">Price</Table.ColumnHeader>
      </Table.Row>
    </Table.Header>
    <Table.Body>
      {sampleData.map((item) => (
        <Table.Row key={item.id}>
          <Table.Cell>{item.name}</Table.Cell>
          <Table.Cell>{item.category}</Table.Cell>
          <Table.Cell>
            <Badge
              colorPalette={
                item.status === "In Stock"
                  ? "green"
                  : item.status === "Low Stock"
                    ? "yellow"
                    : "red"
              }
            >
              {item.status}
            </Badge>
          </Table.Cell>
          <Table.Cell textAlign="end">{item.price}</Table.Cell>
        </Table.Row>
      ))}
    </Table.Body>
  </Table>
);

export const Variants = () => (
  <Stack gap={8}>
    {["line", "outline"].map((variant) => (
      <div key={variant}>
        <strong>Variant: {variant}</strong>
        <Table variant={variant}>
          <Table.Header>
            <Table.Row>
              <Table.ColumnHeader>Product</Table.ColumnHeader>
              <Table.ColumnHeader>Price</Table.ColumnHeader>
            </Table.Row>
          </Table.Header>
          <Table.Body>
            <Table.Row>
              <Table.Cell>Laptop</Table.Cell>
              <Table.Cell>$999</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell>Phone</Table.Cell>
              <Table.Cell>$799</Table.Cell>
            </Table.Row>
          </Table.Body>
        </Table>
      </div>
    ))}
  </Stack>
);
