// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Flex/Flex.stories.js
import { Flex } from "./Flex";
import { Box } from "../Box";

export default {
  title: "Layout/Flex",
  component: Flex,
  tags: ["autodocs"],
  parameters: {
    layout: "padded",
    docs: {
      description: {
        component: "Flex component for creating flexible box layouts.",
      },
    },
  },
  argTypes: {
    gap: {
      control: "select",
      options: ["1", "2", "3", "4", "5", "6", "8", "10"],
      description: "Gap between items",
    },
    direction: {
      control: "select",
      options: ["row", "column", "row-reverse", "column-reverse"],
      description: "Flex direction",
    },
    align: {
      control: "select",
      options: ["flex-start", "center", "flex-end", "stretch", "baseline"],
      description: "Alignment of items",
    },
    justify: {
      control: "select",
      options: ["flex-start", "center", "flex-end", "space-between", "space-around", "space-evenly"],
      description: "Justification of items",
    },
    wrap: {
      control: "select",
      options: ["nowrap", "wrap", "wrap-reverse"],
      description: "Flex wrap behavior",
    },
  },
};

export const Basic = {
  args: {
    gap: "4",
  },
  render: (args) => (
    <Flex {...args}>
      <Box p="4" bg="blue.500" color="white" borderRadius="md">
        Item 1
      </Box>
      <Box p="4" bg="blue.500" color="white" borderRadius="md">
        Item 2
      </Box>
      <Box p="4" bg="blue.500" color="white" borderRadius="md">
        Item 3
      </Box>
    </Flex>
  ),
};

export const WithDirection = {
  args: {
    gap: "4",
    direction: "column",
  },
  render: (args) => (
    <Flex {...args}>
      <Box p="4" bg="green.500" color="white" borderRadius="md">
        Top
      </Box>
      <Box p="4" bg="green.500" color="white" borderRadius="md">
        Middle
      </Box>
      <Box p="4" bg="green.500" color="white" borderRadius="md">
        Bottom
      </Box>
    </Flex>
  ),
};

export const WithAlign = {
  args: {
    gap: "4",
    align: "center",
  },
  render: (args) => (
    <Flex h="32" bg="gray.100" p="4" borderRadius="md" {...args}>
      <Box p="2" bg="purple.500" color="white" borderRadius="md">
        Small
      </Box>
      <Box p="4" bg="purple.500" color="white" borderRadius="md">
        Medium
      </Box>
      <Box p="6" bg="purple.500" color="white" borderRadius="md">
        Large
      </Box>
    </Flex>
  ),
};

export const WithJustify = {
  args: {
    gap: "4",
    justify: "space-between",
  },
  render: (args) => (
    <Flex bg="gray.100" p="4" borderRadius="md" {...args}>
      <Box p="4" bg="teal.500" color="white" borderRadius="md">
        Start
      </Box>
      <Box p="4" bg="teal.500" color="white" borderRadius="md">
        Center
      </Box>
      <Box p="4" bg="teal.500" color="white" borderRadius="md">
        End
      </Box>
    </Flex>
  ),
};

export const WithWrap = {
  args: {
    gap: "4",
    wrap: "wrap",
  },
  render: (args) => (
    <Flex maxW="300px" bg="gray.100" p="4" borderRadius="md" {...args}>
      <Box p="4" bg="orange.500" color="white" borderRadius="md">
        Box 1
      </Box>
      <Box p="4" bg="orange.500" color="white" borderRadius="md">
        Box 2
      </Box>
      <Box p="4" bg="orange.500" color="white" borderRadius="md">
        Box 3
      </Box>
      <Box p="4" bg="orange.500" color="white" borderRadius="md">
        Box 4
      </Box>
    </Flex>
  ),
};

export const ResponsiveDirection = {
  args: {
    gap: "4",
  },
  render: (args) => (
    <Flex direction={{ base: "column", md: "row" }} bg="gray.100" p="4" borderRadius="md" {...args}>
      <Box p="4" flex="1" bg="pink.500" color="white" borderRadius="md" textAlign="center">
        Responsive 1
      </Box>
      <Box p="4" flex="1" bg="pink.500" color="white" borderRadius="md" textAlign="center">
        Responsive 2
      </Box>
      <Box p="4" flex="1" bg="pink.500" color="white" borderRadius="md" textAlign="center">
        Responsive 3
      </Box>
    </Flex>
  ),
};
