// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/HoverCard/HoverCard.stories.js
import React from "react";
import { HoverCard } from "./HoverCard";
import { Stack, HStack, VStack } from "../Stack";
import { Text } from "../Text";
import { Avatar } from "../Avatar";
import { Link } from "../Link";

export default {
  title: "Overlay/HoverCard",
  component: HoverCard,
  parameters: {
    layout: "centered",
  },
};

export const Basic = () => (
  <HoverCard>
    <HoverCard.Trigger asChild>
      <Link href="#">@johndoe</Link>
    </HoverCard.Trigger>
    <HoverCard.Positioner>
      <HoverCard.Content>
        <HoverCard.Arrow />
        <Stack gap="3" p="2">
          <Text fontWeight="semibold">John Doe</Text>
          <Text color="fg.muted" fontSize="sm">
            Software Engineer at Acme Inc.
          </Text>
        </Stack>
      </HoverCard.Content>
    </HoverCard.Positioner>
  </HoverCard>
);

export const UserProfile = () => (
  <HoverCard>
    <HoverCard.Trigger asChild>
      <Link href="#">@janesmith</Link>
    </HoverCard.Trigger>
    <HoverCard.Positioner>
      <HoverCard.Content maxW="300px">
        <HoverCard.Arrow />
        <Stack gap="4" p="3">
          <HStack gap="3">
            <Avatar name="Jane Smith" size="lg" />
            <VStack align="start" gap="0">
              <Text fontWeight="semibold">Jane Smith</Text>
              <Text color="fg.muted" fontSize="sm">
                @janesmith
              </Text>
            </VStack>
          </HStack>
          <Text fontSize="sm">
            Product Designer passionate about creating beautiful user experiences. Currently working
            on design systems.
          </Text>
          <HStack gap="4" fontSize="sm">
            <Text>
              <Text as="span" fontWeight="semibold">
                245
              </Text>{" "}
              Following
            </Text>
            <Text>
              <Text as="span" fontWeight="semibold">
                1.2K
              </Text>{" "}
              Followers
            </Text>
          </HStack>
        </Stack>
      </HoverCard.Content>
    </HoverCard.Positioner>
  </HoverCard>
);

export const WithDelay = () => (
  <HoverCard openDelay={500} closeDelay={200}>
    <HoverCard.Trigger asChild>
      <Link href="#">Hover with delay</Link>
    </HoverCard.Trigger>
    <HoverCard.Positioner>
      <HoverCard.Content>
        <HoverCard.Arrow />
        <Stack gap="2" p="2">
          <Text>Opens after 500ms</Text>
          <Text color="fg.muted" fontSize="sm">
            Closes after 200ms
          </Text>
        </Stack>
      </HoverCard.Content>
    </HoverCard.Positioner>
  </HoverCard>
);
