// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Blockquote/Blockquote.stories.js
import React from "react";
import { Blockquote } from "./Blockquote";
import { Stack } from "../Stack";

export default {
  title: "Typography/Blockquote",
  component: Blockquote,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Blockquote>
    <Blockquote.Content>The only way to do great work is to love what you do.</Blockquote.Content>
  </Blockquote>
);

export const WithCaption = () => (
  <Blockquote>
    <Blockquote.Content>The only way to do great work is to love what you do.</Blockquote.Content>
    <Blockquote.Caption>Steve Jobs</Blockquote.Caption>
  </Blockquote>
);

export const WithIcon = () => (
  <Blockquote>
    <Blockquote.Icon />
    <Blockquote.Content>The only way to do great work is to love what you do.</Blockquote.Content>
    <Blockquote.Caption>Steve Jobs</Blockquote.Caption>
  </Blockquote>
);

export const Variants = () => (
  <Stack gap="8">
    <Blockquote variant="plain">
      <Blockquote.Content>Plain variant blockquote</Blockquote.Content>
      <Blockquote.Caption>Author</Blockquote.Caption>
    </Blockquote>
    <Blockquote variant="solid">
      <Blockquote.Content>Solid variant blockquote</Blockquote.Content>
      <Blockquote.Caption>Author</Blockquote.Caption>
    </Blockquote>
    <Blockquote variant="subtle">
      <Blockquote.Content>Subtle variant blockquote</Blockquote.Content>
      <Blockquote.Caption>Author</Blockquote.Caption>
    </Blockquote>
  </Stack>
);

export const ColorPalettes = () => (
  <Stack gap="8">
    <Blockquote colorPalette="primary">
      <Blockquote.Content>Primary color palette</Blockquote.Content>
    </Blockquote>
    <Blockquote colorPalette="blue">
      <Blockquote.Content>Blue color palette</Blockquote.Content>
    </Blockquote>
    <Blockquote colorPalette="green">
      <Blockquote.Content>Green color palette</Blockquote.Content>
    </Blockquote>
  </Stack>
);
