// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Accordion/Accordion.stories.js
import React from "react";

import { Accordion } from "./Accordion";
import { Stack } from "../Stack";

export default {
  title: "Disclosure/Accordion",
  component: Accordion,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Accordion defaultValue={["item-1"]}>
    <Accordion.Item value="item-1">
      <Accordion.ItemTrigger>
        Section 1
        <Accordion.ItemIndicator />
      </Accordion.ItemTrigger>
      <Accordion.ItemContent>
        <Accordion.ItemBody>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt
          ut labore et dolore magna aliqua.
        </Accordion.ItemBody>
      </Accordion.ItemContent>
    </Accordion.Item>
    <Accordion.Item value="item-2">
      <Accordion.ItemTrigger>
        Section 2
        <Accordion.ItemIndicator />
      </Accordion.ItemTrigger>
      <Accordion.ItemContent>
        <Accordion.ItemBody>
          Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
          commodo consequat.
        </Accordion.ItemBody>
      </Accordion.ItemContent>
    </Accordion.Item>
    <Accordion.Item value="item-3">
      <Accordion.ItemTrigger>
        Section 3
        <Accordion.ItemIndicator />
      </Accordion.ItemTrigger>
      <Accordion.ItemContent>
        <Accordion.ItemBody>
          Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat
          nulla pariatur.
        </Accordion.ItemBody>
      </Accordion.ItemContent>
    </Accordion.Item>
  </Accordion>
);

export const Multiple = () => (
  <Accordion multiple defaultValue={["item-1", "item-2"]}>
    <Accordion.Item value="item-1">
      <Accordion.ItemTrigger>
        First Section
        <Accordion.ItemIndicator />
      </Accordion.ItemTrigger>
      <Accordion.ItemContent>
        <Accordion.ItemBody>Content for first section</Accordion.ItemBody>
      </Accordion.ItemContent>
    </Accordion.Item>
    <Accordion.Item value="item-2">
      <Accordion.ItemTrigger>
        Second Section
        <Accordion.ItemIndicator />
      </Accordion.ItemTrigger>
      <Accordion.ItemContent>
        <Accordion.ItemBody>Content for second section</Accordion.ItemBody>
      </Accordion.ItemContent>
    </Accordion.Item>
    <Accordion.Item value="item-3">
      <Accordion.ItemTrigger>
        Third Section
        <Accordion.ItemIndicator />
      </Accordion.ItemTrigger>
      <Accordion.ItemContent>
        <Accordion.ItemBody>Content for third section</Accordion.ItemBody>
      </Accordion.ItemContent>
    </Accordion.Item>
  </Accordion>
);

export const Collapsible = () => (
  <Accordion collapsible>
    <Accordion.Item value="item-1">
      <Accordion.ItemTrigger>
        Collapsible Section
        <Accordion.ItemIndicator />
      </Accordion.ItemTrigger>
      <Accordion.ItemContent>
        <Accordion.ItemBody>
          This accordion can be fully collapsed. Click the trigger again to close.
        </Accordion.ItemBody>
      </Accordion.ItemContent>
    </Accordion.Item>
  </Accordion>
);

export const FAQ = () => (
  <Accordion collapsible>
    <Accordion.Item value="q1">
      <Accordion.ItemTrigger>
        What is your return policy?
        <Accordion.ItemIndicator />
      </Accordion.ItemTrigger>
      <Accordion.ItemContent>
        <Accordion.ItemBody>
          We offer a 30-day return policy for all unused items in their original packaging.
        </Accordion.ItemBody>
      </Accordion.ItemContent>
    </Accordion.Item>
    <Accordion.Item value="q2">
      <Accordion.ItemTrigger>
        How long does shipping take?
        <Accordion.ItemIndicator />
      </Accordion.ItemTrigger>
      <Accordion.ItemContent>
        <Accordion.ItemBody>
          Standard shipping takes 5-7 business days. Express shipping is available for 2-3 day
          delivery.
        </Accordion.ItemBody>
      </Accordion.ItemContent>
    </Accordion.Item>
    <Accordion.Item value="q3">
      <Accordion.ItemTrigger>
        Do you offer international shipping?
        <Accordion.ItemIndicator />
      </Accordion.ItemTrigger>
      <Accordion.ItemContent>
        <Accordion.ItemBody>
          Yes, we ship to over 50 countries worldwide. Shipping rates and times vary by location.
        </Accordion.ItemBody>
      </Accordion.ItemContent>
    </Accordion.Item>
  </Accordion>
);

export const Variants = () => (
  <Stack gap="8">
    <Accordion variant="plain" collapsible>
      <Accordion.Item value="item-1">
        <Accordion.ItemTrigger>
          Plain Variant
          <Accordion.ItemIndicator />
        </Accordion.ItemTrigger>
        <Accordion.ItemContent>
          <Accordion.ItemBody>Plain variant content</Accordion.ItemBody>
        </Accordion.ItemContent>
      </Accordion.Item>
    </Accordion>
    <Accordion variant="subtle" collapsible>
      <Accordion.Item value="item-1">
        <Accordion.ItemTrigger>
          Subtle Variant
          <Accordion.ItemIndicator />
        </Accordion.ItemTrigger>
        <Accordion.ItemContent>
          <Accordion.ItemBody>Subtle variant content</Accordion.ItemBody>
        </Accordion.ItemContent>
      </Accordion.Item>
    </Accordion>
    <Accordion variant="enclosed" collapsible>
      <Accordion.Item value="item-1">
        <Accordion.ItemTrigger>
          Enclosed Variant
          <Accordion.ItemIndicator />
        </Accordion.ItemTrigger>
        <Accordion.ItemContent>
          <Accordion.ItemBody>Enclosed variant content</Accordion.ItemBody>
        </Accordion.ItemContent>
      </Accordion.Item>
    </Accordion>
  </Stack>
);
