// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Skeleton/Skeleton.stories.js
import React from "react";

import { Box } from "../Box";
import { Skeleton } from "./Skeleton";
import { Stack, HStack } from "../Stack";

export default {
  title: "Feedback/Skeleton",
  component: Skeleton,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component:
          "Skeleton components for displaying loading placeholders. Includes Skeleton, SkeletonText, and SkeletonCircle.",
      },
    },
  },
  argTypes: {
    loading: {
      control: "boolean",
      description: "Toggle loading state",
    },
    height: {
      control: "text",
      description: "Height of the skeleton",
    },
    width: {
      control: "text",
      description: "Width of the skeleton",
    },
  },
};

export const Default = {
  args: {
    height: "20px",
    width: "200px",
  },
};

export const BasicSkeleton = () => (
  <Stack gap={4}>
    <Skeleton height="20px" />
    <Skeleton height="20px" width="80%" />
    <Skeleton height="20px" width="60%" />
  </Stack>
);

export const SkeletonTextExample = () => (
  <Stack gap={6}>
    <Box>
      <p>Default (3 lines):</p>
      <Skeleton.Text />
    </Box>
    <Box>
      <p>5 lines:</p>
      <Skeleton.Text lines={5} />
    </Box>
    <Box>
      <p>Custom gap:</p>
      <Skeleton.Text lines={4} gap={4} />
    </Box>
  </Stack>
);

export const SkeletonCircleExample = () => (
  <HStack gap={4}>
    <Skeleton.Circle size="8" />
    <Skeleton.Circle size="10" />
    <Skeleton.Circle size="12" />
    <Skeleton.Circle size="16" />
  </HStack>
);

export const CardSkeleton = () => (
  <Box borderWidth="1px" borderRadius="lg" p={4} maxW="300px">
    <HStack gap={4} mb={4}>
      <Skeleton.Circle size="12" />
      <Stack gap={2} flex="1">
        <Skeleton height="4" width="120px" />
        <Skeleton height="3" width="80px" />
      </Stack>
    </HStack>
    <Skeleton.Text lines={3} />
  </Box>
);

export const ListSkeleton = () => (
  <Stack gap={4}>
    {[1, 2, 3].map((i) => (
      <HStack key={i} gap={4}>
        <Skeleton.Circle size="10" />
        <Stack gap={2} flex="1">
          <Skeleton height="4" />
          <Skeleton height="3" width="70%" />
        </Stack>
      </HStack>
    ))}
  </Stack>
);

export const WithLoadedContent = () => {
  const [loading, setLoading] = React.useState(true);

  React.useEffect(() => {
    const timer = setTimeout(() => setLoading(false), 2000);
    return () => clearTimeout(timer);
  }, []);

  return (
    <Stack gap={4}>
      <Box>
        <button onClick={() => setLoading(!loading)}>
          Toggle Loading: {loading ? "On" : "Off"}
        </button>
      </Box>
      <Skeleton loading={loading} height="100px" width="200px">
        <Box p={4} bg="blue.100" borderRadius="md">
          Content is loaded!
        </Box>
      </Skeleton>
    </Stack>
  );
};
