// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Highlight/Highlight.stories.js
import React from "react";

import { Highlight } from "./Highlight";
import { Stack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Typography/Highlight",
  component: Highlight,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Text>
    <Highlight query="spotlight">
      With the Highlight component you can spotlight words in text.
    </Highlight>
  </Text>
);

export const MultipleQueries = () => (
  <Text>
    <Highlight query={["spotlight", "emphasize"]}>
      With the Highlight component you can spotlight and emphasize words.
    </Highlight>
  </Text>
);

export const CustomStyles = () => (
  <Text>
    <Highlight
      query="important"
      styles={{ px: "1", py: "0.5", bg: "red.100", color: "red.800", borderRadius: "sm" }}
    >
      This is important text that needs attention.
    </Highlight>
  </Text>
);

export const SearchResults = () => (
  <Stack gap="2">
    <Text>
      <Highlight query="react" styles={{ bg: "yellow.200" }}>
        React is a JavaScript library for building user interfaces.
      </Highlight>
    </Text>
    <Text>
      <Highlight query="react" styles={{ bg: "yellow.200" }}>
        Learning React can help you build modern web applications.
      </Highlight>
    </Text>
  </Stack>
);
