// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/ScrollArea/index.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ScrollArea", {
  enumerable: true,
  get: function get() {
    return _ScrollArea.ScrollArea;
  }
});
Object.defineProperty(exports, "ScrollAreaContent", {
  enumerable: true,
  get: function get() {
    return _ScrollArea.ScrollAreaContent;
  }
});
Object.defineProperty(exports, "ScrollAreaCorner", {
  enumerable: true,
  get: function get() {
    return _ScrollArea.ScrollAreaCorner;
  }
});
Object.defineProperty(exports, "ScrollAreaScrollbar", {
  enumerable: true,
  get: function get() {
    return _ScrollArea.ScrollAreaScrollbar;
  }
});
Object.defineProperty(exports, "ScrollAreaThumb", {
  enumerable: true,
  get: function get() {
    return _ScrollArea.ScrollAreaThumb;
  }
});
Object.defineProperty(exports, "ScrollAreaViewport", {
  enumerable: true,
  get: function get() {
    return _ScrollArea.ScrollAreaViewport;
  }
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _ScrollArea["default"];
  }
});
var _ScrollArea = _interopRequireWildcard(require("./ScrollArea"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, "default": e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
