// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Separator/Separator.stories.js
import React from "react";

import { Separator } from "./Separator";
import { Stack, HStack } from "../Stack";
import { Box } from "../Box";

export default {
  title: "Layout/Separator",
  component: Separator,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Stack gap="4">
    <Box>First section</Box>
    <Separator />
    <Box>Second section</Box>
    <Separator />
    <Box>Third section</Box>
  </Stack>
);

export const Vertical = () => (
  <HStack gap="4" h="10">
    <Box>First</Box>
    <Separator orientation="vertical" />
    <Box>Second</Box>
    <Separator orientation="vertical" />
    <Box>Third</Box>
  </HStack>
);

export const WithSizes = () => (
  <Stack gap="6">
    <Stack gap="2">
      <Box color="fg.muted" fontSize="sm">
        Size: xs
      </Box>
      <Separator size="xs" />
    </Stack>
    <Stack gap="2">
      <Box color="fg.muted" fontSize="sm">
        Size: sm
      </Box>
      <Separator size="sm" />
    </Stack>
    <Stack gap="2">
      <Box color="fg.muted" fontSize="sm">
        Size: md
      </Box>
      <Separator size="md" />
    </Stack>
    <Stack gap="2">
      <Box color="fg.muted" fontSize="sm">
        Size: lg
      </Box>
      <Separator size="lg" />
    </Stack>
  </Stack>
);

export const WithVariants = () => (
  <Stack gap="6">
    <Stack gap="2">
      <Box color="fg.muted" fontSize="sm">
        Variant: solid
      </Box>
      <Separator variant="solid" />
    </Stack>
    <Stack gap="2">
      <Box color="fg.muted" fontSize="sm">
        Variant: dashed
      </Box>
      <Separator variant="dashed" />
    </Stack>
    <Stack gap="2">
      <Box color="fg.muted" fontSize="sm">
        Variant: dotted
      </Box>
      <Separator variant="dotted" />
    </Stack>
  </Stack>
);

export const WithLabel = () => (
  <Stack gap="4">
    <HStack>
      <Box flexShrink="0" color="fg.muted">
        Label (start)
      </Box>
      <Separator flex="1" />
    </HStack>

    <HStack>
      <Separator flex="1" />
      <Box flexShrink="0" color="fg.muted">
        Label (end)
      </Box>
    </HStack>

    <HStack>
      <Separator flex="1" />
      <Box flexShrink="0" color="fg.muted">
        Label (center)
      </Box>
      <Separator flex="1" />
    </HStack>
  </Stack>
);

export const CustomColor = () => (
  <Stack gap="4">
    <Separator borderColor="primary.500" />
    <Separator borderColor="red.500" />
    <Separator borderColor="green.500" />
  </Stack>
);
