// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Container/Container.stories.js
import React from "react";
import { Container } from "./Container";
import { Box } from "../Box";
import { Stack } from "../Stack";

export default {
  title: "Layout/Container",
  component: Container,
  parameters: {
    layout: "fullscreen",
  },
};

export const Basic = () => (
  <Container>
    <Box p="4" bg="bg.muted" borderRadius="md">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam consectetur, tortor in lacinia
      eleifend, dui nisl tristique nunc.
    </Box>
  </Container>
);

export const Fluid = () => (
  <Container fluid>
    <Box p="4" bg="bg.muted" borderRadius="md">
      This container is fluid and takes full width
    </Box>
  </Container>
);

export const WithMaxWidth = () => (
  <Stack gap="6">
    <Container maxW="sm">
      <Box p="4" bg="primary.500" color="white" borderRadius="md">
        maxW="sm"
      </Box>
    </Container>
    <Container maxW="md">
      <Box p="4" bg="primary.500" color="white" borderRadius="md">
        maxW="md"
      </Box>
    </Container>
    <Container maxW="lg">
      <Box p="4" bg="primary.500" color="white" borderRadius="md">
        maxW="lg"
      </Box>
    </Container>
    <Container maxW="xl">
      <Box p="4" bg="primary.500" color="white" borderRadius="md">
        maxW="xl"
      </Box>
    </Container>
  </Stack>
);

export const CenteredContent = () => (
  <Container centerContent>
    <Box p="4" bg="primary.500" color="white" borderRadius="md">
      This content is centered
    </Box>
  </Container>
);
