// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/NativeSelect/NativeSelect.stories.js
import React from "react";

import { NativeSelect } from "./NativeSelect";
import { Stack } from "../Stack";
import { Field } from "../Field";

export default {
  title: "Forms/NativeSelect",
  component: NativeSelect,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <NativeSelect maxW="300px">
    <NativeSelect.Field placeholder="Select option">
      <option value="option1">Option 1</option>
      <option value="option2">Option 2</option>
      <option value="option3">Option 3</option>
    </NativeSelect.Field>
    <NativeSelect.Indicator />
  </NativeSelect>
);

export const WithField = () => (
  <Field maxW="300px">
    <Field.Label>Country</Field.Label>
    <NativeSelect>
      <NativeSelect.Field placeholder="Select country">
        <option value="us">United States</option>
        <option value="uk">United Kingdom</option>
        <option value="ca">Canada</option>
        <option value="au">Australia</option>
      </NativeSelect.Field>
      <NativeSelect.Indicator />
    </NativeSelect>
  </Field>
);

export const Sizes = () => (
  <Stack gap="4" maxW="300px">
    <NativeSelect size="xs">
      <NativeSelect.Field>
        <option value="xs">Size XS</option>
      </NativeSelect.Field>
      <NativeSelect.Indicator />
    </NativeSelect>
    <NativeSelect size="sm">
      <NativeSelect.Field>
        <option value="sm">Size SM</option>
      </NativeSelect.Field>
      <NativeSelect.Indicator />
    </NativeSelect>
    <NativeSelect size="md">
      <NativeSelect.Field>
        <option value="md">Size MD</option>
      </NativeSelect.Field>
      <NativeSelect.Indicator />
    </NativeSelect>
    <NativeSelect size="lg">
      <NativeSelect.Field>
        <option value="lg">Size LG</option>
      </NativeSelect.Field>
      <NativeSelect.Indicator />
    </NativeSelect>
  </Stack>
);

export const Variants = () => (
  <Stack gap="4" maxW="300px">
    <NativeSelect variant="outline">
      <NativeSelect.Field>
        <option value="outline">Outline variant</option>
      </NativeSelect.Field>
      <NativeSelect.Indicator />
    </NativeSelect>
    <NativeSelect variant="subtle">
      <NativeSelect.Field>
        <option value="subtle">Subtle variant</option>
      </NativeSelect.Field>
      <NativeSelect.Indicator />
    </NativeSelect>
  </Stack>
);
