// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/PinInput/PinInput.stories.js
import { PinInput } from "./PinInput";
import { Stack } from "../Stack";

export default {
  title: "Forms/PinInput",
  component: PinInput,
  tags: ["autodocs"],
  parameters: {
    layout: "padded",
    docs: {
      description: {
        component: "PinInput component for entering PIN codes or OTP values.",
      },
    },
  },
  argTypes: {
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg"],
      description: "Size of the pin input fields",
    },
    mask: {
      control: "boolean",
      description: "Mask the input values",
    },
    otp: {
      control: "boolean",
      description: "Enable one-time password mode",
    },
    disabled: {
      control: "boolean",
      description: "Disable the input",
    },
  },
};

export const Basic = () => (
  <PinInput>
    <PinInput.Control>
      <PinInput.Field />
      <PinInput.Field />
      <PinInput.Field />
      <PinInput.Field />
    </PinInput.Control>
  </PinInput>
);

export const WithLabel = () => (
  <PinInput>
    <PinInput.Label>Enter OTP</PinInput.Label>
    <PinInput.Control>
      <PinInput.Field />
      <PinInput.Field />
      <PinInput.Field />
      <PinInput.Field />
    </PinInput.Control>
  </PinInput>
);

export const Sizes = () => (
  <Stack gap="6">
    <PinInput size="xs">
      <PinInput.Control>
        <PinInput.Field />
        <PinInput.Field />
        <PinInput.Field />
        <PinInput.Field />
      </PinInput.Control>
    </PinInput>
    <PinInput size="sm">
      <PinInput.Control>
        <PinInput.Field />
        <PinInput.Field />
        <PinInput.Field />
        <PinInput.Field />
      </PinInput.Control>
    </PinInput>
    <PinInput size="md">
      <PinInput.Control>
        <PinInput.Field />
        <PinInput.Field />
        <PinInput.Field />
        <PinInput.Field />
      </PinInput.Control>
    </PinInput>
    <PinInput size="lg">
      <PinInput.Control>
        <PinInput.Field />
        <PinInput.Field />
        <PinInput.Field />
        <PinInput.Field />
      </PinInput.Control>
    </PinInput>
  </Stack>
);

export const Masked = () => (
  <PinInput mask>
    <PinInput.Label>Enter PIN</PinInput.Label>
    <PinInput.Control>
      <PinInput.Field />
      <PinInput.Field />
      <PinInput.Field />
      <PinInput.Field />
    </PinInput.Control>
  </PinInput>
);

export const OneTimePassword = () => (
  <PinInput otp>
    <PinInput.Label>One-time password</PinInput.Label>
    <PinInput.Control>
      <PinInput.Field />
      <PinInput.Field />
      <PinInput.Field />
      <PinInput.Field />
      <PinInput.Field />
      <PinInput.Field />
    </PinInput.Control>
  </PinInput>
);
