// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Avatar/Avatar.stories.js
import { fn } from "@storybook/test";

import { Avatar } from "./Avatar";
import { Stack, HStack } from "../Stack";

export default {
  title: "Data Display/Avatar",
  component: Avatar,
  tags: ["autodocs"],
  parameters: {
    layout: "padded",
    docs: {
      description: {
        component: "Avatar component for displaying user profile images with fallback support.",
      },
    },
  },
  args: {
    onStatusChange: fn(),
  },
  argTypes: {
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg", "xl", "2xl"],
      description: "Size of the avatar",
    },
    shape: {
      control: "select",
      options: ["square", "rounded", "full"],
      description: "Shape of the avatar",
    },
    name: {
      control: "text",
      description: "Name for fallback initials",
    },
    src: {
      control: "text",
      description: "Image source URL",
    },
    colorPalette: {
      control: "select",
      options: ["gray", "red", "green", "blue", "yellow", "purple", "teal"],
      description: "Color palette for fallback",
    },
    onStatusChange: {
      action: "status changed",
      description: "Callback when image loading status changes (loading, loaded, error)",
    },
  },
};

export const Default = {
  args: {
    name: "John Doe",
    src: "https://i.pravatar.cc/300?u=1",
    size: "md",
  },
};

export const Sizes = {
  args: {
    name: "User",
    src: "https://i.pravatar.cc/300?u=sizes",
  },
  render: (args) => (
    <HStack gap={4} align="center">
      <Avatar {...args} size="xs" name="XS" src="https://i.pravatar.cc/300?u=xs" />
      <Avatar {...args} size="sm" name="SM" src="https://i.pravatar.cc/300?u=sm" />
      <Avatar {...args} size="md" name="MD" src="https://i.pravatar.cc/300?u=md" />
      <Avatar {...args} size="lg" name="LG" src="https://i.pravatar.cc/300?u=lg" />
      <Avatar {...args} size="xl" name="XL" src="https://i.pravatar.cc/300?u=xl" />
      <Avatar {...args} size="2xl" name="2XL" src="https://i.pravatar.cc/300?u=2xl" />
    </HStack>
  ),
};

export const Shapes = {
  args: {
    size: "lg",
    name: "User",
  },
  render: (args) => (
    <HStack gap={4}>
      <Avatar {...args} shape="full" name="Full" src="https://i.pravatar.cc/300?u=full" />
      <Avatar {...args} shape="rounded" name="Rounded" src="https://i.pravatar.cc/300?u=rounded" />
      <Avatar {...args} shape="square" name="Square" src="https://i.pravatar.cc/300?u=square" />
    </HStack>
  ),
};

export const WithFallback = {
  args: {
    size: "md",
  },
  render: (args) => (
    <HStack gap={4}>
      <Avatar {...args} name="John Doe" />
      <Avatar {...args} name="Jane Smith" />
      <Avatar {...args} name="Bob Wilson" />
      <Avatar {...args} />
    </HStack>
  ),
};

export const BrokenImage = {
  args: {
    size: "md",
  },
  render: (args) => (
    <HStack gap={4}>
      <Avatar {...args} name="John Doe" src="https://broken-link.jpg" />
      <Avatar {...args} name="Jane Smith" src="https://another-broken.jpg" />
    </HStack>
  ),
};

export const ColorPalettes = {
  args: {
    size: "md",
  },
  render: (args) => (
    <HStack gap={4}>
      <Avatar {...args} colorPalette="red" name="Red" />
      <Avatar {...args} colorPalette="green" name="Green" />
      <Avatar {...args} colorPalette="blue" name="Blue" />
      <Avatar {...args} colorPalette="purple" name="Purple" />
      <Avatar {...args} colorPalette="teal" name="Teal" />
    </HStack>
  ),
};

export const Group = {
  args: {
    size: "md",
  },
  render: (args) => (
    <Avatar.Group gap="0" spaceX="-3" size={args.size}>
      <Avatar name="User 1" src="https://i.pravatar.cc/300?u=1" />
      <Avatar name="User 2" src="https://i.pravatar.cc/300?u=2" />
      <Avatar name="User 3" src="https://i.pravatar.cc/300?u=3" />
      <Avatar name="User 4" src="https://i.pravatar.cc/300?u=4" />
      <Avatar name="+5" fallback="+5" />
    </Avatar.Group>
  ),
};

export const GroupSizes = {
  args: {},
  render: () => (
    <Stack gap={4}>
      <Avatar.Group size="sm" gap="0" spaceX="-2">
        <Avatar src="https://i.pravatar.cc/300?u=sm1" />
        <Avatar src="https://i.pravatar.cc/300?u=sm2" />
        <Avatar src="https://i.pravatar.cc/300?u=sm3" />
      </Avatar.Group>
      <Avatar.Group size="md" gap="0" spaceX="-3">
        <Avatar src="https://i.pravatar.cc/300?u=md1" />
        <Avatar src="https://i.pravatar.cc/300?u=md2" />
        <Avatar src="https://i.pravatar.cc/300?u=md3" />
      </Avatar.Group>
      <Avatar.Group size="lg" gap="0" spaceX="-4">
        <Avatar src="https://i.pravatar.cc/300?u=lg1" />
        <Avatar src="https://i.pravatar.cc/300?u=lg2" />
        <Avatar src="https://i.pravatar.cc/300?u=lg3" />
      </Avatar.Group>
    </Stack>
  ),
};
