// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/theme/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.osmosTheme = exports.osmosSystem = void 0;
var _react = require("@chakra-ui/react");
var _animationStyles = require("./animation-styles");
var _breakpoints = require("./breakpoints");
var _globalCss = require("./global-css");
var _keyframes = require("./keyframes");
var _layerStyles = require("./layer-styles");
var _recipes = require("./recipes");
var _semanticTokens = require("./semantic-tokens");
var _slotRecipes = require("./slot-recipes");
var _textStyles = require("./text-styles");
var _tokens = require("./tokens");
var themeConfig = (0, _react.defineConfig)({
  preflight: true,
  cssVarsPrefix: "osmos",
  cssVarsRoot: ":where(:root, :host)",
  globalCss: _globalCss.globalCss,
  theme: {
    breakpoints: _breakpoints.breakpoints,
    keyframes: _keyframes.keyframes,
    tokens: _tokens.tokens,
    semanticTokens: _semanticTokens.semanticTokens,
    recipes: _recipes.recipes,
    slotRecipes: _slotRecipes.slotRecipes,
    textStyles: _textStyles.textStyles,
    layerStyles: _layerStyles.layerStyles,
    animationStyles: _animationStyles.animationStyles
  }
});
var osmosTheme = exports.osmosTheme = themeConfig;
var osmosSystem = exports.osmosSystem = (0, _react.createSystem)(_react.defaultBaseConfig, themeConfig);
