// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/theme/Theme.stories.js
import React from "react";
import { Box, Grid, HStack, VStack, Text, SimpleGrid } from "@chakra-ui/react";
import { osmosTheme } from "./index";

// Extract tokens from the theme
const tokens = osmosTheme.theme.tokens || {};
const semanticTokens = osmosTheme.theme.semanticTokens || {};

// Individual token categories
const colors = tokens.colors || {};
const fonts = tokens.fonts || {};
const fontSizes = tokens.fontSizes || {};
const fontWeights = tokens.fontWeights || {};
const spacing = tokens.spacing || {};
const radii = tokens.radii || {};
const shadows = tokens.shadows || {};
const animations = tokens.animations || {};
const aspectRatios = tokens.aspectRatios || {};
const blurs = tokens.blurs || {};
const borders = tokens.borders || {};
const cursor = tokens.cursor || {};
const durations = tokens.durations || {};
const easings = tokens.easings || {};
const letterSpacings = tokens.letterSpacings || {};
const lineHeights = tokens.lineHeights || {};
const sizes = tokens.sizes || {};
const zIndex = tokens.zIndex || {};

// Semantic tokens
const semanticColors = semanticTokens.colors || {};
const semanticShadows = semanticTokens.shadows || {};
const semanticRadii = semanticTokens.radii || {};

export default {
  title: "Theme/Design Tokens",
  tags: ["autodocs"],
};

// ============================================================================
// HELPER COMPONENTS
// ============================================================================

const ColorSwatch = ({ name, shades }) => (
  <VStack align="stretch" gap="0">
    <Text fontWeight="bold" fontSize="sm" mb="2" textTransform="capitalize">
      {name}
    </Text>
    {Object.entries(shades).map(([shade, colorValue]) => {
      const value = colorValue.value || colorValue;
      const isDark = parseInt(shade) >= 500;
      return (
        <HStack key={shade} bg={value} p="3" justify="space-between">
          <Text fontSize="xs" color={isDark ? "white" : "black"} fontWeight="medium">
            {shade}
          </Text>
          <Text fontSize="xs" color={isDark ? "white" : "black"} fontFamily="mono">
            {value}
          </Text>
        </HStack>
      );
    })}
  </VStack>
);

const TokenDisplay = ({ name, value, showValue = true }) => (
  <Box p="3" borderWidth="1px" borderRadius="md">
    <Text fontSize="sm" fontWeight="medium" mb="1">
      {name}
    </Text>
    {showValue && (
      <Text fontSize="xs" color="gray.500" fontFamily="mono">
        {value}
      </Text>
    )}
  </Box>
);

// ============================================================================
// DESIGN TOKENS STORIES
// ============================================================================

export const AllTokens = {
  render: () => (
    <VStack align="stretch" gap="12" pb="12">
      <Box>
        <Text fontSize="3xl" fontWeight="bold" mb="2">
          Design Tokens Overview
        </Text>
        <Text color="gray.600" mb="2">
          Complete design token reference for the Osmos UI component library
        </Text>
        <Text fontSize="sm" color="gray.500">
          Total Categories: {Object.keys(tokens).length}
        </Text>
      </Box>

      {/* Colors */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Colors ({Object.keys(colors).length})
        </Text>
        <SimpleGrid columns={{ base: 1, md: 2, lg: 3, xl: 4 }} gap="6">
          {Object.entries(colors).map(([name, shades]) => (
            <ColorSwatch key={name} name={name} shades={shades} />
          ))}
        </SimpleGrid>
      </Box>

      {/* Typography */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Typography
        </Text>
        <SimpleGrid columns={{ base: 1, lg: 2 }} gap="8">
          <Box>
            <Text fontSize="lg" fontWeight="semibold" mb="4">
              Font Sizes ({Object.keys(fontSizes).length})
            </Text>
            <VStack align="stretch" gap="1">
              {Object.entries(fontSizes).map(([name, size]) => (
                <HStack key={name} justify="space-between" py="1">
                  <Text fontSize="xs" color="gray.500" minW="16">
                    {name}
                  </Text>
                  <Text fontSize={size.value}>Aa</Text>
                  <Text fontSize="xs" color="gray.400" fontFamily="mono">
                    {size.value}
                  </Text>
                </HStack>
              ))}
            </VStack>
          </Box>

          <Box>
            <Text fontSize="lg" fontWeight="semibold" mb="4">
              Font Weights ({Object.keys(fontWeights).length})
            </Text>
            <VStack align="stretch" gap="1">
              {Object.entries(fontWeights).map(([name, weight]) => (
                <HStack key={name} justify="space-between" py="1">
                  <Text fontSize="xs" color="gray.500" minW="20">
                    {name}
                  </Text>
                  <Text fontWeight={weight.value}>Sample</Text>
                  <Text fontSize="xs" color="gray.400" fontFamily="mono">
                    {weight.value}
                  </Text>
                </HStack>
              ))}
            </VStack>
          </Box>
        </SimpleGrid>

        <Box mt="6">
          <Text fontSize="lg" fontWeight="semibold" mb="4">
            Font Families ({Object.keys(fonts).length})
          </Text>
          <VStack align="stretch" gap="4">
            {Object.entries(fonts).map(([name, font]) => (
              <Box key={name} p="4" borderWidth="1px" borderRadius="md">
                <Text fontSize="xs" color="gray.500" mb="1">
                  {name}
                </Text>
                <Text fontFamily={font.value} fontSize="lg">
                  The quick brown fox jumps over the lazy dog
                </Text>
                <Text fontSize="xs" color="gray.400" mt="2" fontFamily="mono">
                  {font.value}
                </Text>
              </Box>
            ))}
          </VStack>
        </Box>
      </Box>

      {/* Spacing */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Spacing ({Object.keys(spacing).length})
        </Text>
        <Grid templateColumns="repeat(auto-fill, minmax(140px, 1fr))" gap="4">
          {Object.entries(spacing).map(([name, space]) => (
            <VStack key={name} p="3" borderWidth="1px" borderRadius="md" align="stretch">
              <HStack justify="space-between">
                <Text fontSize="xs" fontWeight="medium">
                  {name}
                </Text>
                <Text fontSize="xs" color="gray.400" fontFamily="mono">
                  {space.value}
                </Text>
              </HStack>
              <Box bg="blue.500" h="3" w={space.value} borderRadius="sm" minW="2px" maxW="full" />
            </VStack>
          ))}
        </Grid>
      </Box>

      {/* Sizes */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Sizes ({Object.keys(sizes).length})
        </Text>
        <Grid templateColumns="repeat(auto-fill, minmax(140px, 1fr))" gap="4">
          {Object.entries(sizes)
            .slice(0, 20)
            .map(([name, size]) => (
              <TokenDisplay key={name} name={name} value={size.value} />
            ))}
        </Grid>
        {Object.keys(sizes).length > 20 && (
          <Text fontSize="sm" color="gray.500" mt="4">
            Showing 20 of {Object.keys(sizes).length} sizes
          </Text>
        )}
      </Box>

      {/* Border Radius */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Border Radius ({Object.keys(radii).length})
        </Text>
        <HStack gap="4" wrap="wrap">
          {Object.entries(radii).map(([name, radius]) => (
            <VStack key={name} p="4" borderWidth="1px" borderRadius="md">
              <Box w="12" h="12" bg="blue.500" borderRadius={radius.value} />
              <Text fontSize="xs" fontWeight="medium">
                {name}
              </Text>
              <Text fontSize="xs" color="gray.400" fontFamily="mono">
                {radius.value}
              </Text>
            </VStack>
          ))}
        </HStack>
      </Box>

      {/* Shadows */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Shadows ({Object.keys(shadows).length})
        </Text>
        <HStack gap="6" wrap="wrap">
          {Object.entries(shadows).map(([name, shadow]) => (
            <VStack key={name} p="4">
              <Box w="24" h="16" bg="white" borderRadius="md" boxShadow={shadow.value} />
              <Text fontSize="xs" fontWeight="medium" mt="2">
                {name}
              </Text>
            </VStack>
          ))}
        </HStack>
      </Box>

      {/* Animations */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Animations ({Object.keys(animations).length})
        </Text>
        <SimpleGrid columns={{ base: 1, md: 2 }} gap="4">
          {Object.entries(animations).map(([name, anim]) => (
            <TokenDisplay key={name} name={name} value={anim.value} />
          ))}
        </SimpleGrid>
      </Box>

      {/* Aspect Ratios */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Aspect Ratios ({Object.keys(aspectRatios).length})
        </Text>
        <HStack gap="4" wrap="wrap">
          {Object.entries(aspectRatios).map(([name, ratio]) => (
            <VStack key={name} p="4" borderWidth="1px" borderRadius="md">
              <Box
                w="24"
                bg="blue.500"
                borderRadius="md"
                style={{ aspectRatio: ratio.value.replace(" / ", "/") }}
              />
              <Text fontSize="xs" fontWeight="medium">
                {name}
              </Text>
              <Text fontSize="xs" color="gray.400" fontFamily="mono">
                {ratio.value}
              </Text>
            </VStack>
          ))}
        </HStack>
      </Box>

      {/* Blurs */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Blurs ({Object.keys(blurs).length})
        </Text>
        <HStack gap="6" wrap="wrap">
          {Object.entries(blurs).map(([name, blur]) => (
            <VStack key={name} p="4" borderWidth="1px" borderRadius="md">
              <Box
                w="24"
                h="16"
                bg="linear-gradient(135deg, #667eea 0%, #764ba2 100%)"
                borderRadius="md"
                style={{ backdropFilter: `blur(${blur.value})` }}
              >
                <Box
                  w="full"
                  h="full"
                  bg="white"
                  opacity="0.3"
                  borderRadius="md"
                  style={{ backdropFilter: `blur(${blur.value})` }}
                />
              </Box>
              <Text fontSize="xs" fontWeight="medium">
                {name}
              </Text>
              <Text fontSize="xs" color="gray.400" fontFamily="mono">
                {blur.value}
              </Text>
            </VStack>
          ))}
        </HStack>
      </Box>

      {/* Borders */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Borders ({Object.keys(borders).length})
        </Text>
        <SimpleGrid columns={{ base: 1, md: 2 }} gap="4">
          {Object.entries(borders).map(([name, border]) => (
            <Box key={name} p="4" borderWidth="1px" borderRadius="md">
              <HStack justify="space-between">
                <Text fontSize="sm" fontWeight="medium">
                  {name}
                </Text>
                <Text fontSize="xs" color="gray.400" fontFamily="mono">
                  {border.value}
                </Text>
              </HStack>
              <Box mt="3" p="4" borderWidth={border.value} borderColor="gray.400" borderRadius="md">
                Sample
              </Box>
            </Box>
          ))}
        </SimpleGrid>
      </Box>

      {/* Durations */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Durations ({Object.keys(durations).length})
        </Text>
        <SimpleGrid columns={{ base: 2, md: 3, lg: 4 }} gap="4">
          {Object.entries(durations).map(([name, duration]) => (
            <TokenDisplay key={name} name={name} value={duration.value} />
          ))}
        </SimpleGrid>
      </Box>

      {/* Easings */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Easings ({Object.keys(easings).length})
        </Text>
        <SimpleGrid columns={{ base: 1, md: 2 }} gap="4">
          {Object.entries(easings).map(([name, easing]) => (
            <TokenDisplay key={name} name={name} value={easing.value} />
          ))}
        </SimpleGrid>
      </Box>

      {/* Letter Spacings */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Letter Spacings ({Object.keys(letterSpacings).length})
        </Text>
        <VStack align="stretch" gap="3">
          {Object.entries(letterSpacings).map(([name, spacing]) => (
            <Box key={name} p="4" borderWidth="1px" borderRadius="md">
              <HStack justify="space-between" mb="2">
                <Text fontSize="sm" fontWeight="medium">
                  {name}
                </Text>
                <Text fontSize="xs" color="gray.400" fontFamily="mono">
                  {spacing.value}
                </Text>
              </HStack>
              <Text fontSize="lg" letterSpacing={spacing.value}>
                The quick brown fox jumps over the lazy dog
              </Text>
            </Box>
          ))}
        </VStack>
      </Box>

      {/* Line Heights */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Line Heights ({Object.keys(lineHeights).length})
        </Text>
        <SimpleGrid columns={{ base: 1, md: 2 }} gap="4">
          {Object.entries(lineHeights).map(([name, height]) => (
            <Box key={name} p="4" borderWidth="1px" borderRadius="md">
              <HStack justify="space-between" mb="2">
                <Text fontSize="sm" fontWeight="medium">
                  {name}
                </Text>
                <Text fontSize="xs" color="gray.400" fontFamily="mono">
                  {height.value}
                </Text>
              </HStack>
              <Text lineHeight={height.value}>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor
                incididunt ut labore et dolore magna aliqua.
              </Text>
            </Box>
          ))}
        </SimpleGrid>
      </Box>

      {/* Z-Index */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Z-Index ({Object.keys(zIndex).length})
        </Text>
        <SimpleGrid columns={{ base: 2, md: 3, lg: 4 }} gap="4">
          {Object.entries(zIndex).map(([name, z]) => (
            <TokenDisplay key={name} name={name} value={z.value} />
          ))}
        </SimpleGrid>
      </Box>

      {/* Cursor */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Cursor ({Object.keys(cursor).length})
        </Text>
        <SimpleGrid columns={{ base: 2, md: 3, lg: 4 }} gap="4">
          {Object.entries(cursor).map(([name, cur]) => (
            <Box
              key={name}
              p="4"
              borderWidth="1px"
              borderRadius="md"
              cursor={cur.value}
              _hover={{ bg: "gray.50" }}
            >
              <Text fontSize="sm" fontWeight="medium" mb="1">
                {name}
              </Text>
              <Text fontSize="xs" color="gray.500" fontFamily="mono">
                {cur.value}
              </Text>
              <Text fontSize="xs" color="gray.400" mt="2">
                Hover to see cursor
              </Text>
            </Box>
          ))}
        </SimpleGrid>
      </Box>
    </VStack>
  ),
};

// ============================================================================
// SEMANTIC TOKENS STORIES
// ============================================================================

const SemanticTokenDisplay = ({ name, tokenData }) => {
  const hasLightDark =
    tokenData.value &&
    typeof tokenData.value === "object" &&
    (tokenData.value._light || tokenData.value._dark);

  return (
    <Box p="4" borderWidth="1px" borderRadius="md">
      <Text fontSize="sm" fontWeight="medium" mb="2">
        {name}
      </Text>
      {hasLightDark ? (
        <VStack align="stretch" gap="2">
          <HStack>
            <Text fontSize="xs" color="gray.500" minW="12">
              Light:
            </Text>
            <Text fontSize="xs" fontFamily="mono" color="gray.600">
              {tokenData.value._light}
            </Text>
          </HStack>
          <HStack>
            <Text fontSize="xs" color="gray.500" minW="12">
              Dark:
            </Text>
            <Text fontSize="xs" fontFamily="mono" color="gray.600">
              {tokenData.value._dark}
            </Text>
          </HStack>
        </VStack>
      ) : (
        <Text fontSize="xs" color="gray.500" fontFamily="mono">
          {tokenData.value || JSON.stringify(tokenData)}
        </Text>
      )}
    </Box>
  );
};

export const SemanticTokens = {
  render: () => (
    <VStack align="stretch" gap="12" pb="12">
      <Box>
        <Text fontSize="3xl" fontWeight="bold" mb="2">
          Semantic Tokens
        </Text>
        <Text color="gray.600" mb="2">
          Context-aware tokens that adapt to light and dark modes
        </Text>
        <Text fontSize="sm" color="gray.500">
          Total Categories: {Object.keys(semanticTokens).length}
        </Text>
      </Box>

      {/* Semantic Colors */}
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
          Semantic Colors ({Object.keys(semanticColors).length} categories)
        </Text>
        <VStack align="stretch" gap="8">
          {Object.entries(semanticColors).map(([category, tokens]) => (
            <Box key={category}>
              <Text fontSize="lg" fontWeight="semibold" mb="4" textTransform="capitalize">
                {category} ({Object.keys(tokens).length})
              </Text>
              <SimpleGrid columns={{ base: 1, md: 2, lg: 3 }} gap="4">
                {Object.entries(tokens).map(([name, tokenData]) => (
                  <SemanticTokenDisplay key={name} name={name} tokenData={tokenData} />
                ))}
              </SimpleGrid>
            </Box>
          ))}
        </VStack>
      </Box>

      {/* Semantic Shadows */}
      {Object.keys(semanticShadows).length > 0 && (
        <Box>
          <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
            Semantic Shadows ({Object.keys(semanticShadows).length})
          </Text>
          <SimpleGrid columns={{ base: 2, md: 3, lg: 4 }} gap="6">
            {Object.keys(semanticShadows).map((name) => (
              <VStack key={name} p="4">
                <Box w="24" h="16" bg="bg" borderRadius="md" boxShadow={name} />
                <Text fontSize="sm" fontWeight="medium" mt="2">
                  {name}
                </Text>
                <Text fontSize="xs" color="fg.muted" fontFamily="mono">
                  shadow={`"${name}"`}
                </Text>
              </VStack>
            ))}
          </SimpleGrid>
        </Box>
      )}

      {/* Semantic Radii */}
      {Object.keys(semanticRadii).length > 0 && (
        <Box>
          <Text fontSize="2xl" fontWeight="bold" mb="6" pb="2" borderBottomWidth="2px">
            Semantic Radii ({Object.keys(semanticRadii).length})
          </Text>
          <SimpleGrid columns={{ base: 1, md: 2, lg: 3 }} gap="4">
            {Object.entries(semanticRadii).map(([name, tokenData]) => (
              <SemanticTokenDisplay key={name} name={name} tokenData={tokenData} />
            ))}
          </SimpleGrid>
        </Box>
      )}
    </VStack>
  ),
};

// ============================================================================
// INDIVIDUAL TOKEN CATEGORY STORIES
// ============================================================================

export const Colors = {
  render: () => (
    <VStack align="stretch" gap="8">
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6">
          Color Palette ({Object.keys(colors).length})
        </Text>
        <SimpleGrid columns={{ base: 1, md: 2, lg: 3, xl: 4 }} gap="6">
          {Object.entries(colors).map(([name, shades]) => (
            <ColorSwatch key={name} name={name} shades={shades} />
          ))}
        </SimpleGrid>
      </Box>
    </VStack>
  ),
};

export const Typography = {
  render: () => (
    <VStack align="stretch" gap="8">
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6">
          Font Families ({Object.keys(fonts).length})
        </Text>
        <VStack align="stretch" gap="4">
          {Object.entries(fonts).map(([name, font]) => (
            <Box key={name} p="4" borderWidth="1px" borderRadius="md">
              <Text fontSize="xs" color="gray.500" mb="1">
                {name}
              </Text>
              <Text fontFamily={font.value} fontSize="lg">
                The quick brown fox jumps over the lazy dog
              </Text>
              <Text fontSize="xs" color="gray.400" mt="2" fontFamily="mono">
                {font.value}
              </Text>
            </Box>
          ))}
        </VStack>
      </Box>

      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6">
          Font Sizes ({Object.keys(fontSizes).length})
        </Text>
        <VStack align="stretch" gap="2">
          {Object.entries(fontSizes).map(([name, size]) => (
            <HStack key={name} justify="space-between" p="3" borderWidth="1px" borderRadius="md">
              <HStack gap="4">
                <Text fontSize="xs" color="gray.500" minW="12">
                  {name}
                </Text>
                <Text fontSize={size.value}>The quick brown fox</Text>
              </HStack>
              <Text fontSize="xs" color="gray.400" fontFamily="mono">
                {size.value}
              </Text>
            </HStack>
          ))}
        </VStack>
      </Box>

      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6">
          Font Weights ({Object.keys(fontWeights).length})
        </Text>
        <VStack align="stretch" gap="2">
          {Object.entries(fontWeights).map(([name, weight]) => (
            <HStack key={name} justify="space-between" p="3" borderWidth="1px" borderRadius="md">
              <HStack gap="4">
                <Text fontSize="xs" color="gray.500" minW="20">
                  {name}
                </Text>
                <Text fontWeight={weight.value} fontSize="lg">
                  The quick brown fox jumps over the lazy dog
                </Text>
              </HStack>
              <Text fontSize="xs" color="gray.400" fontFamily="mono">
                {weight.value}
              </Text>
            </HStack>
          ))}
        </VStack>
      </Box>
    </VStack>
  ),
};

export const Spacing = {
  render: () => (
    <VStack align="stretch" gap="8">
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="6">
          Spacing Scale ({Object.keys(spacing).length})
        </Text>
        <VStack align="stretch" gap="2">
          {Object.entries(spacing).map(([name, space]) => (
            <HStack key={name} gap="4" p="2" borderWidth="1px" borderRadius="md">
              <Text fontSize="xs" color="gray.500" minW="8" textAlign="right">
                {name}
              </Text>
              <Box bg="blue.500" h="6" w={space.value} borderRadius="sm" />
              <Text fontSize="xs" color="gray.400" fontFamily="mono">
                {space.value}
              </Text>
            </HStack>
          ))}
        </VStack>
      </Box>
    </VStack>
  ),
};

export const Shadows = {
  render: () => (
    <VStack align="stretch" gap="8">
      <Box>
        <Text fontSize="2xl" fontWeight="bold" mb="2">
          Shadows (Semantic Tokens)
        </Text>
        <Text color="fg.muted" mb="6">
          Shadows adapt to light and dark modes automatically
        </Text>
        <SimpleGrid columns={{ base: 2, md: 3, lg: 4 }} gap="8">
          {Object.entries(semanticShadows).map(([name]) => (
            <VStack key={name} gap="3">
              <Box
                w="32"
                h="24"
                bg="bg"
                borderRadius="lg"
                boxShadow={name}
                display="flex"
                alignItems="center"
                justifyContent="center"
              >
                <Text fontSize="xs" color="fg.muted">
                  {name}
                </Text>
              </Box>
              <VStack gap="0">
                <Text fontSize="sm" fontWeight="medium">
                  {name}
                </Text>
                <Text fontSize="xs" color="fg.muted" fontFamily="mono">
                  boxShadow="{name}"
                </Text>
              </VStack>
            </VStack>
          ))}
        </SimpleGrid>
      </Box>

      <Box>
        <Text fontSize="lg" fontWeight="semibold" mb="4">
          Shadow Scale Reference
        </Text>
        <VStack align="stretch" gap="4">
          {Object.entries(semanticShadows).map(([name, tokenData]) => (
            <Box key={name} p="4" borderWidth="1px" borderRadius="md">
              <HStack justify="space-between" mb="3">
                <Text fontWeight="medium">{name}</Text>
                <Box w="16" h="10" bg="bg" borderRadius="md" boxShadow={name} />
              </HStack>
              <VStack align="stretch" gap="1">
                <HStack>
                  <Text fontSize="xs" color="fg.muted" minW="12">
                    Light:
                  </Text>
                  <Text fontSize="xs" fontFamily="mono" color="fg.muted">
                    {tokenData.value?._light || "N/A"}
                  </Text>
                </HStack>
                <HStack>
                  <Text fontSize="xs" color="fg.muted" minW="12">
                    Dark:
                  </Text>
                  <Text fontSize="xs" fontFamily="mono" color="fg.muted">
                    {tokenData.value?._dark || "N/A"}
                  </Text>
                </HStack>
              </VStack>
            </Box>
          ))}
        </VStack>
      </Box>
    </VStack>
  ),
};
