// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Tooltip/Tooltip.stories.js
import React from "react";

import { Tooltip } from "./Tooltip";
import { Button } from "../Button";
import { Stack, HStack } from "../Stack";

export default {
  title: "Overlay/Tooltip",
  component: Tooltip,
  parameters: {
    layout: "centered",
  },
};

export const Basic = () => (
  <Tooltip>
    <Tooltip.Trigger asChild>
      <Button variant="outline">Hover me</Button>
    </Tooltip.Trigger>
    <Tooltip.Positioner>
      <Tooltip.Content>
        <Tooltip.Arrow />
        This is a tooltip
      </Tooltip.Content>
    </Tooltip.Positioner>
  </Tooltip>
);

export const Placements = () => (
  <Stack gap="8" align="center">
    <HStack gap="4">
      <Tooltip positioning={{ placement: "top-start" }}>
        <Tooltip.Trigger asChild>
          <Button size="sm">Top Start</Button>
        </Tooltip.Trigger>
        <Tooltip.Positioner>
          <Tooltip.Content>Top Start</Tooltip.Content>
        </Tooltip.Positioner>
      </Tooltip>
      <Tooltip positioning={{ placement: "top" }}>
        <Tooltip.Trigger asChild>
          <Button size="sm">Top</Button>
        </Tooltip.Trigger>
        <Tooltip.Positioner>
          <Tooltip.Content>Top</Tooltip.Content>
        </Tooltip.Positioner>
      </Tooltip>
      <Tooltip positioning={{ placement: "top-end" }}>
        <Tooltip.Trigger asChild>
          <Button size="sm">Top End</Button>
        </Tooltip.Trigger>
        <Tooltip.Positioner>
          <Tooltip.Content>Top End</Tooltip.Content>
        </Tooltip.Positioner>
      </Tooltip>
    </HStack>
    <HStack gap="4">
      <Tooltip positioning={{ placement: "bottom-start" }}>
        <Tooltip.Trigger asChild>
          <Button size="sm">Bottom Start</Button>
        </Tooltip.Trigger>
        <Tooltip.Positioner>
          <Tooltip.Content>Bottom Start</Tooltip.Content>
        </Tooltip.Positioner>
      </Tooltip>
      <Tooltip positioning={{ placement: "bottom" }}>
        <Tooltip.Trigger asChild>
          <Button size="sm">Bottom</Button>
        </Tooltip.Trigger>
        <Tooltip.Positioner>
          <Tooltip.Content>Bottom</Tooltip.Content>
        </Tooltip.Positioner>
      </Tooltip>
      <Tooltip positioning={{ placement: "bottom-end" }}>
        <Tooltip.Trigger asChild>
          <Button size="sm">Bottom End</Button>
        </Tooltip.Trigger>
        <Tooltip.Positioner>
          <Tooltip.Content>Bottom End</Tooltip.Content>
        </Tooltip.Positioner>
      </Tooltip>
    </HStack>
  </Stack>
);

export const WithArrow = () => (
  <Tooltip>
    <Tooltip.Trigger asChild>
      <Button variant="outline">With Arrow</Button>
    </Tooltip.Trigger>
    <Tooltip.Positioner>
      <Tooltip.Content>
        <Tooltip.Arrow />
        Tooltip with arrow
      </Tooltip.Content>
    </Tooltip.Positioner>
  </Tooltip>
);

export const Disabled = () => (
  <Tooltip disabled>
    <Tooltip.Trigger asChild>
      <Button variant="outline">Disabled Tooltip</Button>
    </Tooltip.Trigger>
    <Tooltip.Positioner>
      <Tooltip.Content>This won't show</Tooltip.Content>
    </Tooltip.Positioner>
  </Tooltip>
);
