// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/NumberInput/NumberInput.stories.js
import React from "react";

import { NumberInput } from "./NumberInput";
import { Stack } from "../Stack";
import { Field } from "../Field";

export default {
  title: "Forms/NumberInput",
  component: NumberInput,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <NumberInput defaultValue={10} maxW="200px">
    <NumberInput.Field />
    <NumberInput.Control>
      <NumberInput.IncrementTrigger />
      <NumberInput.DecrementTrigger />
    </NumberInput.Control>
  </NumberInput>
);

export const WithMinMax = () => (
  <NumberInput defaultValue={5} min={0} max={10} maxW="200px">
    <NumberInput.Field />
    <NumberInput.Control>
      <NumberInput.IncrementTrigger />
      <NumberInput.DecrementTrigger />
    </NumberInput.Control>
  </NumberInput>
);

export const WithStep = () => (
  <NumberInput defaultValue={0} step={5} maxW="200px">
    <NumberInput.Field />
    <NumberInput.Control>
      <NumberInput.IncrementTrigger />
      <NumberInput.DecrementTrigger />
    </NumberInput.Control>
  </NumberInput>
);

export const Sizes = () => (
  <Stack gap="4">
    <NumberInput size="xs" defaultValue={10} maxW="150px">
      <NumberInput.Field />
      <NumberInput.Control>
        <NumberInput.IncrementTrigger />
        <NumberInput.DecrementTrigger />
      </NumberInput.Control>
    </NumberInput>
    <NumberInput size="sm" defaultValue={10} maxW="150px">
      <NumberInput.Field />
      <NumberInput.Control>
        <NumberInput.IncrementTrigger />
        <NumberInput.DecrementTrigger />
      </NumberInput.Control>
    </NumberInput>
    <NumberInput size="md" defaultValue={10} maxW="150px">
      <NumberInput.Field />
      <NumberInput.Control>
        <NumberInput.IncrementTrigger />
        <NumberInput.DecrementTrigger />
      </NumberInput.Control>
    </NumberInput>
    <NumberInput size="lg" defaultValue={10} maxW="150px">
      <NumberInput.Field />
      <NumberInput.Control>
        <NumberInput.IncrementTrigger />
        <NumberInput.DecrementTrigger />
      </NumberInput.Control>
    </NumberInput>
  </Stack>
);

export const WithField = () => (
  <Field maxW="200px">
    <Field.Label>Quantity</Field.Label>
    <NumberInput defaultValue={1} min={1}>
      <NumberInput.Field />
      <NumberInput.Control>
        <NumberInput.IncrementTrigger />
        <NumberInput.DecrementTrigger />
      </NumberInput.Control>
    </NumberInput>
  </Field>
);
