// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Text/Text.stories.js
import React from "react";

import { Text } from "./Text";
import { Stack } from "../Stack";

export default {
  title: "Typography/Text",
  component: Text,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => <Text>This is basic text</Text>;

export const Sizes = () => (
  <Stack gap="2">
    <Text fontSize="xs">Extra small text (xs)</Text>
    <Text fontSize="sm">Small text (sm)</Text>
    <Text fontSize="md">Medium text (md)</Text>
    <Text fontSize="lg">Large text (lg)</Text>
    <Text fontSize="xl">Extra large text (xl)</Text>
    <Text fontSize="2xl">2XL text</Text>
    <Text fontSize="3xl">3XL text</Text>
    <Text fontSize="4xl">4XL text</Text>
  </Stack>
);

export const FontWeights = () => (
  <Stack gap="2">
    <Text fontWeight="light">Light weight</Text>
    <Text fontWeight="normal">Normal weight</Text>
    <Text fontWeight="medium">Medium weight</Text>
    <Text fontWeight="semibold">Semibold weight</Text>
    <Text fontWeight="bold">Bold weight</Text>
  </Stack>
);

export const Truncate = () => (
  <Text maxW="200px" truncate>
    This is a very long text that will be truncated when it exceeds the maximum width
  </Text>
);

export const LineClamp = () => (
  <Text maxW="300px" lineClamp={2}>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut
    labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
    laboris.
  </Text>
);

export const Colors = () => (
  <Stack gap="2">
    <Text color="fg">Default foreground</Text>
    <Text color="fg.muted">Muted foreground</Text>
    <Text color="primary.500">Primary color</Text>
    <Text color="red.500">Red color</Text>
    <Text color="green.500">Green color</Text>
  </Stack>
);

export const AsElement = () => (
  <Stack gap="2">
    <Text as="p">Paragraph element</Text>
    <Text as="span">Span element</Text>
    <Text as="label">Label element</Text>
    <Text as="i">Italic element</Text>
    <Text as="u">Underline element</Text>
    <Text as="s">Strikethrough element</Text>
  </Stack>
);
