// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Tag/Tag.stories.js
import React from "react";

import { Stack, HStack } from "../Stack";
import { Tag } from "./Tag";

export default {
  title: "Data Display/Tag",
  component: Tag,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component:
          "Tag component for displaying labels, categories, or keywords with optional close button.",
      },
    },
  },
  argTypes: {
    variant: {
      control: "select",
      options: ["subtle", "solid", "outline", "surface"],
      description: "Visual variant of the tag",
    },
    colorPalette: {
      control: "select",
      options: ["primary", "gray", "red", "green", "blue", "yellow", "purple", "teal"],
      description: "Color palette for the tag",
    },
    size: {
      control: "select",
      options: ["sm", "md", "lg"],
      description: "Size of the tag",
    },
    closable: {
      control: "boolean",
      description: "Show close button",
    },
  },
};

export const Default = {
  args: {
    children: "Tag",
  },
};

export const Variants = () => (
  <HStack gap={4}>
    <Tag variant="subtle">Subtle</Tag>
    <Tag variant="solid">Solid</Tag>
    <Tag variant="outline">Outline</Tag>
    <Tag variant="surface">Surface</Tag>
  </HStack>
);

export const ColorPalettes = () => (
  <Stack gap={4}>
    <HStack gap={2} wrap="wrap">
      <Tag colorPalette="gray">Gray</Tag>
      <Tag colorPalette="primary">Primary</Tag>
      <Tag colorPalette="red">Red</Tag>
      <Tag colorPalette="green">Green</Tag>
      <Tag colorPalette="blue">Blue</Tag>
    </HStack>
    <HStack gap={2} wrap="wrap">
      <Tag colorPalette="yellow">Yellow</Tag>
      <Tag colorPalette="purple">Purple</Tag>
      <Tag colorPalette="teal">Teal</Tag>
      <Tag colorPalette="orange">Orange</Tag>
    </HStack>
  </Stack>
);

export const Sizes = () => (
  <HStack gap={4} align="center">
    <Tag size="sm">Small</Tag>
    <Tag size="md">Medium</Tag>
    <Tag size="lg">Large</Tag>
  </HStack>
);

export const Closable = () => (
  <HStack gap={4}>
    <Tag closable onClose={() => alert("Closed!")}>
      Closable Tag
    </Tag>
    <Tag closable colorPalette="blue">
      Blue Closable
    </Tag>
    <Tag closable colorPalette="green">
      Green Closable
    </Tag>
  </HStack>
);

export const WithStartElement = () => (
  <HStack gap={4}>
    <Tag startElement={<span>📦</span>}>Package</Tag>
    <Tag startElement={<span>🏷️</span>}>Label</Tag>
    <Tag startElement={<span>⭐</span>}>Featured</Tag>
  </HStack>
);

export const WithEndElement = () => (
  <HStack gap={4}>
    <Tag endElement={<span>✓</span>}>Verified</Tag>
    <Tag endElement={<span>→</span>}>Next</Tag>
  </HStack>
);

export const AllVariantsAndColors = () => (
  <Stack gap={6}>
    {["subtle", "solid", "outline", "surface"].map((variant) => (
      <Stack key={variant} gap={2}>
        <strong>{variant}</strong>
        <HStack gap={2} wrap="wrap">
          {["gray", "primary", "red", "green", "blue", "purple"].map((color) => (
            <Tag key={`${variant}-${color}`} variant={variant} colorPalette={color}>
              {color}
            </Tag>
          ))}
        </HStack>
      </Stack>
    ))}
  </Stack>
);

export const CategoryTags = () => (
  <HStack gap={2} wrap="wrap">
    <Tag colorPalette="blue" variant="solid">
      Technology
    </Tag>
    <Tag colorPalette="green" variant="solid">
      Nature
    </Tag>
    <Tag colorPalette="purple" variant="solid">
      Art
    </Tag>
    <Tag colorPalette="orange" variant="solid">
      Food
    </Tag>
    <Tag colorPalette="teal" variant="solid">
      Travel
    </Tag>
  </HStack>
);
