// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Heading/Heading.stories.js
import React from "react";

import { Heading } from "./Heading";
import { Stack } from "../Stack";

export default {
  title: "Typography/Heading",
  component: Heading,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => <Heading>This is a heading</Heading>;

export const Sizes = () => (
  <Stack gap="4">
    <Heading size="xs">Heading (xs)</Heading>
    <Heading size="sm">Heading (sm)</Heading>
    <Heading size="md">Heading (md)</Heading>
    <Heading size="lg">Heading (lg)</Heading>
    <Heading size="xl">Heading (xl)</Heading>
    <Heading size="2xl">Heading (2xl)</Heading>
    <Heading size="3xl">Heading (3xl)</Heading>
    <Heading size="4xl">Heading (4xl)</Heading>
  </Stack>
);

export const AsElement = () => (
  <Stack gap="4">
    <Heading as="h1" size="2xl">
      H1 Heading
    </Heading>
    <Heading as="h2" size="xl">
      H2 Heading
    </Heading>
    <Heading as="h3" size="lg">
      H3 Heading
    </Heading>
    <Heading as="h4" size="md">
      H4 Heading
    </Heading>
    <Heading as="h5" size="sm">
      H5 Heading
    </Heading>
    <Heading as="h6" size="xs">
      H6 Heading
    </Heading>
  </Stack>
);

export const Truncate = () => (
  <Heading maxW="300px" truncate>
    This is a very long heading that will be truncated
  </Heading>
);

export const Colors = () => (
  <Stack gap="4">
    <Heading color="fg">Default heading</Heading>
    <Heading color="fg.muted">Muted heading</Heading>
    <Heading color="primary.500">Primary heading</Heading>
  </Stack>
);
