// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Grid/Grid.stories.js
import { Grid } from "./Grid";
import { Box } from "../Box";

export default {
  title: "Layout/Grid",
  component: Grid,
  tags: ["autodocs"],
  parameters: {
    layout: "padded",
    docs: {
      description: {
        component: "Grid component for creating CSS Grid layouts.",
      },
    },
  },
  argTypes: {
    gap: {
      control: "select",
      options: ["1", "2", "3", "4", "5", "6", "8", "10"],
      description: "Gap between grid items",
    },
    templateColumns: {
      control: "text",
      description: "CSS grid-template-columns value",
    },
    templateRows: {
      control: "text",
      description: "CSS grid-template-rows value",
    },
  },
};

export const BasicGrid = {
  args: {
    templateColumns: "repeat(3, 1fr)",
    gap: "6",
  },
  render: (args) => (
    <Grid {...args}>
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
    </Grid>
  ),
};

export const WithColSpan = {
  args: {
    templateColumns: "repeat(4, 1fr)",
    gap: "6",
  },
  render: (args) => (
    <Grid {...args}>
      <Grid.Item colSpan={2}>
        <Box h="20" bg="primary.500" borderRadius="md" />
      </Grid.Item>
      <Grid.Item colSpan={1}>
        <Box h="20" bg="primary.600" borderRadius="md" />
      </Grid.Item>
      <Grid.Item colSpan={1}>
        <Box h="20" bg="primary.700" borderRadius="md" />
      </Grid.Item>
    </Grid>
  ),
};

export const SpanningColumnsAndRows = {
  args: {
    gap: "4",
  },
  render: (args) => (
    <Grid h="200px" templateRows="repeat(2, 1fr)" templateColumns="repeat(5, 1fr)" {...args}>
      <Grid.Item rowSpan={2} colSpan={1} bg="primary.500" borderRadius="md" />
      <Grid.Item colSpan={2} bg="primary.600" borderRadius="md" />
      <Grid.Item colSpan={2} bg="primary.700" borderRadius="md" />
      <Grid.Item colSpan={4} bg="primary.800" borderRadius="md" />
    </Grid>
  ),
};

export const WithTemplateAreas = {
  args: {
    gap: "4",
  },
  render: (args) => (
    <Grid
      {...args}
      templateAreas={`
        "header header header"
        "sidebar main main"
        "footer footer footer"
      `}
      templateRows="auto 1fr auto"
      templateColumns="200px 1fr"
      h="400px"
    >
      <Grid.Item area="header" bg="primary.500" p="4" borderRadius="md">
        Header
      </Grid.Item>
      <Grid.Item area="sidebar" bg="primary.600" p="4" borderRadius="md">
        Sidebar
      </Grid.Item>
      <Grid.Item area="main" bg="primary.700" p="4" borderRadius="md">
        Main Content
      </Grid.Item>
      <Grid.Item area="footer" bg="primary.800" p="4" borderRadius="md">
        Footer
      </Grid.Item>
    </Grid>
  ),
};

export const BasicSimpleGrid = {
  args: {
    columns: 2,
    gap: "10",
  },
  render: (args) => (
    <Grid.Simple {...args}>
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
    </Grid.Simple>
  ),
};

export const ResponsiveColumns = {
  args: {
    gap: "6",
  },
  render: (args) => (
    <Grid.Simple columns={{ base: 1, sm: 2, md: 3, lg: 4 }} {...args}>
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
    </Grid.Simple>
  ),
};

export const AutoFit = {
  args: {
    minChildWidth: "200px",
    gap: "6",
  },
  render: (args) => (
    <Grid.Simple {...args}>
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
      <Box h="20" bg="primary.500" borderRadius="md" />
    </Grid.Simple>
  ),
};
