// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Menu/Menu.stories.js
import React, { useState } from "react";

import { Box } from "../Box";
import { Button } from "../Button";
import { Menu } from "./Menu";
import { Stack } from "../Stack";

export default {
  title: "Overlay/Menu",
  component: Menu,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "Menu component for displaying a list of actions or options in a dropdown.",
      },
    },
  },
};

export const Default = () => (
  <Menu>
    <Menu.Trigger>
      <Button variant="outline">Open Menu</Button>
    </Menu.Trigger>
    <Menu.Content>
      <Menu.Item value="new">New File</Menu.Item>
      <Menu.Item value="open">Open...</Menu.Item>
      <Menu.Item value="save">Save</Menu.Item>
      <Menu.Separator />
      <Menu.Item value="export">Export</Menu.Item>
    </Menu.Content>
  </Menu>
);

export const WithCommands = () => (
  <Menu>
    <Menu.Trigger>
      <Button variant="outline">Edit</Button>
    </Menu.Trigger>
    <Menu.Content>
      <Menu.Item value="cut">
        Cut <Menu.ItemCommand>⌘X</Menu.ItemCommand>
      </Menu.Item>
      <Menu.Item value="copy">
        Copy <Menu.ItemCommand>⌘C</Menu.ItemCommand>
      </Menu.Item>
      <Menu.Item value="paste">
        Paste <Menu.ItemCommand>⌘V</Menu.ItemCommand>
      </Menu.Item>
      <Menu.Separator />
      <Menu.Item value="select-all">
        Select All <Menu.ItemCommand>⌘A</Menu.ItemCommand>
      </Menu.Item>
    </Menu.Content>
  </Menu>
);

export const WithGroups = () => (
  <Menu>
    <Menu.Trigger>
      <Button variant="outline">View</Button>
    </Menu.Trigger>
    <Menu.Content>
      <Menu.ItemGroup title="Layout">
        <Menu.Item value="grid">Grid View</Menu.Item>
        <Menu.Item value="list">List View</Menu.Item>
      </Menu.ItemGroup>
      <Menu.Separator />
      <Menu.ItemGroup title="Sort By">
        <Menu.Item value="name">Name</Menu.Item>
        <Menu.Item value="date">Date</Menu.Item>
        <Menu.Item value="size">Size</Menu.Item>
      </Menu.ItemGroup>
    </Menu.Content>
  </Menu>
);

export const WithDisabledItems = () => (
  <Menu>
    <Menu.Trigger>
      <Button variant="outline">Actions</Button>
    </Menu.Trigger>
    <Menu.Content>
      <Menu.Item value="edit">Edit</Menu.Item>
      <Menu.Item value="duplicate">Duplicate</Menu.Item>
      <Menu.Item value="archive" disabled>
        Archive
      </Menu.Item>
      <Menu.Separator />
      <Menu.Item value="delete" color="red.500">
        Delete
      </Menu.Item>
    </Menu.Content>
  </Menu>
);

export const WithCheckboxItems = () => {
  const [checkedItems, setCheckedItems] = useState(["autosave"]);

  const toggleItem = (value) => {
    setCheckedItems((prev) =>
      prev.includes(value) ? prev.filter((item) => item !== value) : [...prev, value],
    );
  };

  return (
    <Menu>
      <Menu.Trigger>
        <Button variant="outline">Settings</Button>
      </Menu.Trigger>
      <Menu.Content>
        <Menu.ItemGroup title="Preferences">
          <Menu.CheckboxItem
            value="autosave"
            checked={checkedItems.includes("autosave")}
            onCheckedChange={() => toggleItem("autosave")}
          >
            Autosave
          </Menu.CheckboxItem>
          <Menu.CheckboxItem
            value="spellcheck"
            checked={checkedItems.includes("spellcheck")}
            onCheckedChange={() => toggleItem("spellcheck")}
          >
            Spellcheck
          </Menu.CheckboxItem>
          <Menu.CheckboxItem
            value="notifications"
            checked={checkedItems.includes("notifications")}
            onCheckedChange={() => toggleItem("notifications")}
          >
            Notifications
          </Menu.CheckboxItem>
        </Menu.ItemGroup>
      </Menu.Content>
    </Menu>
  );
};

export const WithRadioItems = () => {
  const [sortBy, setSortBy] = useState("name");

  return (
    <Menu>
      <Menu.Trigger>
        <Button variant="outline">Sort By: {sortBy}</Button>
      </Menu.Trigger>
      <Menu.Content>
        <Menu.RadioItemGroup value={sortBy} onValueChange={(e) => setSortBy(e.value)}>
          <Menu.RadioItem value="name">Name</Menu.RadioItem>
          <Menu.RadioItem value="date">Date</Menu.RadioItem>
          <Menu.RadioItem value="size">Size</Menu.RadioItem>
          <Menu.RadioItem value="type">Type</Menu.RadioItem>
        </Menu.RadioItemGroup>
      </Menu.Content>
    </Menu>
  );
};

export const Placements = () => (
  <Stack direction="row" gap={4} wrap="wrap">
    {["bottom-start", "bottom", "bottom-end", "top-start", "top", "top-end"].map((placement) => (
      <Menu key={placement} positioning={{ placement }}>
        <Menu.Trigger>
          <Button variant="outline" size="sm">
            {placement}
          </Button>
        </Menu.Trigger>
        <Menu.Content>
          <Menu.Item value="1">Item 1</Menu.Item>
          <Menu.Item value="2">Item 2</Menu.Item>
          <Menu.Item value="3">Item 3</Menu.Item>
        </Menu.Content>
      </Menu>
    ))}
  </Stack>
);

export const DangerItem = () => (
  <Menu>
    <Menu.Trigger>
      <Button variant="outline">More Actions</Button>
    </Menu.Trigger>
    <Menu.Content>
      <Menu.Item value="edit">Edit</Menu.Item>
      <Menu.Item value="share">Share</Menu.Item>
      <Menu.Item value="move">Move to folder</Menu.Item>
      <Menu.Separator />
      <Menu.Item value="delete" color="red.500" _hover={{ bg: "red.50" }}>
        Delete permanently
      </Menu.Item>
    </Menu.Content>
  </Menu>
);

export const WithIcons = () => (
  <Menu>
    <Menu.Trigger>
      <Button variant="outline">File</Button>
    </Menu.Trigger>
    <Menu.Content>
      <Menu.Item value="new">
        <Box as="span" mr={2}>
          📄
        </Box>
        New Document
      </Menu.Item>
      <Menu.Item value="open">
        <Box as="span" mr={2}>
          📂
        </Box>
        Open
      </Menu.Item>
      <Menu.Item value="save">
        <Box as="span" mr={2}>
          💾
        </Box>
        Save
      </Menu.Item>
      <Menu.Separator />
      <Menu.Item value="print">
        <Box as="span" mr={2}>
          🖨️
        </Box>
        Print
      </Menu.Item>
    </Menu.Content>
  </Menu>
);
