// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Menu/Menu.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Menu = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["children", "positioning"],
  _excluded2 = ["children"],
  _excluded3 = ["children", "portalled", "portalRef"],
  _excluded4 = ["children"],
  _excluded5 = ["title", "children"],
  _excluded6 = ["children"],
  _excluded7 = ["children"],
  _excluded8 = ["children"],
  _excluded9 = ["children"],
  _excluded0 = ["children"],
  _excluded1 = ["children"],
  _excluded10 = ["children"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
var Root = function Root(_ref) {
  var children = _ref.children,
    positioning = _ref.positioning,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.Root, _objectSpread(_objectSpread({
    positioning: positioning
  }, props), {}, {
    children: children
  }));
};
Root.displayName = "Menu";
var Trigger = /*#__PURE__*/_react["default"].forwardRef(function (_ref2, ref) {
  var children = _ref2.children,
    props = _objectWithoutProperties(_ref2, _excluded2);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.Trigger, _objectSpread(_objectSpread({
    ref: ref,
    asChild: true
  }, props), {}, {
    children: children
  }));
});
Trigger.displayName = "MenuTrigger";
var Content = /*#__PURE__*/_react["default"].forwardRef(function (_ref3, ref) {
  var children = _ref3.children,
    _ref3$portalled = _ref3.portalled,
    portalled = _ref3$portalled === void 0 ? true : _ref3$portalled,
    portalRef = _ref3.portalRef,
    props = _objectWithoutProperties(_ref3, _excluded3);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Portal, {
    disabled: !portalled,
    container: portalRef,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.Positioner, {
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.Content, _objectSpread(_objectSpread({
        ref: ref
      }, props), {}, {
        children: children
      }))
    })
  });
});
Content.displayName = "MenuContent";
var Item = /*#__PURE__*/_react["default"].forwardRef(function (_ref4, ref) {
  var children = _ref4.children,
    props = _objectWithoutProperties(_ref4, _excluded4);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.Item, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Item.displayName = "MenuItem";
var ItemGroup = /*#__PURE__*/_react["default"].forwardRef(function (_ref5, ref) {
  var title = _ref5.title,
    children = _ref5.children,
    props = _objectWithoutProperties(_ref5, _excluded5);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_react2.Menu.ItemGroup, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: [title && /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.ItemGroupLabel, {
      children: title
    }), children]
  }));
});
ItemGroup.displayName = "MenuItemGroup";
var ItemGroupLabel = /*#__PURE__*/_react["default"].forwardRef(function (_ref6, ref) {
  var children = _ref6.children,
    props = _objectWithoutProperties(_ref6, _excluded6);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.ItemGroupLabel, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
ItemGroupLabel.displayName = "MenuItemGroupLabel";
var Separator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.Separator, _objectSpread({
    ref: ref
  }, props));
});
Separator.displayName = "MenuSeparator";
var ItemCommand = /*#__PURE__*/_react["default"].forwardRef(function (_ref7, ref) {
  var children = _ref7.children,
    props = _objectWithoutProperties(_ref7, _excluded7);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.ItemCommand, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
ItemCommand.displayName = "MenuItemCommand";
var CheckboxItem = /*#__PURE__*/_react["default"].forwardRef(function (_ref8, ref) {
  var children = _ref8.children,
    props = _objectWithoutProperties(_ref8, _excluded8);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_react2.Menu.CheckboxItem, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: [children, /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.ItemIndicator, {})]
  }));
});
CheckboxItem.displayName = "MenuCheckboxItem";
var RadioItemGroup = /*#__PURE__*/_react["default"].forwardRef(function (_ref9, ref) {
  var children = _ref9.children,
    props = _objectWithoutProperties(_ref9, _excluded9);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.RadioItemGroup, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
RadioItemGroup.displayName = "MenuRadioItemGroup";
var RadioItem = /*#__PURE__*/_react["default"].forwardRef(function (_ref0, ref) {
  var children = _ref0.children,
    props = _objectWithoutProperties(_ref0, _excluded0);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_react2.Menu.RadioItem, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: [children, /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.ItemIndicator, {})]
  }));
});
RadioItem.displayName = "MenuRadioItem";
var ContextTrigger = /*#__PURE__*/_react["default"].forwardRef(function (_ref1, ref) {
  var children = _ref1.children,
    props = _objectWithoutProperties(_ref1, _excluded1);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.ContextTrigger, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
ContextTrigger.displayName = "MenuContextTrigger";
var TriggerItem = /*#__PURE__*/_react["default"].forwardRef(function (_ref10, ref) {
  var children = _ref10.children,
    props = _objectWithoutProperties(_ref10, _excluded10);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Menu.TriggerItem, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
TriggerItem.displayName = "MenuTriggerItem";

// Attach sub-components as properties
var Menu = exports.Menu = Object.assign(Root, {
  Trigger: Trigger,
  Content: Content,
  Item: Item,
  ItemGroup: ItemGroup,
  ItemGroupLabel: ItemGroupLabel,
  Separator: Separator,
  ItemCommand: ItemCommand,
  CheckboxItem: CheckboxItem,
  RadioItemGroup: RadioItemGroup,
  RadioItem: RadioItem,
  ContextTrigger: ContextTrigger,
  TriggerItem: TriggerItem
});
var _default = exports["default"] = Menu;
