// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Breadcrumb/Breadcrumb.stories.js
import React from "react";

import { Box } from "../Box";
import { Breadcrumb } from "./Breadcrumb";
import { Stack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Navigation/Breadcrumb",
  component: Breadcrumb,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "Breadcrumb component for showing hierarchical navigation paths.",
      },
    },
  },
  argTypes: {
    size: {
      control: "select",
      options: ["sm", "md", "lg"],
      description: "Size of the breadcrumb",
    },
    variant: {
      control: "select",
      options: ["plain", "underline"],
      description: "Visual variant of the breadcrumb",
    },
  },
};

export const Default = () => (
  <Breadcrumb>
    <Breadcrumb.Link href="#">Home</Breadcrumb.Link>
    <Breadcrumb.Link href="#">Products</Breadcrumb.Link>
    <Breadcrumb.CurrentLink>Category</Breadcrumb.CurrentLink>
  </Breadcrumb>
);

export const WithCustomSeparator = () => (
  <Stack gap={4}>
    <Breadcrumb separator="/">
      <Breadcrumb.Link href="#">Home</Breadcrumb.Link>
      <Breadcrumb.Link href="#">Library</Breadcrumb.Link>
      <Breadcrumb.CurrentLink>Data</Breadcrumb.CurrentLink>
    </Breadcrumb>
    <Breadcrumb separator=">">
      <Breadcrumb.Link href="#">Home</Breadcrumb.Link>
      <Breadcrumb.Link href="#">Library</Breadcrumb.Link>
      <Breadcrumb.CurrentLink>Data</Breadcrumb.CurrentLink>
    </Breadcrumb>
    <Breadcrumb separator="→">
      <Breadcrumb.Link href="#">Home</Breadcrumb.Link>
      <Breadcrumb.Link href="#">Library</Breadcrumb.Link>
      <Breadcrumb.CurrentLink>Data</Breadcrumb.CurrentLink>
    </Breadcrumb>
    <Breadcrumb separator="|">
      <Breadcrumb.Link href="#">Home</Breadcrumb.Link>
      <Breadcrumb.Link href="#">Library</Breadcrumb.Link>
      <Breadcrumb.CurrentLink>Data</Breadcrumb.CurrentLink>
    </Breadcrumb>
  </Stack>
);

export const Sizes = () => (
  <Stack gap={4}>
    {["sm", "md", "lg"].map((size) => (
      <Box key={size}>
        <Text fontWeight="bold" mb={2}>
          Size: {size}
        </Text>
        <Breadcrumb size={size}>
          <Breadcrumb.Link href="#">Home</Breadcrumb.Link>
          <Breadcrumb.Link href="#">Products</Breadcrumb.Link>
          <Breadcrumb.CurrentLink>Item</Breadcrumb.CurrentLink>
        </Breadcrumb>
      </Box>
    ))}
  </Stack>
);

export const Variants = () => (
  <Stack gap={4}>
    {["plain", "underline"].map((variant) => (
      <Box key={variant}>
        <Text fontWeight="bold" mb={2}>
          Variant: {variant}
        </Text>
        <Breadcrumb variant={variant}>
          <Breadcrumb.Link href="#">Home</Breadcrumb.Link>
          <Breadcrumb.Link href="#">Products</Breadcrumb.Link>
          <Breadcrumb.CurrentLink>Item</Breadcrumb.CurrentLink>
        </Breadcrumb>
      </Box>
    ))}
  </Stack>
);

export const LongPath = () => (
  <Breadcrumb>
    <Breadcrumb.Link href="#">Home</Breadcrumb.Link>
    <Breadcrumb.Link href="#">Products</Breadcrumb.Link>
    <Breadcrumb.Link href="#">Electronics</Breadcrumb.Link>
    <Breadcrumb.Link href="#">Computers</Breadcrumb.Link>
    <Breadcrumb.Link href="#">Laptops</Breadcrumb.Link>
    <Breadcrumb.CurrentLink>MacBook Pro</Breadcrumb.CurrentLink>
  </Breadcrumb>
);

export const WithEllipsis = () => (
  <Breadcrumb.Root>
    <Breadcrumb.List>
      <Breadcrumb.Item>
        <Breadcrumb.Link href="#">Home</Breadcrumb.Link>
      </Breadcrumb.Item>
      <Breadcrumb.Separator />
      <Breadcrumb.Item>
        <Breadcrumb.Ellipsis />
      </Breadcrumb.Item>
      <Breadcrumb.Separator />
      <Breadcrumb.Item>
        <Breadcrumb.Link href="#">Category</Breadcrumb.Link>
      </Breadcrumb.Item>
      <Breadcrumb.Separator />
      <Breadcrumb.Item>
        <Breadcrumb.CurrentLink>Current Page</Breadcrumb.CurrentLink>
      </Breadcrumb.Item>
    </Breadcrumb.List>
  </Breadcrumb.Root>
);

export const InPageHeader = () => (
  <Box borderBottom="1px" borderColor="gray.200" pb={4}>
    <Breadcrumb>
      <Breadcrumb.Link href="#">Dashboard</Breadcrumb.Link>
      <Breadcrumb.Link href="#">Settings</Breadcrumb.Link>
      <Breadcrumb.CurrentLink>Profile</Breadcrumb.CurrentLink>
    </Breadcrumb>
    <Text fontSize="2xl" fontWeight="bold" mt={2}>
      Profile Settings
    </Text>
  </Box>
);

export const WithIcons = () => (
  <Breadcrumb>
    <Breadcrumb.Link href="#">
      <Box as="span" mr={1}>
        🏠
      </Box>
      Home
    </Breadcrumb.Link>
    <Breadcrumb.Link href="#">
      <Box as="span" mr={1}>
        📁
      </Box>
      Documents
    </Breadcrumb.Link>
    <Breadcrumb.CurrentLink>
      <Box as="span" mr={1}>
        📄
      </Box>
      File.txt
    </Breadcrumb.CurrentLink>
  </Breadcrumb>
);
