// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/SegmentedControl/SegmentedControl.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.SegmentedControl = exports.SegmentGroup = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["children"],
  _excluded2 = ["children"],
  _excluded3 = ["children"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
var Root = /*#__PURE__*/_react["default"].forwardRef(function (_ref, ref) {
  var children = _ref.children,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.SegmentGroup.Root, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Root.displayName = "SegmentedControl";
var Item = /*#__PURE__*/_react["default"].forwardRef(function (_ref2, ref) {
  var children = _ref2.children,
    props = _objectWithoutProperties(_ref2, _excluded2);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.SegmentGroup.Item, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Item.displayName = "SegmentedControlItem";
var Items = function Items(_ref3) {
  var children = _ref3.children,
    props = _objectWithoutProperties(_ref3, _excluded3);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.SegmentGroup.Items, _objectSpread(_objectSpread({}, props), {}, {
    children: children
  }));
};
Items.displayName = "SegmentedControlItems";
var Indicator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.SegmentGroup.Indicator, _objectSpread({
    ref: ref
  }, props));
});
Indicator.displayName = "SegmentedControlIndicator";
var ItemHiddenInput = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.SegmentGroup.ItemHiddenInput, _objectSpread({
    ref: ref
  }, props));
});
ItemHiddenInput.displayName = "SegmentedControlItemHiddenInput";
var ItemText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.SegmentGroup.ItemText, _objectSpread({
    ref: ref
  }, props));
});
ItemText.displayName = "SegmentedControlItemText";

// Attach sub-components as properties
var SegmentedControl = exports.SegmentedControl = Object.assign(Root, {
  Item: Item,
  Items: Items,
  Indicator: Indicator,
  ItemHiddenInput: ItemHiddenInput,
  ItemText: ItemText
});

// SegmentGroup alias with same nested structure
var SegmentGroup = exports.SegmentGroup = Object.assign(Root, {
  Item: Item,
  Items: Items,
  Indicator: Indicator,
  ItemHiddenInput: ItemHiddenInput,
  ItemText: ItemText
});
var _default = exports["default"] = SegmentedControl;
