// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Badge/Badge.stories.js
import { Badge } from "./Badge";
import { Stack, HStack } from "../Stack";

export default {
  title: "Data Display/Badge",
  component: Badge,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "Badge component for displaying status, labels, or counts.",
      },
    },
  },
  argTypes: {
    variant: {
      control: "select",
      options: ["subtle", "solid", "outline", "surface"],
      description: "Visual variant of the badge",
    },
    colorPalette: {
      control: "select",
      options: ["primary", "gray", "red", "green", "blue", "yellow", "purple", "teal", "orange"],
      description: "Color palette for the badge",
    },
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg"],
      description: "Size of the badge",
    },
  },
};

export const Default = {
  args: {
    children: "Badge",
  },
};

export const Variants = () => (
  <HStack gap={4}>
    <Badge variant="subtle">Subtle</Badge>
    <Badge variant="solid">Solid</Badge>
    <Badge variant="outline">Outline</Badge>
    <Badge variant="surface">Surface</Badge>
  </HStack>
);

export const ColorPalettes = () => (
  <Stack gap={4}>
    <HStack gap={2}>
      <Badge colorPalette="gray">Gray</Badge>
      <Badge colorPalette="primary">Primary</Badge>
      <Badge colorPalette="red">Red</Badge>
      <Badge colorPalette="green">Green</Badge>
      <Badge colorPalette="blue">Blue</Badge>
    </HStack>
    <HStack gap={2}>
      <Badge colorPalette="yellow">Yellow</Badge>
      <Badge colorPalette="purple">Purple</Badge>
      <Badge colorPalette="teal">Teal</Badge>
      <Badge colorPalette="orange">Orange</Badge>
    </HStack>
  </Stack>
);

export const Sizes = () => (
  <HStack gap={4} align="center">
    <Badge size="xs">Extra Small</Badge>
    <Badge size="sm">Small</Badge>
    <Badge size="md">Medium</Badge>
    <Badge size="lg">Large</Badge>
  </HStack>
);

export const StatusBadges = () => (
  <HStack gap={4}>
    <Badge colorPalette="green" variant="solid">
      Active
    </Badge>
    <Badge colorPalette="yellow" variant="solid">
      Pending
    </Badge>
    <Badge colorPalette="red" variant="solid">
      Inactive
    </Badge>
    <Badge colorPalette="blue" variant="solid">
      New
    </Badge>
  </HStack>
);

export const AllVariantsAndColors = () => (
  <Stack gap={6}>
    {["subtle", "solid", "outline", "surface"].map((variant) => (
      <Stack key={variant} gap={2}>
        <strong>{variant}</strong>
        <HStack gap={2} wrap="wrap">
          {["gray", "primary", "red", "green", "blue", "yellow", "purple"].map((color) => (
            <Badge key={`${variant}-${color}`} variant={variant} colorPalette={color}>
              {color}
            </Badge>
          ))}
        </HStack>
      </Stack>
    ))}
  </Stack>
);
