// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Tabs/Tabs.stories.js
import React, { useState } from "react";

import { Box } from "../Box";
import { Stack } from "../Stack";
import { Tabs } from "./Tabs";
import { Text } from "../Text";

export default {
  title: "Navigation/Tabs",
  component: Tabs,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component:
          "Tabs component for organizing content into separate views where only one view is visible at a time.",
      },
    },
  },
  argTypes: {
    variant: {
      control: "select",
      options: ["line", "enclosed", "outline", "plain"],
      description: "Visual variant of the tabs",
    },
    size: {
      control: "select",
      options: ["sm", "md", "lg"],
      description: "Size of the tabs",
    },
    orientation: {
      control: "select",
      options: ["horizontal", "vertical"],
      description: "Orientation of the tabs",
    },
    fitted: {
      control: "boolean",
      description: "Whether tabs should stretch to fill container",
    },
  },
};

export const Default = () => (
  <Tabs defaultValue="tab1">
    <Tabs.List>
      <Tabs.Trigger value="tab1">Account</Tabs.Trigger>
      <Tabs.Trigger value="tab2">Security</Tabs.Trigger>
      <Tabs.Trigger value="tab3">Notifications</Tabs.Trigger>
    </Tabs.List>
    <Tabs.Content value="tab1">
      <Box p={4}>
        <Text>Manage your account settings and preferences.</Text>
      </Box>
    </Tabs.Content>
    <Tabs.Content value="tab2">
      <Box p={4}>
        <Text>Configure your security and privacy settings.</Text>
      </Box>
    </Tabs.Content>
    <Tabs.Content value="tab3">
      <Box p={4}>
        <Text>Manage your notification preferences.</Text>
      </Box>
    </Tabs.Content>
  </Tabs>
);

export const Variants = () => (
  <Stack gap={8}>
    {["line", "enclosed", "outline", "plain"].map((variant) => (
      <Box key={variant}>
        <Text fontWeight="bold" mb={2}>
          {variant}
        </Text>
        <Tabs defaultValue="tab1" variant={variant}>
          <Tabs.List>
            <Tabs.Trigger value="tab1">Tab 1</Tabs.Trigger>
            <Tabs.Trigger value="tab2">Tab 2</Tabs.Trigger>
            <Tabs.Trigger value="tab3">Tab 3</Tabs.Trigger>
          </Tabs.List>
          <Tabs.Content value="tab1">
            <Box p={4}>Content for Tab 1</Box>
          </Tabs.Content>
          <Tabs.Content value="tab2">
            <Box p={4}>Content for Tab 2</Box>
          </Tabs.Content>
          <Tabs.Content value="tab3">
            <Box p={4}>Content for Tab 3</Box>
          </Tabs.Content>
        </Tabs>
      </Box>
    ))}
  </Stack>
);

export const Sizes = () => (
  <Stack gap={8}>
    {["sm", "md", "lg"].map((size) => (
      <Box key={size}>
        <Text fontWeight="bold" mb={2}>
          Size: {size}
        </Text>
        <Tabs defaultValue="tab1" size={size}>
          <Tabs.List>
            <Tabs.Trigger value="tab1">Tab 1</Tabs.Trigger>
            <Tabs.Trigger value="tab2">Tab 2</Tabs.Trigger>
            <Tabs.Trigger value="tab3">Tab 3</Tabs.Trigger>
          </Tabs.List>
          <Tabs.Content value="tab1">
            <Box p={4}>Content for Tab 1</Box>
          </Tabs.Content>
          <Tabs.Content value="tab2">
            <Box p={4}>Content for Tab 2</Box>
          </Tabs.Content>
          <Tabs.Content value="tab3">
            <Box p={4}>Content for Tab 3</Box>
          </Tabs.Content>
        </Tabs>
      </Box>
    ))}
  </Stack>
);

export const Fitted = () => (
  <Box maxW="600px">
    <Tabs defaultValue="tab1" fitted>
      <Tabs.List>
        <Tabs.Trigger value="tab1">Overview</Tabs.Trigger>
        <Tabs.Trigger value="tab2">Details</Tabs.Trigger>
        <Tabs.Trigger value="tab3">Settings</Tabs.Trigger>
      </Tabs.List>
      <Tabs.Content value="tab1">
        <Box p={4}>
          <Text>Tabs stretch to fill the available width.</Text>
        </Box>
      </Tabs.Content>
      <Tabs.Content value="tab2">
        <Box p={4}>
          <Text>Each tab takes equal width.</Text>
        </Box>
      </Tabs.Content>
      <Tabs.Content value="tab3">
        <Box p={4}>
          <Text>Useful for navigation-style tabs.</Text>
        </Box>
      </Tabs.Content>
    </Tabs>
  </Box>
);

export const Vertical = () => (
  <Tabs defaultValue="tab1" orientation="vertical">
    <Tabs.List>
      <Tabs.Trigger value="tab1">Profile</Tabs.Trigger>
      <Tabs.Trigger value="tab2">Settings</Tabs.Trigger>
      <Tabs.Trigger value="tab3">Billing</Tabs.Trigger>
      <Tabs.Trigger value="tab4">Support</Tabs.Trigger>
    </Tabs.List>
    <Tabs.Content value="tab1">
      <Box p={4}>
        <Text>Profile content goes here.</Text>
      </Box>
    </Tabs.Content>
    <Tabs.Content value="tab2">
      <Box p={4}>
        <Text>Settings content goes here.</Text>
      </Box>
    </Tabs.Content>
    <Tabs.Content value="tab3">
      <Box p={4}>
        <Text>Billing content goes here.</Text>
      </Box>
    </Tabs.Content>
    <Tabs.Content value="tab4">
      <Box p={4}>
        <Text>Support content goes here.</Text>
      </Box>
    </Tabs.Content>
  </Tabs>
);

export const WithDisabledTab = () => (
  <Tabs defaultValue="tab1">
    <Tabs.List>
      <Tabs.Trigger value="tab1">Active</Tabs.Trigger>
      <Tabs.Trigger value="tab2">Available</Tabs.Trigger>
      <Tabs.Trigger value="tab3" disabled>
        Disabled
      </Tabs.Trigger>
    </Tabs.List>
    <Tabs.Content value="tab1">
      <Box p={4}>
        <Text>This tab is active.</Text>
      </Box>
    </Tabs.Content>
    <Tabs.Content value="tab2">
      <Box p={4}>
        <Text>This tab is available.</Text>
      </Box>
    </Tabs.Content>
    <Tabs.Content value="tab3">
      <Box p={4}>
        <Text>This tab is disabled.</Text>
      </Box>
    </Tabs.Content>
  </Tabs>
);

export const Controlled = () => {
  const [value, setValue] = useState("tab1");

  return (
    <Stack gap={4}>
      <Text>Current tab: {value}</Text>
      <Tabs value={value} onValueChange={(e) => setValue(e.value)}>
        <Tabs.List>
          <Tabs.Trigger value="tab1">First</Tabs.Trigger>
          <Tabs.Trigger value="tab2">Second</Tabs.Trigger>
          <Tabs.Trigger value="tab3">Third</Tabs.Trigger>
        </Tabs.List>
        <Tabs.Content value="tab1">
          <Box p={4}>First tab content</Box>
        </Tabs.Content>
        <Tabs.Content value="tab2">
          <Box p={4}>Second tab content</Box>
        </Tabs.Content>
        <Tabs.Content value="tab3">
          <Box p={4}>Third tab content</Box>
        </Tabs.Content>
      </Tabs>
    </Stack>
  );
};

export const WithIndicator = () => (
  <Tabs defaultValue="tab1" variant="plain">
    <Tabs.List>
      <Tabs.Trigger value="tab1">Overview</Tabs.Trigger>
      <Tabs.Trigger value="tab2">Analytics</Tabs.Trigger>
      <Tabs.Trigger value="tab3">Reports</Tabs.Trigger>
      <Tabs.Indicator />
    </Tabs.List>
    <Tabs.Content value="tab1">
      <Box p={4}>Overview content</Box>
    </Tabs.Content>
    <Tabs.Content value="tab2">
      <Box p={4}>Analytics content</Box>
    </Tabs.Content>
    <Tabs.Content value="tab3">
      <Box p={4}>Reports content</Box>
    </Tabs.Content>
  </Tabs>
);

export const LazyMounted = () => (
  <Tabs defaultValue="tab1" lazyMount unmountOnExit>
    <Tabs.List>
      <Tabs.Trigger value="tab1">Lazy Tab 1</Tabs.Trigger>
      <Tabs.Trigger value="tab2">Lazy Tab 2</Tabs.Trigger>
      <Tabs.Trigger value="tab3">Lazy Tab 3</Tabs.Trigger>
    </Tabs.List>
    <Tabs.Content value="tab1">
      <Box p={4}>
        <Text>This content is lazily mounted. Check the DOM to verify.</Text>
      </Box>
    </Tabs.Content>
    <Tabs.Content value="tab2">
      <Box p={4}>
        <Text>Content is mounted only when this tab is active.</Text>
      </Box>
    </Tabs.Content>
    <Tabs.Content value="tab3">
      <Box p={4}>
        <Text>Previous content is unmounted on exit.</Text>
      </Box>
    </Tabs.Content>
  </Tabs>
);
