// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Progress/Progress.stories.js
import React from "react";

import { Progress } from "./Progress";
import { Stack } from "../Stack";

export default {
  title: "Feedback/Progress",
  component: Progress,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "Progress component for showing completion status of a task or process.",
      },
    },
  },
  argTypes: {
    value: {
      control: { type: "range", min: 0, max: 100 },
      description: "Current progress value (0-100)",
    },
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg"],
      description: "Size of the progress bar",
    },
    colorPalette: {
      control: "select",
      options: ["primary", "gray", "red", "green", "blue", "yellow", "purple", "teal"],
      description: "Color palette for the progress bar",
    },
    label: {
      control: "text",
      description: "Label for the progress bar",
    },
    showValueLabel: {
      control: "boolean",
      description: "Show percentage value",
    },
  },
};

export const Default = {
  args: {
    value: 65,
  },
};

export const WithLabel = {
  args: {
    value: 45,
    label: "Upload Progress",
  },
};

export const WithValueLabel = {
  args: {
    value: 75,
    label: "Loading",
    showValueLabel: true,
  },
};

export const Sizes = () => (
  <Stack gap={4}>
    <Progress value={60} size="xs" />
    <Progress value={60} size="sm" />
    <Progress value={60} size="md" />
    <Progress value={60} size="lg" />
  </Stack>
);

export const ColorPalettes = () => (
  <Stack gap={4}>
    <Progress value={60} colorPalette="primary" />
    <Progress value={60} colorPalette="blue" />
    <Progress value={60} colorPalette="green" />
    <Progress value={60} colorPalette="red" />
    <Progress value={60} colorPalette="yellow" />
    <Progress value={60} colorPalette="purple" />
  </Stack>
);

export const DifferentValues = () => (
  <Stack gap={4}>
    <Progress value={0} label="0%" showValueLabel />
    <Progress value={25} label="25%" showValueLabel />
    <Progress value={50} label="50%" showValueLabel />
    <Progress value={75} label="75%" showValueLabel />
    <Progress value={100} label="Complete" showValueLabel />
  </Stack>
);

export const Striped = {
  args: {
    value: 65,
    striped: true,
  },
};

export const Animated = {
  args: {
    value: 65,
    striped: true,
    animated: true,
  },
};
