// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Modal/Modal.stories.js
import React, { useState } from "react";
import { fn, userEvent, within, expect } from "@storybook/test";

import { Button } from "../Button";
import { Input } from "../Input";
import { Stack } from "../Stack";
import { Text } from "../Text";
import { Modal } from "./Modal";

export default {
  title: "Overlay/Modal",
  component: Modal,
  tags: ["autodocs"],
  parameters: {
    layout: "padded",
    docs: {
      description: {
        component: "Modal (Dialog) component for displaying content in a layer above the page.",
      },
    },
  },
  args: {
    onOpenChange: fn(),
    onExitComplete: fn(),
  },
  argTypes: {
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg", "xl", "full"],
      description: "Size of the modal",
    },
    placement: {
      control: "select",
      options: ["top", "center", "bottom"],
      description: "Vertical placement of the modal",
    },
    scrollBehavior: {
      control: "select",
      options: ["inside", "outside"],
      description: "Scroll behavior for long content",
    },
    onOpenChange: {
      action: "open changed",
      description: "Callback when modal open state changes",
    },
    onExitComplete: {
      action: "exit complete",
      description: "Callback when modal exit animation completes",
    },
  },
};

export const Default = {
  args: {
    size: "md",
  },
  render: (args) => (
    <Modal {...args}>
      <Modal.Trigger>
        <Button>Open Modal</Button>
      </Modal.Trigger>
      <Modal.Content>
        <Modal.Header>
          <Modal.Title>Modal Title</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Text>
            This is the modal content. You can put any content here including forms, text, images,
            etc.
          </Text>
        </Modal.Body>
        <Modal.Footer>
          <Modal.ActionTrigger>
            <Button variant="outline">Cancel</Button>
          </Modal.ActionTrigger>
          <Button>Save</Button>
        </Modal.Footer>
        <Modal.CloseTrigger />
      </Modal.Content>
    </Modal>
  ),
  play: async ({ canvasElement, args }) => {
    const canvas = within(canvasElement);
    const openButton = canvas.getByRole("button", { name: /open modal/i });

    // Click to open modal
    await userEvent.click(openButton);
    await expect(args.onOpenChange).toHaveBeenCalled();
  },
};

export const WithDescription = {
  args: {
    size: "md",
  },
  render: (args) => (
    <Modal {...args}>
      <Modal.Trigger>
        <Button>Open Modal</Button>
      </Modal.Trigger>
      <Modal.Content>
        <Modal.Header>
          <Modal.Title>Confirm Action</Modal.Title>
          <Modal.Description>
            Please review the following information before proceeding.
          </Modal.Description>
        </Modal.Header>
        <Modal.Body>
          <Text>Are you sure you want to continue with this action?</Text>
        </Modal.Body>
        <Modal.Footer>
          <Modal.ActionTrigger>
            <Button variant="outline">Cancel</Button>
          </Modal.ActionTrigger>
          <Button colorPalette="primary">Confirm</Button>
        </Modal.Footer>
        <Modal.CloseTrigger />
      </Modal.Content>
    </Modal>
  ),
};

export const WithForm = {
  args: {
    size: "md",
  },
  render: (args) => (
    <Modal {...args}>
      <Modal.Trigger>
        <Button>Open Form Modal</Button>
      </Modal.Trigger>
      <Modal.Content>
        <Modal.Header>
          <Modal.Title>Sign Up</Modal.Title>
          <Modal.Description>Create a new account</Modal.Description>
        </Modal.Header>
        <Modal.Body>
          <Stack gap={4}>
            <Input label="First Name" placeholder="Enter first name" />
            <Input label="Last Name" placeholder="Enter last name" />
            <Input label="Email" placeholder="Enter email" type="email" />
          </Stack>
        </Modal.Body>
        <Modal.Footer>
          <Modal.ActionTrigger>
            <Button variant="outline">Cancel</Button>
          </Modal.ActionTrigger>
          <Button>Create Account</Button>
        </Modal.Footer>
        <Modal.CloseTrigger />
      </Modal.Content>
    </Modal>
  ),
};

export const Sizes = {
  args: {},
  render: (args) => (
    <Stack direction="row" gap={4} wrap="wrap">
      {["xs", "sm", "md", "lg", "xl"].map((size) => (
        <Modal key={size} size={size} {...args}>
          <Modal.Trigger>
            <Button variant="outline">{size.toUpperCase()}</Button>
          </Modal.Trigger>
          <Modal.Content>
            <Modal.Header>
              <Modal.Title>Size: {size}</Modal.Title>
            </Modal.Header>
            <Modal.Body>
              <Text>This is a {size} modal.</Text>
            </Modal.Body>
            <Modal.Footer>
              <Modal.ActionTrigger>
                <Button variant="outline">Close</Button>
              </Modal.ActionTrigger>
            </Modal.Footer>
            <Modal.CloseTrigger />
          </Modal.Content>
        </Modal>
      ))}
    </Stack>
  ),
};

export const Placements = {
  args: {},
  render: (args) => (
    <Stack direction="row" gap={4}>
      {["top", "center", "bottom"].map((placement) => (
        <Modal key={placement} placement={placement} {...args}>
          <Modal.Trigger>
            <Button variant="outline">{placement}</Button>
          </Modal.Trigger>
          <Modal.Content>
            <Modal.Header>
              <Modal.Title>Placement: {placement}</Modal.Title>
            </Modal.Header>
            <Modal.Body>
              <Text>Modal positioned at {placement}.</Text>
            </Modal.Body>
            <Modal.CloseTrigger />
          </Modal.Content>
        </Modal>
      ))}
    </Stack>
  ),
};

export const Controlled = () => {
  const [open, setOpen] = useState(false);

  return (
    <>
      <Button onClick={() => setOpen(true)}>Open Controlled Modal</Button>
      <Modal open={open} onClose={() => setOpen(false)}>
        <Modal.Content>
          <Modal.Header>
            <Modal.Title>Controlled Modal</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <Text>This modal is controlled externally.</Text>
          </Modal.Body>
          <Modal.Footer>
            <Button onClick={() => setOpen(false)}>Close</Button>
          </Modal.Footer>
        </Modal.Content>
      </Modal>
    </>
  );
};

export const AlertDialog = {
  args: {
    size: "md",
  },
  render: (args) => (
    <Modal {...args}>
      <Modal.Trigger>
        <Button colorPalette="red">Delete Item</Button>
      </Modal.Trigger>
      <Modal.Content>
        <Modal.Header>
          <Modal.Title>Are you sure?</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Text>
            This action cannot be undone. This will permanently delete your item and remove the data
            from our servers.
          </Text>
        </Modal.Body>
        <Modal.Footer>
          <Modal.ActionTrigger>
            <Button variant="outline">Cancel</Button>
          </Modal.ActionTrigger>
          <Button colorPalette="red">Delete</Button>
        </Modal.Footer>
        <Modal.CloseTrigger />
      </Modal.Content>
    </Modal>
  ),
};
