// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Modal/Modal.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Modal = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["open", "onOpenChange", "onClose", "size", "placement", "scrollBehavior", "motionPreset", "closeOnInteractOutside", "closeOnEscape", "children"],
  _excluded2 = ["children"],
  _excluded3 = ["children", "showBackdrop"],
  _excluded4 = ["children"],
  _excluded5 = ["children"],
  _excluded6 = ["children"],
  _excluded7 = ["children"],
  _excluded8 = ["children"],
  _excluded9 = ["children"],
  _excluded0 = ["children"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
var Root = function Root(_ref) {
  var open = _ref.open,
    onOpenChange = _ref.onOpenChange,
    onClose = _ref.onClose,
    size = _ref.size,
    _ref$placement = _ref.placement,
    placement = _ref$placement === void 0 ? "center" : _ref$placement,
    _ref$scrollBehavior = _ref.scrollBehavior,
    scrollBehavior = _ref$scrollBehavior === void 0 ? "inside" : _ref$scrollBehavior,
    _ref$motionPreset = _ref.motionPreset,
    motionPreset = _ref$motionPreset === void 0 ? "scale" : _ref$motionPreset,
    _ref$closeOnInteractO = _ref.closeOnInteractOutside,
    closeOnInteractOutside = _ref$closeOnInteractO === void 0 ? true : _ref$closeOnInteractO,
    _ref$closeOnEscape = _ref.closeOnEscape,
    closeOnEscape = _ref$closeOnEscape === void 0 ? true : _ref$closeOnEscape,
    children = _ref.children,
    props = _objectWithoutProperties(_ref, _excluded);
  var handleOpenChange = function handleOpenChange(details) {
    onOpenChange === null || onOpenChange === void 0 || onOpenChange(details);
    if (!details.open && onClose) {
      onClose();
    }
  };
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Dialog.Root, _objectSpread(_objectSpread({
    open: open,
    onOpenChange: handleOpenChange,
    size: size,
    placement: placement,
    scrollBehavior: scrollBehavior,
    motionPreset: motionPreset,
    closeOnInteractOutside: closeOnInteractOutside,
    closeOnEscape: closeOnEscape
  }, props), {}, {
    children: children
  }));
};
Root.displayName = "Modal";
var Trigger = /*#__PURE__*/_react["default"].forwardRef(function (_ref2, ref) {
  var children = _ref2.children,
    props = _objectWithoutProperties(_ref2, _excluded2);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Dialog.Trigger, _objectSpread(_objectSpread({
    ref: ref,
    asChild: true
  }, props), {}, {
    children: children
  }));
});
Trigger.displayName = "ModalTrigger";
var Content = /*#__PURE__*/_react["default"].forwardRef(function (_ref3, ref) {
  var children = _ref3.children,
    _ref3$showBackdrop = _ref3.showBackdrop,
    showBackdrop = _ref3$showBackdrop === void 0 ? true : _ref3$showBackdrop,
    props = _objectWithoutProperties(_ref3, _excluded3);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_react2.Portal, {
    children: [showBackdrop && /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Dialog.Backdrop, {}), /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Dialog.Positioner, {
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Dialog.Content, _objectSpread(_objectSpread({
        ref: ref
      }, props), {}, {
        children: children
      }))
    })]
  });
});
Content.displayName = "ModalContent";
var Header = /*#__PURE__*/_react["default"].forwardRef(function (_ref4, ref) {
  var children = _ref4.children,
    props = _objectWithoutProperties(_ref4, _excluded4);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Dialog.Header, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Header.displayName = "ModalHeader";
var Body = /*#__PURE__*/_react["default"].forwardRef(function (_ref5, ref) {
  var children = _ref5.children,
    props = _objectWithoutProperties(_ref5, _excluded5);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Dialog.Body, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Body.displayName = "ModalBody";
var Footer = /*#__PURE__*/_react["default"].forwardRef(function (_ref6, ref) {
  var children = _ref6.children,
    props = _objectWithoutProperties(_ref6, _excluded6);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Dialog.Footer, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Footer.displayName = "ModalFooter";
var Title = /*#__PURE__*/_react["default"].forwardRef(function (_ref7, ref) {
  var children = _ref7.children,
    props = _objectWithoutProperties(_ref7, _excluded7);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Dialog.Title, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Title.displayName = "ModalTitle";
var Description = /*#__PURE__*/_react["default"].forwardRef(function (_ref8, ref) {
  var children = _ref8.children,
    props = _objectWithoutProperties(_ref8, _excluded8);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Dialog.Description, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Description.displayName = "ModalDescription";
var CloseTrigger = /*#__PURE__*/_react["default"].forwardRef(function (_ref9, ref) {
  var children = _ref9.children,
    props = _objectWithoutProperties(_ref9, _excluded9);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Dialog.CloseTrigger, _objectSpread(_objectSpread({
    ref: ref,
    position: "absolute",
    top: "2",
    insetEnd: "2",
    asChild: true
  }, props), {}, {
    children: children || /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CloseButton, {
      size: "sm"
    })
  }));
});
CloseTrigger.displayName = "ModalCloseTrigger";
var ActionTrigger = /*#__PURE__*/_react["default"].forwardRef(function (_ref0, ref) {
  var children = _ref0.children,
    props = _objectWithoutProperties(_ref0, _excluded0);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Dialog.ActionTrigger, _objectSpread(_objectSpread({
    ref: ref,
    asChild: true
  }, props), {}, {
    children: children
  }));
});
ActionTrigger.displayName = "ModalActionTrigger";

// Attach sub-components as properties
var Modal = exports.Modal = Object.assign(Root, {
  Trigger: Trigger,
  Content: Content,
  Header: Header,
  Body: Body,
  Footer: Footer,
  Title: Title,
  Description: Description,
  CloseTrigger: CloseTrigger,
  ActionTrigger: ActionTrigger
});
var _default = exports["default"] = Modal;
