// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/theme/recipes/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.recipes = void 0;
var _badge = require("./badge");
var _button = require("./button");
var _code = require("./code");
var _container = require("./container");
var _heading = require("./heading");
var _input = require("./input");
var _inputAddon = require("./input-addon");
var _kbd = require("./kbd");
var _link = require("./link");
var _mark = require("./mark");
var _separator = require("./separator");
var _skeleton = require("./skeleton");
var _skipNavLink = require("./skip-nav-link");
var _spinner = require("./spinner");
var _textarea = require("./textarea");
var _icon = require("./icon");
var _checkmark = require("./checkmark");
var _radiomark = require("./radiomark");
var _colorSwatch = require("./color-swatch");
var recipes = exports.recipes = {
  badge: _badge.badgeRecipe,
  button: _button.buttonRecipe,
  code: _code.codeRecipe,
  container: _container.containerRecipe,
  heading: _heading.headingRecipe,
  input: _input.inputRecipe,
  inputAddon: _inputAddon.inputAddonRecipe,
  kbd: _kbd.kbdRecipe,
  link: _link.linkRecipe,
  mark: _mark.markRecipe,
  separator: _separator.separatorRecipe,
  skeleton: _skeleton.skeletonRecipe,
  skipNavLink: _skipNavLink.skipNavLinkRecipe,
  spinner: _spinner.spinnerRecipe,
  textarea: _textarea.textareaRecipe,
  icon: _icon.iconRecipe,
  checkmark: _checkmark.checkmarkRecipe,
  radiomark: _radiomark.radiomarkRecipe,
  colorSwatch: _colorSwatch.colorSwatchRecipe
};
