// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Button/Button.stories.js
import React from "react";
import { LuMail, LuArrowRight, LuPlus, LuDownload, LuTrash2 } from "react-icons/lu";

import Button from "./Button";
import { HStack, VStack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Components/Button",
  component: Button,
  tags: ["autodocs"],
  argTypes: {
    variant: {
      control: "select",
      options: ["solid", "subtle", "surface", "outline", "ghost", "plain"],
      description: "The visual style of the button",
      table: {
        defaultValue: { summary: "solid" },
      },
    },
    size: {
      control: "select",
      options: ["2xs", "xs", "sm", "md", "lg", "xl", "2xl"],
      description: "The size of the button",
      table: {
        defaultValue: { summary: "md" },
      },
    },
    colorPalette: {
      control: "select",
      options: [
        "primary",
        "secondary",
        "success",
        "warning",
        "danger",
        "info",
        "gray",
        "red",
        "orange",
        "yellow",
        "green",
        "teal",
        "blue",
        "cyan",
        "purple",
        "pink",
      ],
      description: "The color palette to use",
      table: {
        defaultValue: { summary: "primary" },
      },
    },
    disabled: {
      control: "boolean",
      description: "Whether the button is disabled",
      table: {
        defaultValue: { summary: false },
      },
    },
    loading: {
      control: "boolean",
      description: "Whether the button is in loading state",
      table: {
        defaultValue: { summary: false },
      },
    },
    loadingText: {
      control: "text",
      description: "Text to show while loading",
    },
    spinnerPlacement: {
      control: "select",
      options: ["start", "end"],
      description: "Placement of the loading spinner",
      table: {
        defaultValue: { summary: "start" },
      },
    },
  },
};

export const Default = {
  args: {
    children: "Button",
  },
};

export const Variants = {
  render: () => (
    <HStack gap="4" wrap="wrap">
      <Button variant="solid">Solid</Button>
      <Button variant="subtle">Subtle</Button>
      <Button variant="surface">Surface</Button>
      <Button variant="outline">Outline</Button>
      <Button variant="ghost">Ghost</Button>
      <Button variant="plain">Plain</Button>
    </HStack>
  ),
};

export const Sizes = {
  render: () => (
    <HStack gap="4" wrap="wrap" align="center">
      <Button size="2xs">2XS</Button>
      <Button size="xs">XS</Button>
      <Button size="sm">Small</Button>
      <Button size="md">Medium</Button>
      <Button size="lg">Large</Button>
      <Button size="xl">XL</Button>
      <Button size="2xl">2XL</Button>
    </HStack>
  ),
};

export const ThemeColors = {
  render: () => (
    <VStack gap="4" align="flex-start">
      <Text fontWeight="bold" fontSize="md" mb="2">
        Custom Theme Colors
      </Text>
      {["primary", "secondary", "success", "warning", "danger", "info"].map((color) => (
        <HStack key={color} gap="4" align="center">
          <Text minW="20" fontSize="sm" fontWeight="medium">
            {color}
          </Text>
          <Button colorPalette={color} variant="solid">
            Solid
          </Button>
          <Button colorPalette={color} variant="subtle">
            Subtle
          </Button>
          <Button colorPalette={color} variant="surface">
            Surface
          </Button>
          <Button colorPalette={color} variant="outline">
            Outline
          </Button>
          <Button colorPalette={color} variant="ghost">
            Ghost
          </Button>
        </HStack>
      ))}
    </VStack>
  ),
};

export const ColorPalettes = {
  render: () => (
    <VStack gap="4" align="flex-start">
      <Text fontWeight="bold" fontSize="md" mb="2">
        Standard Chakra Colors
      </Text>
      {["gray", "red", "orange", "yellow", "green", "teal", "blue", "cyan", "purple", "pink"].map(
        (color) => (
          <HStack key={color} gap="4" align="center">
            <Text minW="20" fontSize="sm">
              {color}
            </Text>
            <Button colorPalette={color} variant="solid">
              Solid
            </Button>
            <Button colorPalette={color} variant="subtle">
              Subtle
            </Button>
            <Button colorPalette={color} variant="surface">
              Surface
            </Button>
            <Button colorPalette={color} variant="outline">
              Outline
            </Button>
            <Button colorPalette={color} variant="ghost">
              Ghost
            </Button>
          </HStack>
        ),
      )}
    </VStack>
  ),
};

export const WithIcons = {
  render: () => (
    <HStack gap="4" wrap="wrap">
      <Button colorPalette="teal">
        <LuMail /> Email
      </Button>
      <Button colorPalette="blue" variant="outline">
        Download <LuDownload />
      </Button>
      <Button colorPalette="green">
        <LuPlus /> Add New
      </Button>
      <Button colorPalette="purple" variant="subtle">
        Next <LuArrowRight />
      </Button>
      <Button colorPalette="red" variant="ghost">
        <LuTrash2 /> Delete
      </Button>
    </HStack>
  ),
};

export const Disabled = {
  render: () => (
    <HStack gap="4" wrap="wrap">
      <Button disabled>Disabled Solid</Button>
      <Button disabled variant="outline">
        Disabled Outline
      </Button>
      <Button disabled variant="ghost">
        Disabled Ghost
      </Button>
      <Button disabled variant="subtle">
        Disabled Subtle
      </Button>
    </HStack>
  ),
};

export const Loading = {
  render: () => (
    <HStack gap="4" wrap="wrap">
      <Button loading>Loading</Button>
      <Button loading loadingText="Saving...">
        Save
      </Button>
      <Button loading colorPalette="blue">
        Submit
      </Button>
      <Button loading variant="outline" colorPalette="green">
        Processing
      </Button>
    </HStack>
  ),
};

export const LoadingWithSpinnerPlacement = {
  render: () => (
    <HStack gap="4" wrap="wrap">
      <Button loading loadingText="Loading..." spinnerPlacement="start" colorPalette="teal">
        Submit
      </Button>
      <Button loading loadingText="Loading..." spinnerPlacement="end" colorPalette="blue">
        Continue
      </Button>
    </HStack>
  ),
};

export const AsLink = {
  render: () => (
    <HStack gap="4" wrap="wrap">
      <Button asChild>
        <a href="https://example.com" target="_blank" rel="noopener noreferrer">
          Link Button
        </a>
      </Button>
      <Button asChild variant="outline" colorPalette="blue">
        <a href="https://example.com" target="_blank" rel="noopener noreferrer">
          Outline Link <LuArrowRight />
        </a>
      </Button>
    </HStack>
  ),
};

export const CustomStyling = {
  render: () => (
    <HStack gap="4" wrap="wrap">
      <Button height="48px" width="200px" border="2px solid" borderColor="green.500">
        Custom Size
      </Button>
      <Button rounded="full" colorPalette="purple">
        Rounded Full
      </Button>
      <Button rounded="none" colorPalette="red">
        No Radius
      </Button>
    </HStack>
  ),
};

export const AllVariantsAndSizes = {
  render: () => (
    <VStack gap="6" align="flex-start">
      {["solid", "subtle", "surface", "outline", "ghost", "plain"].map((variant) => (
        <VStack key={variant} gap="2" align="flex-start">
          <Text fontWeight="bold" fontSize="sm">
            {variant.toUpperCase()}
          </Text>
          <HStack gap="2" wrap="wrap" align="center">
            <Button variant={variant} size="2xs" colorPalette="blue">
              2XS
            </Button>
            <Button variant={variant} size="xs" colorPalette="blue">
              XS
            </Button>
            <Button variant={variant} size="sm" colorPalette="blue">
              SM
            </Button>
            <Button variant={variant} size="md" colorPalette="blue">
              MD
            </Button>
            <Button variant={variant} size="lg" colorPalette="blue">
              LG
            </Button>
            <Button variant={variant} size="xl" colorPalette="blue">
              XL
            </Button>
            <Button variant={variant} size="2xl" colorPalette="blue">
              2XL
            </Button>
          </HStack>
        </VStack>
      ))}
    </VStack>
  ),
};

export const Playground = {
  args: {
    children: "Click Me",
    variant: "solid",
    size: "md",
    colorPalette: "primary",
    disabled: false,
    loading: false,
  },
};
