// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Show/Show.stories.js
import React, { useState } from "react";

import { Show } from "./Show";
import { Button } from "../Button";
import { Box } from "../Box";
import { Stack } from "../Stack";

export default {
  title: "Utility/Show",
  component: Show,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => {
  const [isVisible, setIsVisible] = useState(true);

  return (
    <Stack gap="4">
      <Button onClick={() => setIsVisible(!isVisible)}>Toggle visibility</Button>
      <Show when={isVisible}>
        <Box p="4" bg="primary.100" borderRadius="md">
          This content is conditionally rendered
        </Box>
      </Show>
    </Stack>
  );
};

export const WithFallback = () => {
  const [isVisible, setIsVisible] = useState(false);

  return (
    <Stack gap="4">
      <Button onClick={() => setIsVisible(!isVisible)}>Toggle visibility</Button>
      <Show
        when={isVisible}
        fallback={
          <Box p="4" bg="gray.100" borderRadius="md">
            Fallback content shown when condition is false
          </Box>
        }
      >
        <Box p="4" bg="green.100" borderRadius="md">
          Main content shown when condition is true
        </Box>
      </Show>
    </Stack>
  );
};

export const Responsive = () => (
  <Stack gap="4">
    <Show above="md">
      <Box p="4" bg="blue.100" borderRadius="md">
        Visible on md and above
      </Box>
    </Show>
    <Show below="md">
      <Box p="4" bg="red.100" borderRadius="md">
        Visible below md
      </Box>
    </Show>
  </Stack>
);
