// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Spacer/Spacer.stories.js
import { Spacer } from "./Spacer";
import { Flex } from "../Flex";
import { Box } from "../Box";
import { HStack } from "../Stack";

export default {
  title: "Layout/Spacer",
  component: Spacer,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Flex bg="bg.muted" p="4" borderRadius="md">
    <Box p="4" bg="primary.500" color="white" borderRadius="md">
      Box 1
    </Box>
    <Spacer />
    <Box p="4" bg="primary.600" color="white" borderRadius="md">
      Box 2
    </Box>
  </Flex>
);

export const MultipleSpacer = () => (
  <Flex bg="bg.muted" p="4" borderRadius="md">
    <Box p="4" bg="primary.500" color="white" borderRadius="md">
      Box 1
    </Box>
    <Spacer />
    <Box p="4" bg="primary.600" color="white" borderRadius="md">
      Box 2
    </Box>
    <Spacer />
    <Box p="4" bg="primary.700" color="white" borderRadius="md">
      Box 3
    </Box>
  </Flex>
);

export const InHStack = () => (
  <HStack bg="bg.muted" p="4" borderRadius="md">
    <Box p="4" bg="primary.500" color="white" borderRadius="md">
      Left
    </Box>
    <Spacer />
    <Box p="4" bg="primary.600" color="white" borderRadius="md">
      Right
    </Box>
  </HStack>
);

export const NavbarExample = () => (
  <Flex as="nav" bg="bg.muted" p="4" borderRadius="md" align="center">
    <Box fontWeight="bold" fontSize="lg">
      Logo
    </Box>
    <Spacer />
    <HStack gap="4">
      <Box cursor="pointer">Home</Box>
      <Box cursor="pointer">About</Box>
      <Box cursor="pointer">Contact</Box>
    </HStack>
    <Spacer />
    <Box p="2" bg="primary.500" color="white" borderRadius="md" cursor="pointer">
      Sign In
    </Box>
  </Flex>
);
