// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Checkbox/Checkbox.stories.js
import React, { useState } from "react";

import { Box } from "../Box";
import { Checkbox } from "./Checkbox";
import { Stack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Form/Checkbox",
  component: Checkbox,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "Checkbox component for boolean input with support for indeterminate state.",
      },
    },
  },
  argTypes: {
    size: {
      control: "select",
      options: ["sm", "md", "lg"],
      description: "Size of the checkbox",
    },
    colorPalette: {
      control: "select",
      options: ["primary", "gray", "red", "green", "blue", "yellow", "purple", "teal"],
      description: "Color palette for the checkbox",
    },
    disabled: {
      control: "boolean",
      description: "Disable the checkbox",
    },
    defaultChecked: {
      control: "boolean",
      description: "Default checked state",
    },
    animated: {
      control: "boolean",
      description: "Enable/disable animations",
      defaultValue: true,
    },
  },
};

export const Default = {
  args: {
    children: "Accept terms and conditions",
  },
};

export const Checked = {
  args: {
    children: "I agree",
    defaultChecked: true,
  },
};

export const Sizes = () => (
  <Stack gap={4}>
    <Checkbox size="sm">Small checkbox</Checkbox>
    <Checkbox size="md">Medium checkbox (default)</Checkbox>
    <Checkbox size="lg">Large checkbox</Checkbox>
  </Stack>
);

export const ColorPalettes = () => (
  <Stack gap={4}>
    <Checkbox colorPalette="primary" defaultChecked>
      Primary
    </Checkbox>
    <Checkbox colorPalette="green" defaultChecked>
      Green
    </Checkbox>
    <Checkbox colorPalette="red" defaultChecked>
      Red
    </Checkbox>
    <Checkbox colorPalette="blue" defaultChecked>
      Blue
    </Checkbox>
    <Checkbox colorPalette="purple" defaultChecked>
      Purple
    </Checkbox>
    <Checkbox colorPalette="teal" defaultChecked>
      Teal
    </Checkbox>
  </Stack>
);

export const Disabled = () => (
  <Stack gap={4}>
    <Checkbox disabled>Disabled unchecked</Checkbox>
    <Checkbox disabled defaultChecked>
      Disabled checked
    </Checkbox>
  </Stack>
);

export const Invalid = {
  args: {
    children: "Required checkbox",
    invalid: true,
  },
};

export const Group = () => (
  <Stack gap={2}>
    <Checkbox defaultChecked>Option 1</Checkbox>
    <Checkbox>Option 2</Checkbox>
    <Checkbox>Option 3</Checkbox>
  </Stack>
);

export const Animations = () => {
  const [checked, setChecked] = useState(false);

  return (
    <Stack gap={6}>
      <Box>
        <Text fontWeight="bold" mb={2}>
          Pop Animation
        </Text>
        <Text fontSize="sm" color="gray.600" mb={3}>
          Click the checkbox to see the pop animation effect
        </Text>
        <Checkbox checked={checked} onCheckedChange={(e) => setChecked(e.checked)}>
          Click me to see animation
        </Checkbox>
      </Box>

      <Box>
        <Text fontWeight="bold" mb={2}>
          Hover Animation
        </Text>
        <Text fontSize="sm" color="gray.600" mb={3}>
          Hover over the checkbox to see scale effect
        </Text>
        <Checkbox>Hover over me</Checkbox>
      </Box>

      <Box>
        <Text fontWeight="bold" mb={2}>
          Different Colors
        </Text>
        <Text fontSize="sm" color="gray.600" mb={3}>
          Animations work with all color palettes
        </Text>
        <Stack gap={3}>
          <Checkbox colorPalette="primary">Primary</Checkbox>
          <Checkbox colorPalette="green">Green</Checkbox>
          <Checkbox colorPalette="purple">Purple</Checkbox>
        </Stack>
      </Box>

      <Box>
        <Text fontWeight="bold" mb={2}>
          Disabled State
        </Text>
        <Checkbox disabled>Disabled (no animations)</Checkbox>
      </Box>
    </Stack>
  );
};

export const WithoutAnimations = () => (
  <Stack gap={4}>
    <Text fontSize="sm" color="gray.600">
      Set animated=false to disable all animations
    </Text>
    <Checkbox animated={false}>No animations</Checkbox>
    <Checkbox animated={false} defaultChecked>
      No animations (checked)
    </Checkbox>
  </Stack>
);
