// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/DatePicker/DatePicker.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.DatePickerViewTrigger = exports.DatePickerViewControl = exports.DatePickerView = exports.DatePickerTrigger = exports.DatePickerTableRow = exports.DatePickerTableHeader = exports.DatePickerTableHead = exports.DatePickerTableCellTrigger = exports.DatePickerTableCell = exports.DatePickerTableBody = exports.DatePickerTable = exports.DatePickerRangeText = exports.DatePickerPrevTrigger = exports.DatePickerPresetTrigger = exports.DatePickerPositioner = exports.DatePickerNextTrigger = exports.DatePickerLabel = exports.DatePickerInput = exports.DatePickerControl = exports.DatePickerContent = exports.DatePickerClearTrigger = exports.DatePicker = void 0;
var _react = _interopRequireDefault(require("react"));
var _datePicker = require("@ark-ui/react/date-picker");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["children"],
  _excluded2 = ["children"],
  _excluded3 = ["children"],
  _excluded4 = ["children"],
  _excluded5 = ["children"],
  _excluded6 = ["children"],
  _excluded7 = ["children"],
  _excluded8 = ["children"],
  _excluded9 = ["children"],
  _excluded0 = ["children"],
  _excluded1 = ["children"],
  _excluded10 = ["children"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
var Root = /*#__PURE__*/_react["default"].forwardRef(function (_ref, ref) {
  var children = _ref.children,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.Root, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Root.displayName = "DatePicker";
var Control = /*#__PURE__*/_react["default"].forwardRef(function (_ref2, ref) {
  var children = _ref2.children,
    props = _objectWithoutProperties(_ref2, _excluded2);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.Control, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Control.displayName = "DatePickerControl";
var Input = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.Input, _objectSpread({
    ref: ref
  }, props));
});
Input.displayName = "DatePickerInput";
var Trigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.Trigger, _objectSpread({
    ref: ref
  }, props));
});
Trigger.displayName = "DatePickerTrigger";
var Content = /*#__PURE__*/_react["default"].forwardRef(function (_ref3, ref) {
  var children = _ref3.children,
    props = _objectWithoutProperties(_ref3, _excluded3);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.Content, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Content.displayName = "DatePickerContent";
var Positioner = /*#__PURE__*/_react["default"].forwardRef(function (_ref4, ref) {
  var children = _ref4.children,
    props = _objectWithoutProperties(_ref4, _excluded4);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.Positioner, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Positioner.displayName = "DatePickerPositioner";
var Label = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.Label, _objectSpread({
    ref: ref
  }, props));
});
Label.displayName = "DatePickerLabel";
var Table = /*#__PURE__*/_react["default"].forwardRef(function (_ref5, ref) {
  var children = _ref5.children,
    props = _objectWithoutProperties(_ref5, _excluded5);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.Table, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
Table.displayName = "DatePickerTable";
var TableHeader = /*#__PURE__*/_react["default"].forwardRef(function (_ref6, ref) {
  var children = _ref6.children,
    props = _objectWithoutProperties(_ref6, _excluded6);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.TableHeader, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
TableHeader.displayName = "DatePickerTableHeader";
var TableBody = /*#__PURE__*/_react["default"].forwardRef(function (_ref7, ref) {
  var children = _ref7.children,
    props = _objectWithoutProperties(_ref7, _excluded7);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.TableBody, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
TableBody.displayName = "DatePickerTableBody";
var TableRow = /*#__PURE__*/_react["default"].forwardRef(function (_ref8, ref) {
  var children = _ref8.children,
    props = _objectWithoutProperties(_ref8, _excluded8);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.TableRow, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
TableRow.displayName = "DatePickerTableRow";
var TableCell = /*#__PURE__*/_react["default"].forwardRef(function (_ref9, ref) {
  var children = _ref9.children,
    props = _objectWithoutProperties(_ref9, _excluded9);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.TableCell, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
TableCell.displayName = "DatePickerTableCell";
var TableCellTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.TableCellTrigger, _objectSpread({
    ref: ref
  }, props));
});
TableCellTrigger.displayName = "DatePickerTableCellTrigger";
var TableHead = /*#__PURE__*/_react["default"].forwardRef(function (_ref0, ref) {
  var children = _ref0.children,
    props = _objectWithoutProperties(_ref0, _excluded0);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.TableHead, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
TableHead.displayName = "DatePickerTableHead";
var ViewControl = /*#__PURE__*/_react["default"].forwardRef(function (_ref1, ref) {
  var children = _ref1.children,
    props = _objectWithoutProperties(_ref1, _excluded1);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.ViewControl, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
ViewControl.displayName = "DatePickerViewControl";
var PrevTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.PrevTrigger, _objectSpread({
    ref: ref
  }, props));
});
PrevTrigger.displayName = "DatePickerPrevTrigger";
var NextTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.NextTrigger, _objectSpread({
    ref: ref
  }, props));
});
NextTrigger.displayName = "DatePickerNextTrigger";
var ViewTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.ViewTrigger, _objectSpread({
    ref: ref
  }, props));
});
ViewTrigger.displayName = "DatePickerViewTrigger";
var View = /*#__PURE__*/_react["default"].forwardRef(function (_ref10, ref) {
  var children = _ref10.children,
    props = _objectWithoutProperties(_ref10, _excluded10);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.View, _objectSpread(_objectSpread({
    ref: ref
  }, props), {}, {
    children: children
  }));
});
View.displayName = "DatePickerView";
var ClearTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.ClearTrigger, _objectSpread({
    ref: ref
  }, props));
});
ClearTrigger.displayName = "DatePickerClearTrigger";
var PresetTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.PresetTrigger, _objectSpread({
    ref: ref
  }, props));
});
PresetTrigger.displayName = "DatePickerPresetTrigger";
var RangeText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_datePicker.DatePicker.RangeText, _objectSpread({
    ref: ref
  }, props));
});
RangeText.displayName = "DatePickerRangeText";

// Attach sub-components as properties
var DatePicker = exports.DatePicker = Object.assign(Root, {
  Control: Control,
  Input: Input,
  Trigger: Trigger,
  Content: Content,
  Positioner: Positioner,
  Label: Label,
  Table: Table,
  TableHeader: TableHeader,
  TableBody: TableBody,
  TableRow: TableRow,
  TableCell: TableCell,
  TableCellTrigger: TableCellTrigger,
  TableHead: TableHead,
  ViewControl: ViewControl,
  PrevTrigger: PrevTrigger,
  NextTrigger: NextTrigger,
  ViewTrigger: ViewTrigger,
  View: View,
  ClearTrigger: ClearTrigger,
  PresetTrigger: PresetTrigger,
  RangeText: RangeText
});

// Named exports for backward compatibility
var DatePickerControl = exports.DatePickerControl = Control;
var DatePickerInput = exports.DatePickerInput = Input;
var DatePickerTrigger = exports.DatePickerTrigger = Trigger;
var DatePickerContent = exports.DatePickerContent = Content;
var DatePickerPositioner = exports.DatePickerPositioner = Positioner;
var DatePickerLabel = exports.DatePickerLabel = Label;
var DatePickerTable = exports.DatePickerTable = Table;
var DatePickerTableHeader = exports.DatePickerTableHeader = TableHeader;
var DatePickerTableBody = exports.DatePickerTableBody = TableBody;
var DatePickerTableRow = exports.DatePickerTableRow = TableRow;
var DatePickerTableCell = exports.DatePickerTableCell = TableCell;
var DatePickerTableCellTrigger = exports.DatePickerTableCellTrigger = TableCellTrigger;
var DatePickerTableHead = exports.DatePickerTableHead = TableHead;
var DatePickerViewControl = exports.DatePickerViewControl = ViewControl;
var DatePickerPrevTrigger = exports.DatePickerPrevTrigger = PrevTrigger;
var DatePickerNextTrigger = exports.DatePickerNextTrigger = NextTrigger;
var DatePickerViewTrigger = exports.DatePickerViewTrigger = ViewTrigger;
var DatePickerView = exports.DatePickerView = View;
var DatePickerClearTrigger = exports.DatePickerClearTrigger = ClearTrigger;
var DatePickerPresetTrigger = exports.DatePickerPresetTrigger = PresetTrigger;
var DatePickerRangeText = exports.DatePickerRangeText = RangeText;
var _default = exports["default"] = DatePicker;
