// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Input/Input.stories.js
import React, { useState } from "react";

import { Box } from "../Box";
import { Button } from "../Button";
import { Input } from "./Input";
import { Stack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Form/Input",
  component: Input,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component:
          "Input component for text entry with optional label, helper text, error states, and smooth animations.",
      },
    },
  },
  argTypes: {
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg"],
      description: "Size of the input",
    },
    variant: {
      control: "select",
      options: ["outline", "filled", "flushed"],
      description: "Visual variant of the input",
    },
    placeholder: {
      control: "text",
      description: "Placeholder text",
    },
    disabled: {
      control: "boolean",
      description: "Disable the input",
    },
    invalid: {
      control: "boolean",
      description: "Mark input as invalid",
    },
    label: {
      control: "text",
      description: "Label for the input field",
    },
    helperText: {
      control: "text",
      description: "Helper text below the input",
    },
    errorText: {
      control: "text",
      description: "Error message when invalid",
    },
    animated: {
      control: "boolean",
      description: "Enable/disable animations",
      defaultValue: true,
    },
    colorPalette: {
      control: "select",
      options: ["primary", "gray", "blue", "green", "red", "purple", "teal", "orange"],
      description: "Color palette for hover/focus states",
      defaultValue: "primary",
    },
  },
};

export const Default = {
  args: {
    placeholder: "Enter text...",
  },
};

export const WithLabel = {
  args: {
    label: "Email",
    placeholder: "Enter your email",
  },
};

export const WithHelperText = {
  args: {
    label: "Password",
    placeholder: "Enter password",
    helperText: "Must be at least 8 characters",
  },
};

export const WithError = {
  args: {
    label: "Email",
    placeholder: "Enter your email",
    invalid: true,
    errorText: "Email is required",
  },
};

export const Sizes = () => (
  <Stack gap={4}>
    <Input size="xs" placeholder="Extra small" />
    <Input size="sm" placeholder="Small" />
    <Input size="md" placeholder="Medium (default)" />
    <Input size="lg" placeholder="Large" />
  </Stack>
);

export const Variants = () => (
  <Stack gap={4}>
    <Input variant="outline" placeholder="Outline variant" />
    <Input variant="filled" placeholder="Filled variant" />
    <Input variant="flushed" placeholder="Flushed variant" />
  </Stack>
);

export const Disabled = {
  args: {
    placeholder: "Disabled input",
    disabled: true,
  },
};

export const Required = {
  args: {
    label: "Required Field",
    placeholder: "This field is required",
    required: true,
  },
};

export const ColorPalettes = () => (
  <Stack gap={4}>
    <Input
      label="Primary (default)"
      placeholder="Hover and focus to see primary color"
      colorPalette="primary"
    />
    <Input label="Blue" placeholder="Hover and focus to see blue color" colorPalette="blue" />
    <Input label="Green" placeholder="Hover and focus to see green color" colorPalette="green" />
    <Input label="Purple" placeholder="Hover and focus to see purple color" colorPalette="purple" />
    <Input label="Teal" placeholder="Hover and focus to see teal color" colorPalette="teal" />
  </Stack>
);

export const Animations = () => {
  const [showError, setShowError] = useState(false);

  return (
    <Stack gap={6}>
      <Box>
        <Text fontWeight="bold" mb={2}>
          Focus Animation
        </Text>
        <Text fontSize="sm" color="gray.600" mb={3}>
          Click on the input to see the glow effect
        </Text>
        <Input
          label="Focus me"
          placeholder="Click to see focus animation"
          helperText="Input shows a subtle glow effect on focus"
        />
      </Box>

      <Box>
        <Text fontWeight="bold" mb={2}>
          Hover Animation
        </Text>
        <Text fontSize="sm" color="gray.600" mb={3}>
          Hover over the input to see border color change
        </Text>
        <Input label="Hover me" placeholder="Hover to see border animation" />
      </Box>

      <Box>
        <Text fontWeight="bold" mb={2}>
          Error Shake Animation
        </Text>
        <Text fontSize="sm" color="gray.600" mb={3}>
          Toggle error state to see shake animation
        </Text>
        <Stack gap={3}>
          <Input
            label="Email"
            placeholder="Enter email"
            invalid={showError}
            errorText={showError ? "Please enter a valid email address" : undefined}
          />
          <Button onClick={() => setShowError(!showError)} size="sm" width="fit-content">
            {showError ? "Clear Error" : "Show Error"}
          </Button>
        </Stack>
      </Box>

      <Box>
        <Text fontWeight="bold" mb={2}>
          Disabled State
        </Text>
        <Input label="Disabled input" placeholder="This input is disabled" disabled />
      </Box>
    </Stack>
  );
};

export const WithoutAnimations = () => (
  <Stack gap={4}>
    <Text fontSize="sm" color="gray.600">
      Set animated=false to disable all animations
    </Text>
    <Input
      label="No animations"
      placeholder="Animations disabled"
      helperText="This input has animations disabled"
      animated={false}
    />
  </Stack>
);

export const InteractiveForm = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [submitted, setSubmitted] = useState(false);

  const emailInvalid = submitted && !email.includes("@");
  const passwordInvalid = submitted && password.length < 8;

  const handleSubmit = () => {
    setSubmitted(true);
  };

  return (
    <Stack gap={4} maxW="400px">
      <Input
        label="Email"
        placeholder="Enter your email"
        value={email}
        onChange={(e) => setEmail(e.target.value)}
        invalid={emailInvalid}
        errorText={emailInvalid ? "Please enter a valid email" : undefined}
      />
      <Input
        label="Password"
        type="password"
        placeholder="Enter your password"
        value={password}
        onChange={(e) => setPassword(e.target.value)}
        invalid={passwordInvalid}
        errorText={passwordInvalid ? "Password must be at least 8 characters" : undefined}
        helperText={!passwordInvalid ? "Must be at least 8 characters" : undefined}
      />
      <Button onClick={handleSubmit}>Submit</Button>
      {submitted && !emailInvalid && !passwordInvalid && (
        <Text color="green.500">Form submitted successfully!</Text>
      )}
    </Stack>
  );
};
