// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Pagination/index.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Pagination", {
  enumerable: true,
  get: function get() {
    return _Pagination.Pagination;
  }
});
Object.defineProperty(exports, "PaginationContext", {
  enumerable: true,
  get: function get() {
    return _Pagination.PaginationContext;
  }
});
Object.defineProperty(exports, "PaginationEllipsis", {
  enumerable: true,
  get: function get() {
    return _Pagination.PaginationEllipsis;
  }
});
Object.defineProperty(exports, "PaginationItem", {
  enumerable: true,
  get: function get() {
    return _Pagination.PaginationItem;
  }
});
Object.defineProperty(exports, "PaginationItems", {
  enumerable: true,
  get: function get() {
    return _Pagination.PaginationItems;
  }
});
Object.defineProperty(exports, "PaginationNextTrigger", {
  enumerable: true,
  get: function get() {
    return _Pagination.PaginationNextTrigger;
  }
});
Object.defineProperty(exports, "PaginationPageText", {
  enumerable: true,
  get: function get() {
    return _Pagination.PaginationPageText;
  }
});
Object.defineProperty(exports, "PaginationPrevTrigger", {
  enumerable: true,
  get: function get() {
    return _Pagination.PaginationPrevTrigger;
  }
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _Pagination["default"];
  }
});
var _Pagination = _interopRequireWildcard(require("./Pagination"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, "default": e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
