// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Pagination/Pagination.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.PaginationPrevTrigger = exports.PaginationPageText = exports.PaginationNextTrigger = exports.PaginationItems = exports.PaginationItem = exports.PaginationEllipsis = exports.PaginationContext = exports.Pagination = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["count", "pageSize", "page", "defaultPage", "onPageChange", "siblingCount", "children"],
  _excluded2 = ["children"],
  _excluded3 = ["children"],
  _excluded4 = ["render"],
  _excluded5 = ["format"],
  _excluded6 = ["value", "children"];
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(e, t) { if (null == e) return {}; var o, r, i = _objectWithoutPropertiesLoose(e, t); if (Object.getOwnPropertySymbols) { var n = Object.getOwnPropertySymbols(e); for (r = 0; r < n.length; r++) o = n[r], -1 === t.indexOf(o) && {}.propertyIsEnumerable.call(e, o) && (i[o] = e[o]); } return i; }
function _objectWithoutPropertiesLoose(r, e) { if (null == r) return {}; var t = {}; for (var n in r) if ({}.hasOwnProperty.call(r, n)) { if (-1 !== e.indexOf(n)) continue; t[n] = r[n]; } return t; }
var Pagination = exports.Pagination = /*#__PURE__*/_react["default"].forwardRef(function (_ref, ref) {
  var count = _ref.count,
    _ref$pageSize = _ref.pageSize,
    pageSize = _ref$pageSize === void 0 ? 10 : _ref$pageSize,
    page = _ref.page,
    _ref$defaultPage = _ref.defaultPage,
    defaultPage = _ref$defaultPage === void 0 ? 1 : _ref$defaultPage,
    onPageChange = _ref.onPageChange,
    _ref$siblingCount = _ref.siblingCount,
    siblingCount = _ref$siblingCount === void 0 ? 1 : _ref$siblingCount,
    children = _ref.children,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Pagination.Root, _objectSpread(_objectSpread({
    ref: ref,
    count: count,
    pageSize: pageSize,
    page: page,
    defaultPage: defaultPage,
    onPageChange: onPageChange,
    siblingCount: siblingCount
  }, props), {}, {
    children: children
  }));
});
Pagination.displayName = "Pagination";

// Pagination sub-components
var PaginationPrevTrigger = exports.PaginationPrevTrigger = /*#__PURE__*/_react["default"].forwardRef(function (_ref2, ref) {
  var children = _ref2.children,
    props = _objectWithoutProperties(_ref2, _excluded2);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Pagination.PrevTrigger, _objectSpread(_objectSpread({
    ref: ref,
    asChild: true
  }, props), {}, {
    children: children
  }));
});
PaginationPrevTrigger.displayName = "PaginationPrevTrigger";
var PaginationNextTrigger = exports.PaginationNextTrigger = /*#__PURE__*/_react["default"].forwardRef(function (_ref3, ref) {
  var children = _ref3.children,
    props = _objectWithoutProperties(_ref3, _excluded3);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Pagination.NextTrigger, _objectSpread(_objectSpread({
    ref: ref,
    asChild: true
  }, props), {}, {
    children: children
  }));
});
PaginationNextTrigger.displayName = "PaginationNextTrigger";
var PaginationItems = exports.PaginationItems = function PaginationItems(_ref4) {
  var render = _ref4.render,
    props = _objectWithoutProperties(_ref4, _excluded4);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Pagination.Items, _objectSpread({
    render: render
  }, props));
};
PaginationItems.displayName = "PaginationItems";
var PaginationPageText = exports.PaginationPageText = /*#__PURE__*/_react["default"].forwardRef(function (_ref5, ref) {
  var _ref5$format = _ref5.format,
    format = _ref5$format === void 0 ? "compact" : _ref5$format,
    props = _objectWithoutProperties(_ref5, _excluded5);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Pagination.PageText, _objectSpread({
    ref: ref,
    format: format
  }, props));
});
PaginationPageText.displayName = "PaginationPageText";
var PaginationEllipsis = exports.PaginationEllipsis = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Pagination.Ellipsis, _objectSpread({
    ref: ref
  }, props));
});
PaginationEllipsis.displayName = "PaginationEllipsis";
var PaginationItem = exports.PaginationItem = /*#__PURE__*/_react["default"].forwardRef(function (_ref6, ref) {
  var value = _ref6.value,
    children = _ref6.children,
    props = _objectWithoutProperties(_ref6, _excluded6);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Pagination.Item, _objectSpread(_objectSpread({
    ref: ref,
    value: value,
    asChild: true
  }, props), {}, {
    children: children
  }));
});
PaginationItem.displayName = "PaginationItem";
var PaginationContext = exports.PaginationContext = _react2.Pagination.Context;
var _default = exports["default"] = Pagination;
