// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Pagination/Pagination.stories.js
import React, { useState } from "react";

import { Box } from "../Box";
import { Button } from "../Button";
import {
  Pagination,
  PaginationPrevTrigger,
  PaginationNextTrigger,
  PaginationItems,
  PaginationPageText,
  PaginationContext,
} from "./Pagination";
import { Stack, HStack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Navigation/Pagination",
  component: Pagination,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "Pagination component for navigating through pages of content.",
      },
    },
  },
  argTypes: {
    count: {
      control: "number",
      description: "Total number of items",
    },
    pageSize: {
      control: "number",
      description: "Number of items per page",
    },
    siblingCount: {
      control: "number",
      description: "Number of sibling pages to show",
    },
  },
};

export const Default = () => (
  <Pagination count={100} pageSize={10} defaultPage={1}>
    <HStack gap={2}>
      <PaginationPrevTrigger>
        <Button variant="outline" size="sm">
          Previous
        </Button>
      </PaginationPrevTrigger>
      <PaginationItems
        render={(page) => (
          <Button key={page.value} variant={page.type === "page" ? "outline" : "ghost"} size="sm">
            {page.type === "ellipsis" ? "..." : page.value}
          </Button>
        )}
      />
      <PaginationNextTrigger>
        <Button variant="outline" size="sm">
          Next
        </Button>
      </PaginationNextTrigger>
    </HStack>
  </Pagination>
);

export const WithPageText = () => (
  <Pagination count={100} pageSize={10} defaultPage={1}>
    <HStack gap={4}>
      <PaginationPrevTrigger>
        <Button variant="outline" size="sm">
          ←
        </Button>
      </PaginationPrevTrigger>
      <PaginationPageText format="compact" />
      <PaginationNextTrigger>
        <Button variant="outline" size="sm">
          →
        </Button>
      </PaginationNextTrigger>
    </HStack>
  </Pagination>
);

export const LongPageText = () => (
  <Pagination count={200} pageSize={10} defaultPage={5}>
    <HStack gap={4}>
      <PaginationPrevTrigger>
        <Button variant="outline" size="sm">
          Previous
        </Button>
      </PaginationPrevTrigger>
      <PaginationPageText format="long" />
      <PaginationNextTrigger>
        <Button variant="outline" size="sm">
          Next
        </Button>
      </PaginationNextTrigger>
    </HStack>
  </Pagination>
);

export const Controlled = () => {
  const [page, setPage] = useState(1);
  const pageSize = 10;
  const count = 100;

  return (
    <Stack gap={4}>
      <Text>
        Current Page: {page} of {Math.ceil(count / pageSize)}
      </Text>
      <Pagination
        count={count}
        pageSize={pageSize}
        page={page}
        onPageChange={(e) => setPage(e.page)}
      >
        <HStack gap={2}>
          <PaginationPrevTrigger>
            <Button variant="outline" size="sm">
              Previous
            </Button>
          </PaginationPrevTrigger>
          <PaginationItems
            render={(page) => (
              <Button
                key={page.value}
                variant={page.type === "page" ? "outline" : "ghost"}
                size="sm"
              >
                {page.type === "ellipsis" ? "..." : page.value}
              </Button>
            )}
          />
          <PaginationNextTrigger>
            <Button variant="outline" size="sm">
              Next
            </Button>
          </PaginationNextTrigger>
        </HStack>
      </Pagination>
    </Stack>
  );
};

export const CustomSiblingCount = () => (
  <Stack gap={6}>
    <Box>
      <Text fontWeight="bold" mb={2}>
        siblingCount: 1 (default)
      </Text>
      <Pagination count={200} pageSize={10} defaultPage={10} siblingCount={1}>
        <HStack gap={2}>
          <PaginationPrevTrigger>
            <Button variant="outline" size="sm">
              ←
            </Button>
          </PaginationPrevTrigger>
          <PaginationItems
            render={(page) => (
              <Button
                key={page.value}
                variant={page.type === "page" ? "outline" : "ghost"}
                size="sm"
              >
                {page.type === "ellipsis" ? "..." : page.value}
              </Button>
            )}
          />
          <PaginationNextTrigger>
            <Button variant="outline" size="sm">
              →
            </Button>
          </PaginationNextTrigger>
        </HStack>
      </Pagination>
    </Box>
    <Box>
      <Text fontWeight="bold" mb={2}>
        siblingCount: 2
      </Text>
      <Pagination count={200} pageSize={10} defaultPage={10} siblingCount={2}>
        <HStack gap={2}>
          <PaginationPrevTrigger>
            <Button variant="outline" size="sm">
              ←
            </Button>
          </PaginationPrevTrigger>
          <PaginationItems
            render={(page) => (
              <Button
                key={page.value}
                variant={page.type === "page" ? "outline" : "ghost"}
                size="sm"
              >
                {page.type === "ellipsis" ? "..." : page.value}
              </Button>
            )}
          />
          <PaginationNextTrigger>
            <Button variant="outline" size="sm">
              →
            </Button>
          </PaginationNextTrigger>
        </HStack>
      </Pagination>
    </Box>
  </Stack>
);

export const DifferentPageSizes = () => (
  <Stack gap={6}>
    {[5, 10, 20].map((size) => (
      <Box key={size}>
        <Text fontWeight="bold" mb={2}>
          Page Size: {size} (Total: 100 items)
        </Text>
        <Pagination count={100} pageSize={size} defaultPage={1}>
          <HStack gap={4}>
            <PaginationPrevTrigger>
              <Button variant="outline" size="sm">
                ←
              </Button>
            </PaginationPrevTrigger>
            <PaginationPageText format="compact" />
            <PaginationNextTrigger>
              <Button variant="outline" size="sm">
                →
              </Button>
            </PaginationNextTrigger>
          </HStack>
        </Pagination>
      </Box>
    ))}
  </Stack>
);

export const MinimalPagination = () => (
  <Pagination count={50} pageSize={10} defaultPage={1}>
    <HStack gap={2}>
      <PaginationPrevTrigger>
        <Button variant="ghost" size="sm">
          ← Prev
        </Button>
      </PaginationPrevTrigger>
      <PaginationNextTrigger>
        <Button variant="ghost" size="sm">
          Next →
        </Button>
      </PaginationNextTrigger>
    </HStack>
  </Pagination>
);

export const WithTableContext = () => {
  const [page, setPage] = useState(1);
  const pageSize = 5;
  const totalItems = 47;
  const totalPages = Math.ceil(totalItems / pageSize);
  const startItem = (page - 1) * pageSize + 1;
  const endItem = Math.min(page * pageSize, totalItems);

  return (
    <Box border="1px" borderColor="gray.200" borderRadius="md" p={4}>
      <Stack gap={4}>
        <Text>
          Showing {startItem}-{endItem} of {totalItems} items
        </Text>
        <Pagination
          count={totalItems}
          pageSize={pageSize}
          page={page}
          onPageChange={(e) => setPage(e.page)}
        >
          <HStack gap={2} justify="center">
            <PaginationPrevTrigger>
              <Button variant="outline" size="sm" disabled={page === 1}>
                Previous
              </Button>
            </PaginationPrevTrigger>
            <Text mx={4}>
              Page {page} of {totalPages}
            </Text>
            <PaginationNextTrigger>
              <Button variant="outline" size="sm" disabled={page === totalPages}>
                Next
              </Button>
            </PaginationNextTrigger>
          </HStack>
        </Pagination>
      </Stack>
    </Box>
  );
};
