// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Status/Status.stories.js
import React from "react";

import { Status } from "./Status";
import { Stack, HStack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Feedback/Status",
  component: Status,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Status value="success">
    <Status.Indicator />
    Online
  </Status>
);

export const AllStatuses = () => (
  <Stack gap="4">
    <Status value="success">
      <Status.Indicator />
      Success / Online
    </Status>
    <Status value="warning">
      <Status.Indicator />
      Warning / Away
    </Status>
    <Status value="error">
      <Status.Indicator />
      Error / Offline
    </Status>
    <Status value="info">
      <Status.Indicator />
      Info / Busy
    </Status>
  </Stack>
);

export const Sizes = () => (
  <Stack gap="4">
    <Status value="success" size="sm">
      <Status.Indicator />
      Small
    </Status>
    <Status value="success" size="md">
      <Status.Indicator />
      Medium
    </Status>
    <Status value="success" size="lg">
      <Status.Indicator />
      Large
    </Status>
  </Stack>
);

export const UserStatus = () => (
  <Stack gap="3">
    <HStack gap="3">
      <Status value="success">
        <Status.Indicator />
      </Status>
      <Text>John Doe - Online</Text>
    </HStack>
    <HStack gap="3">
      <Status value="warning">
        <Status.Indicator />
      </Status>
      <Text>Jane Smith - Away</Text>
    </HStack>
    <HStack gap="3">
      <Status value="error">
        <Status.Indicator />
      </Status>
      <Text>Bob Wilson - Offline</Text>
    </HStack>
  </Stack>
);
