// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/theme/slot-recipes/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.slotRecipes = void 0;
var _accordion = require("./accordion");
var _actionBar = require("./action-bar");
var _alert = require("./alert");
var _avatar = require("./avatar");
var _blockquote = require("./blockquote");
var _breadcrumb = require("./breadcrumb");
var _card = require("./card");
var _carousel = require("./carousel");
var _checkbox = require("./checkbox");
var _checkboxCard = require("./checkbox-card");
var _codeBlock = require("./code-block");
var _collapsible = require("./collapsible");
var _dataList = require("./data-list");
var _dialog = require("./dialog");
var _drawer = require("./drawer");
var _editable = require("./editable");
var _emptyState = require("./empty-state");
var _field = require("./field");
var _fieldset = require("./fieldset");
var _fileUpload = require("./file-upload");
var _hoverCard = require("./hover-card");
var _list = require("./list");
var _listbox = require("./listbox");
var _menu = require("./menu");
var _nativeSelect = require("./native-select");
var _numberInput = require("./number-input");
var _pinInput = require("./pin-input");
var _popover = require("./popover");
var _progress = require("./progress");
var _progressCircle = require("./progress-circle");
var _radioCard = require("./radio-card");
var _radioGroup = require("./radio-group");
var _ratingGroup = require("./rating-group");
var _scrollArea = require("./scroll-area");
var _segmentGroup = require("./segment-group");
var _select = require("./select");
var _combobox = require("./combobox");
var _slider = require("./slider");
var _splitter = require("./splitter");
var _stat = require("./stat");
var _steps = require("./steps");
var _switch = require("./switch");
var _table = require("./table");
var _tabs = require("./tabs");
var _tag = require("./tag");
var _tagsInput = require("./tags-input");
var _toast = require("./toast");
var _tooltip = require("./tooltip");
var _status = require("./status");
var _timeline = require("./timeline");
var _colorPicker = require("./color-picker");
var _qrCode = require("./qr-code");
var _treeView = require("./tree-view");
var slotRecipes = exports.slotRecipes = {
  accordion: _accordion.accordionSlotRecipe,
  actionBar: _actionBar.actionBarSlotRecipe,
  alert: _alert.alertSlotRecipe,
  avatar: _avatar.avatarSlotRecipe,
  blockquote: _blockquote.blockquoteSlotRecipe,
  breadcrumb: _breadcrumb.breadcrumbSlotRecipe,
  card: _card.cardSlotRecipe,
  carousel: _carousel.carouselSlotRecipe,
  checkbox: _checkbox.checkboxSlotRecipe,
  checkboxCard: _checkboxCard.checkboxCardSlotRecipe,
  codeBlock: _codeBlock.codeBlockSlotRecipe,
  collapsible: _collapsible.collapsibleSlotRecipe,
  dataList: _dataList.dataListSlotRecipe,
  dialog: _dialog.dialogSlotRecipe,
  drawer: _drawer.drawerSlotRecipe,
  editable: _editable.editableSlotRecipe,
  emptyState: _emptyState.emptyStateSlotRecipe,
  field: _field.fieldSlotRecipe,
  fieldset: _fieldset.fieldsetSlotRecipe,
  fileUpload: _fileUpload.fileUploadSlotRecipe,
  hoverCard: _hoverCard.hoverCardSlotRecipe,
  list: _list.listSlotRecipe,
  listbox: _listbox.listboxSlotRecipe,
  menu: _menu.menuSlotRecipe,
  nativeSelect: _nativeSelect.nativeSelectSlotRecipe,
  numberInput: _numberInput.numberInputSlotRecipe,
  pinInput: _pinInput.pinInputSlotRecipe,
  popover: _popover.popoverSlotRecipe,
  progress: _progress.progressSlotRecipe,
  progressCircle: _progressCircle.progressCircleSlotRecipe,
  radioCard: _radioCard.radioCardSlotRecipe,
  radioGroup: _radioGroup.radioGroupSlotRecipe,
  ratingGroup: _ratingGroup.ratingGroupSlotRecipe,
  scrollArea: _scrollArea.scrollAreaSlotRecipe,
  segmentGroup: _segmentGroup.segmentGroupSlotRecipe,
  select: _select.selectSlotRecipe,
  combobox: _combobox.comboboxSlotRecipe,
  slider: _slider.sliderSlotRecipe,
  splitter: _splitter.splitterSlotRecipe,
  stat: _stat.statSlotRecipe,
  steps: _steps.stepsSlotRecipe,
  "switch": _switch.switchSlotRecipe,
  table: _table.tableSlotRecipe,
  tabs: _tabs.tabsSlotRecipe,
  tag: _tag.tagSlotRecipe,
  tagsInput: _tagsInput.tagsInputSlotRecipe,
  toast: _toast.toastSlotRecipe,
  tooltip: _tooltip.tooltipSlotRecipe,
  status: _status.statusSlotRecipe,
  timeline: _timeline.timelineSlotRecipe,
  colorPicker: _colorPicker.colorPickerSlotRecipe,
  qrCode: _qrCode.qrCodeSlotRecipe,
  treeView: _treeView.treeViewSlotRecipe
};
