// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/theme/slot-recipes/tooltip.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.tooltipSlotRecipe = void 0;
var _react = require("@chakra-ui/react");
var tooltipSlotRecipe = exports.tooltipSlotRecipe = (0, _react.defineSlotRecipe)({
  slots: ["trigger", "arrow", "arrowTip", "positioner", "content"],
  className: "chakra-tooltip",
  base: {
    content: {
      "--tooltip-bg": "colors.bg.inverted",
      bg: "var(--tooltip-bg)",
      color: "fg.inverted",
      px: "2.5",
      py: "1",
      borderRadius: "l2",
      fontWeight: "medium",
      textStyle: "xs",
      boxShadow: "md",
      maxW: "xs",
      zIndex: "tooltip",
      transformOrigin: "var(--transform-origin)",
      _open: {
        animationStyle: "scale-fade-in",
        animationDuration: "fast"
      },
      _closed: {
        animationStyle: "scale-fade-out",
        animationDuration: "fast"
      }
    },
    arrow: {
      "--arrow-size": "sizes.2",
      "--arrow-background": "var(--tooltip-bg)"
    },
    arrowTip: {
      borderTopWidth: "1px",
      borderLeftWidth: "1px",
      borderColor: "var(--tooltip-bg)"
    }
  }
});
