// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Spinner/Spinner.stories.js
import { Box } from "../Box";
import { Spinner } from "./Spinner";
import { Stack } from "../Stack";

export default {
  title: "Feedback/Spinner",
  component: Spinner,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component: "Spinner component for indicating loading states.",
      },
    },
  },
  argTypes: {
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg", "xl"],
      description: "Size of the spinner",
    },
    colorPalette: {
      control: "select",
      options: ["primary", "gray", "red", "green", "blue", "yellow", "purple", "teal"],
      description: "Color palette for the spinner",
    },
    borderWidth: {
      control: "text",
      description: "Border width of the spinner",
    },
  },
};

export const Default = {
  args: {
    size: "md",
  },
};

export const Sizes = {
  args: {},
  render: (args) => (
    <Stack direction="row" gap={4} align="center">
      <Spinner {...args} size="xs" />
      <Spinner {...args} size="sm" />
      <Spinner {...args} size="md" />
      <Spinner {...args} size="lg" />
      <Spinner {...args} size="xl" />
    </Stack>
  ),
};

export const ColorPalettes = {
  args: {
    size: "md",
  },
  render: (args) => (
    <Stack direction="row" gap={4} align="center">
      <Spinner {...args} colorPalette="primary" />
      <Spinner {...args} colorPalette="blue" />
      <Spinner {...args} colorPalette="green" />
      <Spinner {...args} colorPalette="red" />
      <Spinner {...args} colorPalette="purple" />
      <Spinner {...args} colorPalette="teal" />
    </Stack>
  ),
};

export const CustomBorderWidth = {
  args: {
    size: "lg",
  },
  render: (args) => (
    <Stack direction="row" gap={4} align="center">
      <Spinner {...args} borderWidth="2px" />
      <Spinner {...args} borderWidth="4px" />
      <Spinner {...args} borderWidth="6px" />
    </Stack>
  ),
};

export const WithLabel = {
  args: {
    size: "sm",
  },
  render: (args) => (
    <Stack direction="row" gap={2} align="center">
      <Spinner {...args} />
      <span>Loading...</span>
    </Stack>
  ),
};

export const InButton = {
  args: {
    size: "sm",
    color: "white",
  },
  render: (args) => (
    <Box
      as="button"
      display="flex"
      alignItems="center"
      gap={2}
      px={4}
      py={2}
      bg="primary.500"
      color="white"
      borderRadius="md"
      disabled
      opacity={0.7}
    >
      <Spinner {...args} />
      Processing...
    </Box>
  ),
};
