// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Stack/Stack.stories.js
import { Stack } from "./Stack";
import { Box } from "../Box";

export default {
  title: "Layout/Stack",
  component: Stack,
  tags: ["autodocs"],
  parameters: {
    layout: "padded",
    docs: {
      description: {
        component: "Stack component for arranging elements in a vertical or horizontal layout.",
      },
    },
  },
  argTypes: {
    gap: {
      control: "select",
      options: ["1", "2", "3", "4", "5", "6", "8", "10"],
      description: "Gap between items",
    },
    direction: {
      control: "select",
      options: ["row", "column"],
      description: "Direction of the stack",
    },
    align: {
      control: "select",
      options: ["flex-start", "center", "flex-end", "stretch"],
      description: "Alignment of items",
    },
    justify: {
      control: "select",
      options: ["flex-start", "center", "flex-end", "space-between", "space-around"],
      description: "Justification of items",
    },
  },
};

export const BasicStack = () => (
  <Stack gap="4">
    <Box p="4" bg="blue.500" color="white" borderRadius="md">
      Item 1
    </Box>
    <Box p="4" bg="blue.500" color="white" borderRadius="md">
      Item 2
    </Box>
    <Box p="4" bg="blue.500" color="white" borderRadius="md">
      Item 3
    </Box>
  </Stack>
);

export const HorizontalStack = () => (
  <Stack.Horizontal gap="4">
    <Box p="4" bg="green.500" color="white" borderRadius="md">
      Left
    </Box>
    <Box p="4" bg="green.500" color="white" borderRadius="md">
      Center
    </Box>
    <Box p="4" bg="green.500" color="white" borderRadius="md">
      Right
    </Box>
  </Stack.Horizontal>
);

export const VerticalStack = () => (
  <Stack.Vertical gap="4">
    <Box p="4" bg="purple.500" color="white" borderRadius="md">
      Top
    </Box>
    <Box p="4" bg="purple.500" color="white" borderRadius="md">
      Middle
    </Box>
    <Box p="4" bg="purple.500" color="white" borderRadius="md">
      Bottom
    </Box>
  </Stack.Vertical>
);

export const WithSeparator = () => (
  <Stack gap="4" separator={<Stack.Separator />}>
    <Box p="4" bg="teal.500" color="white" borderRadius="md">
      Section 1
    </Box>
    <Box p="4" bg="teal.500" color="white" borderRadius="md">
      Section 2
    </Box>
    <Box p="4" bg="teal.500" color="white" borderRadius="md">
      Section 3
    </Box>
  </Stack>
);

export const HStackWithSeparator = () => (
  <Stack.Horizontal gap="4" separator={<Stack.Separator />}>
    <Box p="4" bg="orange.500" color="white" borderRadius="md">
      A
    </Box>
    <Box p="4" bg="orange.500" color="white" borderRadius="md">
      B
    </Box>
    <Box p="4" bg="orange.500" color="white" borderRadius="md">
      C
    </Box>
  </Stack.Horizontal>
);

export const ResponsiveDirection = () => (
  <Stack direction={{ base: "column", md: "row" }} gap="4">
    <Box p="6" flex="1" bg="pink.500" color="white" borderRadius="md" textAlign="center">
      Responsive 1
    </Box>
    <Box p="6" flex="1" bg="pink.500" color="white" borderRadius="md" textAlign="center">
      Responsive 2
    </Box>
    <Box p="6" flex="1" bg="pink.500" color="white" borderRadius="md" textAlign="center">
      Responsive 3
    </Box>
  </Stack>
);

export const WithAlign = () => (
  <Stack.Horizontal gap="4" align="center" p="4" bg="gray.100" borderRadius="md" minH="32">
    <Box p="2" bg="red.500" color="white" borderRadius="md">
      Small
    </Box>
    <Box p="4" bg="red.500" color="white" borderRadius="md">
      Medium
    </Box>
    <Box p="6" bg="red.500" color="white" borderRadius="md">
      Large
    </Box>
  </Stack.Horizontal>
);

export const WithJustify = () => (
  <Stack.Horizontal gap="4" justify="space-between" p="4" bg="gray.100" borderRadius="md">
    <Box p="4" bg="cyan.500" color="white" borderRadius="md">
      Start
    </Box>
    <Box p="4" bg="cyan.500" color="white" borderRadius="md">
      Center
    </Box>
    <Box p="4" bg="cyan.500" color="white" borderRadius="md">
      End
    </Box>
  </Stack.Horizontal>
);
