// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Editable/Editable.stories.js
import React from "react";
import { Editable } from "./Editable";
import { Stack, HStack } from "../Stack";
import { Button } from "../Button";

export default {
  title: "Forms/Editable",
  component: Editable,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Editable defaultValue="Click to edit">
    <Editable.Preview />
    <Editable.Input />
  </Editable>
);

export const WithControls = () => (
  <Editable defaultValue="Click to edit">
    <Editable.Preview />
    <Editable.Input />
    <Editable.Control>
      <Editable.EditTrigger asChild>
        <Button size="sm" variant="outline">
          Edit
        </Button>
      </Editable.EditTrigger>
      <Editable.SubmitTrigger asChild>
        <Button size="sm" variant="solid">
          Save
        </Button>
      </Editable.SubmitTrigger>
      <Editable.CancelTrigger asChild>
        <Button size="sm" variant="ghost">
          Cancel
        </Button>
      </Editable.CancelTrigger>
    </Editable.Control>
  </Editable>
);

export const WithTextarea = () => (
  <Editable defaultValue="This is a longer text that can be edited. Click to edit this paragraph.">
    <Editable.Preview />
    <Editable.Textarea />
  </Editable>
);

export const Placeholder = () => (
  <Editable placeholder="Click to add text">
    <Editable.Preview />
    <Editable.Input />
  </Editable>
);

export const Disabled = () => (
  <Editable defaultValue="This cannot be edited" disabled>
    <Editable.Preview />
    <Editable.Input />
  </Editable>
);
