// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Editable/Editable.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Editable = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var Root = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Editable.Root, _objectSpread({
    ref: ref
  }, props));
});
Root.displayName = "Editable";
var Preview = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Editable.Preview, _objectSpread({
    ref: ref
  }, props));
});
Preview.displayName = "EditablePreview";
var Input = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Editable.Input, _objectSpread({
    ref: ref
  }, props));
});
Input.displayName = "EditableInput";
var Textarea = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Editable.Textarea, _objectSpread({
    ref: ref
  }, props));
});
Textarea.displayName = "EditableTextarea";
var Control = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Editable.Control, _objectSpread({
    ref: ref
  }, props));
});
Control.displayName = "EditableControl";
var EditTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Editable.EditTrigger, _objectSpread({
    ref: ref
  }, props));
});
EditTrigger.displayName = "EditableEditTrigger";
var SubmitTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Editable.SubmitTrigger, _objectSpread({
    ref: ref
  }, props));
});
SubmitTrigger.displayName = "EditableSubmitTrigger";
var CancelTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.Editable.CancelTrigger, _objectSpread({
    ref: ref
  }, props));
});
CancelTrigger.displayName = "EditableCancelTrigger";

// Attach sub-components as properties
var Editable = exports.Editable = Object.assign(Root, {
  Preview: Preview,
  Input: Input,
  Textarea: Textarea,
  Control: Control,
  EditTrigger: EditTrigger,
  SubmitTrigger: SubmitTrigger,
  CancelTrigger: CancelTrigger
});
var _default = exports["default"] = Editable;
