// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Collapsible/Collapsible.stories.js
import React from "react";
import { Collapsible } from "./Collapsible";
import { Button } from "../Button";
import { Box } from "../Box";
import { Stack } from "../Stack";

export default {
  title: "Disclosure/Collapsible",
  component: Collapsible,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Collapsible>
    <Collapsible.Trigger asChild>
      <Button variant="outline">Toggle Content</Button>
    </Collapsible.Trigger>
    <Collapsible.Content>
      <Box mt="4" p="4" bg="bg.muted" borderRadius="md">
        This content can be expanded and collapsed. Lorem ipsum dolor sit amet, consectetur
        adipiscing elit.
      </Box>
    </Collapsible.Content>
  </Collapsible>
);

export const DefaultOpen = () => (
  <Collapsible defaultOpen>
    <Collapsible.Trigger asChild>
      <Button variant="outline">Toggle (Default Open)</Button>
    </Collapsible.Trigger>
    <Collapsible.Content>
      <Box mt="4" p="4" bg="bg.muted" borderRadius="md">
        This content is visible by default.
      </Box>
    </Collapsible.Content>
  </Collapsible>
);

export const WithAnimation = () => (
  <Collapsible>
    <Collapsible.Trigger asChild>
      <Button variant="solid">Show More</Button>
    </Collapsible.Trigger>
    <Collapsible.Content>
      <Stack mt="4" gap="2">
        <Box p="3" bg="primary.50" borderRadius="md">
          Item 1
        </Box>
        <Box p="3" bg="primary.100" borderRadius="md">
          Item 2
        </Box>
        <Box p="3" bg="primary.200" borderRadius="md">
          Item 3
        </Box>
        <Box p="3" bg="primary.300" borderRadius="md">
          Item 4
        </Box>
      </Stack>
    </Collapsible.Content>
  </Collapsible>
);

export const ReadMore = () => (
  <Box maxW="400px">
    <Box>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut
      labore et dolore magna aliqua.
    </Box>
    <Collapsible>
      <Collapsible.Content>
        <Box mt="2">
          Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
          commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
          dolore eu fugiat nulla pariatur.
        </Box>
      </Collapsible.Content>
      <Collapsible.Trigger asChild>
        <Button variant="link" size="sm" mt="2">
          Read more
        </Button>
      </Collapsible.Trigger>
    </Collapsible>
  </Box>
);
