// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Stat/Stat.stories.js
import React from "react";

import { Stat } from "./Stat";
import { Stack, HStack } from "../Stack";
import { SimpleGrid } from "../Grid";

export default {
  title: "Data Display/Stat",
  component: Stat,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Stat>
    <Stat.Label>Total Sales</Stat.Label>
    <Stat.ValueText>$45,670</Stat.ValueText>
  </Stat>
);

export const WithHelpText = () => (
  <Stat>
    <Stat.Label>Monthly Revenue</Stat.Label>
    <Stat.ValueText>$12,450</Stat.ValueText>
    <Stat.HelpText>Since last month</Stat.HelpText>
  </Stat>
);

export const WithIndicators = () => (
  <HStack gap="8">
    <Stat>
      <Stat.Label>Revenue</Stat.Label>
      <Stat.ValueText>$45,670</Stat.ValueText>
      <Stat.HelpText>
        <Stat.UpIndicator /> 23.5%
      </Stat.HelpText>
    </Stat>
    <Stat>
      <Stat.Label>Expenses</Stat.Label>
      <Stat.ValueText>$12,340</Stat.ValueText>
      <Stat.HelpText>
        <Stat.DownIndicator /> 12.3%
      </Stat.HelpText>
    </Stat>
  </HStack>
);

export const Dashboard = () => (
  <SimpleGrid columns={{ base: 1, md: 2, lg: 4 }} gap="6">
    <Stat>
      <Stat.Label>Total Users</Stat.Label>
      <Stat.ValueText>12,543</Stat.ValueText>
      <Stat.HelpText>
        <Stat.UpIndicator /> 15.2%
      </Stat.HelpText>
    </Stat>
    <Stat>
      <Stat.Label>Active Sessions</Stat.Label>
      <Stat.ValueText>1,234</Stat.ValueText>
      <Stat.HelpText>
        <Stat.UpIndicator /> 8.1%
      </Stat.HelpText>
    </Stat>
    <Stat>
      <Stat.Label>Bounce Rate</Stat.Label>
      <Stat.ValueText>24.3%</Stat.ValueText>
      <Stat.HelpText>
        <Stat.DownIndicator /> 3.2%
      </Stat.HelpText>
    </Stat>
    <Stat>
      <Stat.Label>Avg. Duration</Stat.Label>
      <Stat.ValueText>4m 32s</Stat.ValueText>
      <Stat.HelpText>
        <Stat.UpIndicator /> 12.5%
      </Stat.HelpText>
    </Stat>
  </SimpleGrid>
);
