// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Rating/Rating.stories.js
import React from "react";

import { Rating } from "./Rating";
import { Stack, HStack } from "../Stack";

export default {
  title: "Forms/Rating",
  component: Rating,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <Rating defaultValue={3} count={5}>
    <Rating.Control>
      {[1, 2, 3, 4, 5].map((index) => (
        <Rating.Item key={index} index={index}>
          <Rating.ItemIndicator />
        </Rating.Item>
      ))}
    </Rating.Control>
  </Rating>
);

export const WithLabel = () => (
  <Rating defaultValue={3} count={5}>
    <Rating.Label>Rate this product</Rating.Label>
    <Rating.Control>
      {[1, 2, 3, 4, 5].map((index) => (
        <Rating.Item key={index} index={index}>
          <Rating.ItemIndicator />
        </Rating.Item>
      ))}
    </Rating.Control>
  </Rating>
);

export const Sizes = () => (
  <Stack gap="4">
    <Rating defaultValue={3} count={5} size="xs">
      <Rating.Control>
        {[1, 2, 3, 4, 5].map((index) => (
          <Rating.Item key={index} index={index}>
            <Rating.ItemIndicator />
          </Rating.Item>
        ))}
      </Rating.Control>
    </Rating>
    <Rating defaultValue={3} count={5} size="sm">
      <Rating.Control>
        {[1, 2, 3, 4, 5].map((index) => (
          <Rating.Item key={index} index={index}>
            <Rating.ItemIndicator />
          </Rating.Item>
        ))}
      </Rating.Control>
    </Rating>
    <Rating defaultValue={3} count={5} size="md">
      <Rating.Control>
        {[1, 2, 3, 4, 5].map((index) => (
          <Rating.Item key={index} index={index}>
            <Rating.ItemIndicator />
          </Rating.Item>
        ))}
      </Rating.Control>
    </Rating>
    <Rating defaultValue={3} count={5} size="lg">
      <Rating.Control>
        {[1, 2, 3, 4, 5].map((index) => (
          <Rating.Item key={index} index={index}>
            <Rating.ItemIndicator />
          </Rating.Item>
        ))}
      </Rating.Control>
    </Rating>
  </Stack>
);

export const ReadOnly = () => (
  <Rating defaultValue={4} count={5} readOnly>
    <Rating.Control>
      {[1, 2, 3, 4, 5].map((index) => (
        <Rating.Item key={index} index={index}>
          <Rating.ItemIndicator />
        </Rating.Item>
      ))}
    </Rating.Control>
  </Rating>
);

export const HalfStars = () => (
  <Rating defaultValue={3.5} count={5} allowHalf>
    <Rating.Control>
      {[1, 2, 3, 4, 5].map((index) => (
        <Rating.Item key={index} index={index}>
          <Rating.ItemIndicator />
        </Rating.Item>
      ))}
    </Rating.Control>
  </Rating>
);

export const ColorPalettes = () => (
  <Stack gap="4">
    <Rating defaultValue={3} count={5} colorPalette="yellow">
      <Rating.Control>
        {[1, 2, 3, 4, 5].map((index) => (
          <Rating.Item key={index} index={index}>
            <Rating.ItemIndicator />
          </Rating.Item>
        ))}
      </Rating.Control>
    </Rating>
    <Rating defaultValue={3} count={5} colorPalette="orange">
      <Rating.Control>
        {[1, 2, 3, 4, 5].map((index) => (
          <Rating.Item key={index} index={index}>
            <Rating.ItemIndicator />
          </Rating.Item>
        ))}
      </Rating.Control>
    </Rating>
    <Rating defaultValue={3} count={5} colorPalette="red">
      <Rating.Control>
        {[1, 2, 3, 4, 5].map((index) => (
          <Rating.Item key={index} index={index}>
            <Rating.ItemIndicator />
          </Rating.Item>
        ))}
      </Rating.Control>
    </Rating>
  </Stack>
);
