// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/CodeBlock/CodeBlock.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.CodeBlockTitle = exports.CodeBlockOverlay = exports.CodeBlockHeader = exports.CodeBlockFooter = exports.CodeBlockCopyTrigger = exports.CodeBlockCopyIndicator = exports.CodeBlockControl = exports.CodeBlockContent = exports.CodeBlockCollapseTrigger = exports.CodeBlockCollapseText = exports.CodeBlockCollapseIndicator = exports.CodeBlockCodeText = exports.CodeBlockCode = exports.CodeBlockAdapterProvider = exports.CodeBlock = void 0;
var _react = _interopRequireDefault(require("react"));
var _react2 = require("@chakra-ui/react");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var Root = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.Root, _objectSpread({
    ref: ref
  }, props));
});
Root.displayName = "CodeBlock";
var Content = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.Content, _objectSpread({
    ref: ref
  }, props));
});
Content.displayName = "CodeBlockContent";
var Code = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.Code, _objectSpread({
    ref: ref
  }, props));
});
Code.displayName = "CodeBlockCode";
var CodeText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.CodeText, _objectSpread({
    ref: ref
  }, props));
});
CodeText.displayName = "CodeBlockCodeText";
var Header = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.Header, _objectSpread({
    ref: ref
  }, props));
});
Header.displayName = "CodeBlockHeader";
var Title = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.Title, _objectSpread({
    ref: ref
  }, props));
});
Title.displayName = "CodeBlockTitle";
var CopyTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.CopyTrigger, _objectSpread({
    ref: ref
  }, props));
});
CopyTrigger.displayName = "CodeBlockCopyTrigger";
var CopyIndicator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.CopyIndicator, _objectSpread({
    ref: ref
  }, props));
});
CopyIndicator.displayName = "CodeBlockCopyIndicator";
var AdapterProvider = function AdapterProvider(props) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.AdapterProvider, _objectSpread({}, props));
};
AdapterProvider.displayName = "CodeBlockAdapterProvider";
var Control = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.Control, _objectSpread({
    ref: ref
  }, props));
});
Control.displayName = "CodeBlockControl";
var Overlay = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.Overlay, _objectSpread({
    ref: ref
  }, props));
});
Overlay.displayName = "CodeBlockOverlay";
var CollapseTrigger = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.CollapseTrigger, _objectSpread({
    ref: ref
  }, props));
});
CollapseTrigger.displayName = "CodeBlockCollapseTrigger";
var CollapseIndicator = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.CollapseIndicator, _objectSpread({
    ref: ref
  }, props));
});
CollapseIndicator.displayName = "CodeBlockCollapseIndicator";
var CollapseText = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.CollapseText, _objectSpread({
    ref: ref
  }, props));
});
CollapseText.displayName = "CodeBlockCollapseText";
var Footer = /*#__PURE__*/_react["default"].forwardRef(function (props, ref) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_react2.CodeBlock.Footer, _objectSpread({
    ref: ref
  }, props));
});
Footer.displayName = "CodeBlockFooter";

// Attach sub-components as properties

var CodeBlock = exports.CodeBlock = Object.assign(Root, {
  Content: Content,
  Code: Code,
  CodeText: CodeText,
  Header: Header,
  Title: Title,
  CopyTrigger: CopyTrigger,
  CopyIndicator: CopyIndicator,
  AdapterProvider: AdapterProvider,
  Control: Control,
  Overlay: Overlay,
  CollapseTrigger: CollapseTrigger,
  CollapseIndicator: CollapseIndicator,
  CollapseText: CollapseText,
  Footer: Footer
});

// Named exports for backward compatibility
var CodeBlockContent = exports.CodeBlockContent = Content;
var CodeBlockCode = exports.CodeBlockCode = Code;
var CodeBlockCodeText = exports.CodeBlockCodeText = CodeText;
var CodeBlockHeader = exports.CodeBlockHeader = Header;
var CodeBlockTitle = exports.CodeBlockTitle = Title;
var CodeBlockCopyTrigger = exports.CodeBlockCopyTrigger = CopyTrigger;
var CodeBlockCopyIndicator = exports.CodeBlockCopyIndicator = CopyIndicator;
var CodeBlockAdapterProvider = exports.CodeBlockAdapterProvider = AdapterProvider;
var CodeBlockControl = exports.CodeBlockControl = Control;
var CodeBlockOverlay = exports.CodeBlockOverlay = Overlay;
var CodeBlockCollapseTrigger = exports.CodeBlockCollapseTrigger = CollapseTrigger;
var CodeBlockCollapseIndicator = exports.CodeBlockCollapseIndicator = CollapseIndicator;
var CodeBlockCollapseText = exports.CodeBlockCollapseText = CollapseText;
var CodeBlockFooter = exports.CodeBlockFooter = Footer;
var _default = exports["default"] = CodeBlock;
