// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/CodeBlock/index.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "CodeBlock", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlock;
  }
});
Object.defineProperty(exports, "CodeBlockAdapterProvider", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockAdapterProvider;
  }
});
Object.defineProperty(exports, "CodeBlockCode", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockCode;
  }
});
Object.defineProperty(exports, "CodeBlockCodeText", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockCodeText;
  }
});
Object.defineProperty(exports, "CodeBlockCollapseIndicator", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockCollapseIndicator;
  }
});
Object.defineProperty(exports, "CodeBlockCollapseText", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockCollapseText;
  }
});
Object.defineProperty(exports, "CodeBlockCollapseTrigger", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockCollapseTrigger;
  }
});
Object.defineProperty(exports, "CodeBlockContent", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockContent;
  }
});
Object.defineProperty(exports, "CodeBlockControl", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockControl;
  }
});
Object.defineProperty(exports, "CodeBlockCopyIndicator", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockCopyIndicator;
  }
});
Object.defineProperty(exports, "CodeBlockCopyTrigger", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockCopyTrigger;
  }
});
Object.defineProperty(exports, "CodeBlockFooter", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockFooter;
  }
});
Object.defineProperty(exports, "CodeBlockHeader", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockHeader;
  }
});
Object.defineProperty(exports, "CodeBlockOverlay", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockOverlay;
  }
});
Object.defineProperty(exports, "CodeBlockTitle", {
  enumerable: true,
  get: function get() {
    return _CodeBlock.CodeBlockTitle;
  }
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function get() {
    return _CodeBlock["default"];
  }
});
var _CodeBlock = _interopRequireWildcard(require("./CodeBlock"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, "default": e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
