// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Popover/Popover.stories.js
import React, { useState } from "react";
import { fn, userEvent, within, expect } from "@storybook/test";

import { Box } from "../Box";
import { Button } from "../Button";
import { Input } from "../Input";
import { Popover } from "./Popover";
import { Stack, HStack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Overlay/Popover",
  component: Popover,
  tags: ["autodocs"],
  parameters: {
    layout: "padded",
    docs: {
      description: {
        component:
          "Popover component for displaying floating content relative to a trigger element.",
      },
    },
  },
  args: {
    onOpenChange: fn(),
  },
  argTypes: {
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg"],
      description: "Size of the popover",
    },
    onOpenChange: {
      action: "open changed",
      description: "Callback when popover open state changes",
    },
  },
};

export const Default = {
  args: {
    size: "md",
  },
  render: (args) => (
    <Popover {...args}>
      <Popover.Trigger>
        <Button variant="outline">Open Popover</Button>
      </Popover.Trigger>
      <Popover.Content>
        <Popover.Arrow />
        <Popover.Body>
          <Popover.Title fontWeight="medium">Popover Title</Popover.Title>
          <Text mt={2}>This is the popover content. You can put any content here.</Text>
        </Popover.Body>
      </Popover.Content>
    </Popover>
  ),
  play: async ({ canvasElement, args }) => {
    const canvas = within(canvasElement);
    const openButton = canvas.getByRole("button", { name: /open popover/i });

    // Click to open popover
    await userEvent.click(openButton);
    await expect(args.onOpenChange).toHaveBeenCalled();
  },
};

export const WithHeaderAndFooter = {
  args: {
    size: "md",
  },
  render: (args) => (
    <Popover {...args}>
      <Popover.Trigger>
        <Button>Manage Channels</Button>
      </Popover.Trigger>
      <Popover.Content>
        <Popover.Arrow />
        <Popover.Header>
          <Popover.Title>Channel Settings</Popover.Title>
        </Popover.Header>
        <Popover.Body>
          <Text>Configure your notification channels and preferences.</Text>
        </Popover.Body>
        <Popover.Footer>
          <HStack justify="flex-end" gap={2}>
            <Button variant="outline" size="sm">
              Cancel
            </Button>
            <Button size="sm">Save</Button>
          </HStack>
        </Popover.Footer>
        <Popover.CloseTrigger />
      </Popover.Content>
    </Popover>
  ),
};

export const WithForm = {
  args: {
    size: "md",
  },
  render: (args) => (
    <Popover {...args}>
      <Popover.Trigger>
        <Button>Edit Dimensions</Button>
      </Popover.Trigger>
      <Popover.Content>
        <Popover.Arrow />
        <Popover.Body>
          <Stack gap={4}>
            <Input label="Width" placeholder="e.g., 100px" size="sm" />
            <Input label="Height" placeholder="e.g., 100px" size="sm" />
          </Stack>
        </Popover.Body>
        <Popover.CloseTrigger />
      </Popover.Content>
    </Popover>
  ),
};

export const Sizes = {
  args: {},
  render: (args) => (
    <Stack direction="row" gap={4}>
      {["xs", "sm", "md", "lg"].map((size) => (
        <Popover key={size} size={size} {...args}>
          <Popover.Trigger>
            <Button variant="outline">{size.toUpperCase()}</Button>
          </Popover.Trigger>
          <Popover.Content>
            <Popover.Arrow />
            <Popover.Body>
              <Popover.Title>Size: {size}</Popover.Title>
              <Text mt={2}>This is a {size} popover.</Text>
            </Popover.Body>
          </Popover.Content>
        </Popover>
      ))}
    </Stack>
  ),
};

export const Placements = {
  args: {},
  render: (args) => (
    <Box py={20} px={10}>
      <Stack direction="row" gap={4} wrap="wrap" justify="center">
        {[
          "top-start",
          "top",
          "top-end",
          "bottom-start",
          "bottom",
          "bottom-end",
          "left-start",
          "left",
          "left-end",
          "right-start",
          "right",
          "right-end",
        ].map((placement) => (
          <Popover key={placement} positioning={{ placement }} {...args}>
            <Popover.Trigger>
              <Button variant="outline" size="sm">
                {placement}
              </Button>
            </Popover.Trigger>
            <Popover.Content>
              <Popover.Arrow />
              <Popover.Body>
                <Text>Placement: {placement}</Text>
              </Popover.Body>
            </Popover.Content>
          </Popover>
        ))}
      </Stack>
    </Box>
  ),
};

export const Controlled = () => {
  const [open, setOpen] = useState(false);

  return (
    <Stack direction="row" gap={4} align="center">
      <Popover open={open} onOpenChange={(e) => setOpen(e.open)}>
        <Popover.Trigger>
          <Button>Click Me</Button>
        </Popover.Trigger>
        <Popover.Content>
          <Popover.Arrow />
          <Popover.Body>
            <Text>Controlled popover</Text>
            <Button size="sm" mt={2} onClick={() => setOpen(false)}>
              Close
            </Button>
          </Popover.Body>
        </Popover.Content>
      </Popover>
      <Button variant="outline" onClick={() => setOpen(!open)}>
        Toggle: {open ? "Open" : "Closed"}
      </Button>
    </Stack>
  );
};

export const ConfirmationPopover = {
  args: {
    size: "md",
  },
  render: (args) => (
    <Popover {...args}>
      <Popover.Trigger>
        <Button colorPalette="red">Delete</Button>
      </Popover.Trigger>
      <Popover.Content>
        <Popover.Arrow />
        <Popover.Header>
          <Popover.Title>Confirm Deletion</Popover.Title>
        </Popover.Header>
        <Popover.Body>
          <Text>Are you sure you want to delete this item?</Text>
        </Popover.Body>
        <Popover.Footer>
          <HStack justify="flex-end" gap={2}>
            <Button variant="outline" size="sm">
              Cancel
            </Button>
            <Button colorPalette="red" size="sm">
              Delete
            </Button>
          </HStack>
        </Popover.Footer>
        <Popover.CloseTrigger />
      </Popover.Content>
    </Popover>
  ),
};

export const WithCustomBackground = {
  args: {
    size: "md",
  },
  render: (args) => (
    <Popover {...args}>
      <Popover.Trigger>
        <Button>Custom Background</Button>
      </Popover.Trigger>
      <Popover.Content css={{ "--popover-bg": "var(--chakra-colors-blue-50)" }}>
        <Popover.Arrow />
        <Popover.Body>
          <Text>This popover has a custom background color.</Text>
        </Popover.Body>
      </Popover.Content>
    </Popover>
  ),
};
