// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Card/Card.stories.js
import React from "react";

import { Box } from "../Box";
import { Button } from "../Button";
import { Card } from "./Card";
import { Stack, HStack } from "../Stack";
import { Text } from "../Text";

export default {
  title: "Data Display/Card",
  component: Card,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component:
          "Card component for grouping related content with header, body, and footer sections.",
      },
    },
  },
  argTypes: {
    variant: {
      control: "select",
      options: ["subtle", "outline", "elevated"],
      description: "Visual variant of the card",
    },
    size: {
      control: "select",
      options: ["sm", "md", "lg"],
      description: "Size of the card",
    },
  },
};

export const Default = () => (
  <Card maxW="320px">
    <Card.Body>
      <Text>This is a basic card with just body content.</Text>
    </Card.Body>
  </Card>
);

export const WithHeaderAndFooter = () => (
  <Card maxW="320px">
    <Card.Header>
      <Card.Title>Card Title</Card.Title>
    </Card.Header>
    <Card.Body>
      <Text>This card has a header, body, and footer.</Text>
    </Card.Body>
    <Card.Footer>
      <Button size="sm">Action</Button>
    </Card.Footer>
  </Card>
);

export const WithDescription = () => (
  <Card maxW="320px">
    <Card.Header>
      <Card.Title>Project Alpha</Card.Title>
      <Card.Description>Last updated 3 days ago</Card.Description>
    </Card.Header>
    <Card.Body>
      <Text>This is a project card with title and description in the header.</Text>
    </Card.Body>
    <Card.Footer gap={2}>
      <Button variant="outline" size="sm">
        Cancel
      </Button>
      <Button size="sm">Save</Button>
    </Card.Footer>
  </Card>
);

export const Variants = () => (
  <HStack gap={4} wrap="wrap">
    {["subtle", "outline", "elevated"].map((variant) => (
      <Card key={variant} variant={variant} maxW="240px">
        <Card.Header>
          <Card.Title>{variant}</Card.Title>
        </Card.Header>
        <Card.Body>
          <Text>This is the {variant} variant.</Text>
        </Card.Body>
      </Card>
    ))}
  </HStack>
);

export const Sizes = () => (
  <Stack gap={4}>
    {["sm", "md", "lg"].map((size) => (
      <Card key={size} size={size} maxW="320px">
        <Card.Header>
          <Card.Title>Size: {size}</Card.Title>
        </Card.Header>
        <Card.Body>
          <Text>Card content with {size} padding.</Text>
        </Card.Body>
      </Card>
    ))}
  </Stack>
);

export const HorizontalCard = () => (
  <Card flexDirection="row" overflow="hidden" maxW="xl">
    <Box w="200px" bg="gray.200" display="flex" alignItems="center" justifyContent="center">
      Image
    </Box>
    <Box>
      <Card.Body>
        <Card.Title mb={2}>Horizontal Card</Card.Title>
        <Text>This card has a horizontal layout with an image placeholder on the left.</Text>
      </Card.Body>
      <Card.Footer>
        <Button size="sm">Learn More</Button>
      </Card.Footer>
    </Box>
  </Card>
);

export const InteractiveCard = () => (
  <Card
    maxW="320px"
    cursor="pointer"
    _hover={{ shadow: "lg", transform: "translateY(-2px)" }}
    transition="all 0.2s"
  >
    <Card.Body>
      <Card.Title mb={2}>Hover Me</Card.Title>
      <Text>This card has hover effects for interactivity.</Text>
    </Card.Body>
  </Card>
);
