// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Switch/Switch.stories.js
import { fn, userEvent, within, expect } from "@storybook/test";

import { Stack } from "../Stack";
import { Switch } from "./Switch";

export default {
  title: "Form/Switch",
  component: Switch,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component:
          "Switch component for toggling between two states, similar to a checkbox but with a different UI.",
      },
    },
  },
  args: {
    onCheckedChange: fn(),
  },
  argTypes: {
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg"],
      description: "Size of the switch",
    },
    colorPalette: {
      control: "select",
      options: ["primary", "gray", "red", "green", "blue", "yellow", "purple", "teal"],
      description: "Color palette for the switch",
    },
    disabled: {
      control: "boolean",
      description: "Disable the switch",
    },
    defaultChecked: {
      control: "boolean",
      description: "Default checked state",
    },
    variant: {
      control: "select",
      options: ["solid", "raised"],
      description: "Visual variant of the switch",
    },
    onCheckedChange: {
      action: "checked changed",
      description: "Callback when switch state changes",
    },
  },
};

export const Default = {
  args: {
    children: "Enable notifications",
  },
  play: async ({ canvasElement, args }) => {
    const canvas = within(canvasElement);
    const switchElement = canvas.getByRole("checkbox");

    // Click to toggle on
    await userEvent.click(switchElement);
    await expect(args.onCheckedChange).toHaveBeenCalled();
  },
};

export const Checked = {
  args: {
    children: "Feature enabled",
    defaultChecked: true,
  },
};

export const Sizes = {
  args: {
    children: "Switch label",
    size: "md",
  },
  render: (args) => (
    <Stack gap={4}>
      <Switch {...args} size="xs">
        Extra small
      </Switch>
      <Switch {...args} size="sm">
        Small
      </Switch>
      <Switch {...args} size="md">
        Medium (default)
      </Switch>
      <Switch {...args} size="lg">
        Large
      </Switch>
    </Stack>
  ),
};

export const ColorPalettes = {
  args: {
    defaultChecked: true,
  },
  render: (args) => (
    <Stack gap={4}>
      <Switch {...args} colorPalette="primary">
        Primary
      </Switch>
      <Switch {...args} colorPalette="green">
        Green
      </Switch>
      <Switch {...args} colorPalette="red">
        Red
      </Switch>
      <Switch {...args} colorPalette="blue">
        Blue
      </Switch>
      <Switch {...args} colorPalette="purple">
        Purple
      </Switch>
      <Switch {...args} colorPalette="teal">
        Teal
      </Switch>
    </Stack>
  ),
};

export const Disabled = {
  args: {
    disabled: true,
    children: "Disabled switch",
  },
  render: (args) => (
    <Stack gap={4}>
      <Switch {...args}>Disabled off</Switch>
      <Switch {...args} defaultChecked>
        Disabled on
      </Switch>
    </Stack>
  ),
};

export const Variants = {
  args: {
    children: "Switch label",
  },
  render: (args) => (
    <Stack gap={4}>
      <Switch {...args} variant="solid">
        Solid variant
      </Switch>
      <Switch {...args} variant="raised">
        Raised variant
      </Switch>
    </Stack>
  ),
};
