// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/DataList/DataList.stories.js
import React from "react";

import { DataList } from "./DataList";
import { Stack } from "../Stack";

export default {
  title: "Data Display/DataList",
  component: DataList,
  parameters: {
    layout: "padded",
  },
};

export const Basic = () => (
  <DataList>
    <DataList.Item>
      <DataList.ItemLabel>Name</DataList.ItemLabel>
      <DataList.ItemValue>John Doe</DataList.ItemValue>
    </DataList.Item>
    <DataList.Item>
      <DataList.ItemLabel>Email</DataList.ItemLabel>
      <DataList.ItemValue>john@example.com</DataList.ItemValue>
    </DataList.Item>
    <DataList.Item>
      <DataList.ItemLabel>Phone</DataList.ItemLabel>
      <DataList.ItemValue>+1 234 567 890</DataList.ItemValue>
    </DataList.Item>
  </DataList>
);

export const Horizontal = () => (
  <DataList orientation="horizontal">
    <DataList.Item>
      <DataList.ItemLabel>Status</DataList.ItemLabel>
      <DataList.ItemValue>Active</DataList.ItemValue>
    </DataList.Item>
    <DataList.Item>
      <DataList.ItemLabel>Role</DataList.ItemLabel>
      <DataList.ItemValue>Admin</DataList.ItemValue>
    </DataList.Item>
    <DataList.Item>
      <DataList.ItemLabel>Team</DataList.ItemLabel>
      <DataList.ItemValue>Engineering</DataList.ItemValue>
    </DataList.Item>
  </DataList>
);

export const UserProfile = () => (
  <DataList maxW="400px">
    <DataList.Item>
      <DataList.ItemLabel>Full Name</DataList.ItemLabel>
      <DataList.ItemValue>Jane Smith</DataList.ItemValue>
    </DataList.Item>
    <DataList.Item>
      <DataList.ItemLabel>Email Address</DataList.ItemLabel>
      <DataList.ItemValue>jane.smith@company.com</DataList.ItemValue>
    </DataList.Item>
    <DataList.Item>
      <DataList.ItemLabel>Department</DataList.ItemLabel>
      <DataList.ItemValue>Product Design</DataList.ItemValue>
    </DataList.Item>
    <DataList.Item>
      <DataList.ItemLabel>Location</DataList.ItemLabel>
      <DataList.ItemValue>San Francisco, CA</DataList.ItemValue>
    </DataList.Item>
    <DataList.Item>
      <DataList.ItemLabel>Member Since</DataList.ItemLabel>
      <DataList.ItemValue>January 2024</DataList.ItemValue>
    </DataList.Item>
  </DataList>
);
