// @rishikeshjoshi-morpheus/ui@1.1.0/dist/src/ui/Textarea/Textarea.stories.js
import React, { useState } from "react";

import { Box } from "../Box";
import { Button } from "../Button";
import { Stack } from "../Stack";
import { Text } from "../Text";
import { Textarea } from "./Textarea";

export default {
  title: "Form/Textarea",
  component: Textarea,
  tags: ["autodocs"],
  parameters: {
    docs: {
      description: {
        component:
          "Textarea component for multi-line text input with optional label and validation.",
      },
    },
  },
  argTypes: {
    size: {
      control: "select",
      options: ["xs", "sm", "md", "lg"],
      description: "Size of the textarea",
    },
    variant: {
      control: "select",
      options: ["outline", "filled", "flushed"],
      description: "Visual variant",
    },
    placeholder: {
      control: "text",
      description: "Placeholder text",
    },
    disabled: {
      control: "boolean",
      description: "Disable the textarea",
    },
    rows: {
      control: "number",
      description: "Number of visible rows",
    },
    label: {
      control: "text",
      description: "Label for the textarea",
    },
    animated: {
      control: "boolean",
      description: "Enable/disable animations",
      defaultValue: true,
    },
    colorPalette: {
      control: "select",
      options: ["primary", "gray", "blue", "green", "red", "purple", "teal", "orange"],
      description: "Color palette for hover/focus states",
      defaultValue: "primary",
    },
  },
};

export const Default = {
  args: {
    placeholder: "Enter your message...",
  },
};

export const WithLabel = {
  args: {
    label: "Description",
    placeholder: "Enter description...",
  },
};

export const WithHelperText = {
  args: {
    label: "Bio",
    placeholder: "Tell us about yourself",
    helperText: "Maximum 500 characters",
  },
};

export const WithError = {
  args: {
    label: "Comments",
    placeholder: "Enter your comments",
    invalid: true,
    errorText: "Comments are required",
  },
};

export const Sizes = () => (
  <Stack gap={4}>
    <Textarea size="xs" placeholder="Extra small" />
    <Textarea size="sm" placeholder="Small" />
    <Textarea size="md" placeholder="Medium (default)" />
    <Textarea size="lg" placeholder="Large" />
  </Stack>
);

export const Variants = () => (
  <Stack gap={4}>
    <Textarea variant="outline" placeholder="Outline variant" />
    <Textarea variant="filled" placeholder="Filled variant" />
    <Textarea variant="flushed" placeholder="Flushed variant" />
  </Stack>
);

export const CustomRows = {
  args: {
    placeholder: "Custom height (6 rows)",
    rows: 6,
  },
};

export const Disabled = {
  args: {
    placeholder: "Disabled textarea",
    disabled: true,
  },
};

export const Resize = () => (
  <Stack gap={4}>
    <Textarea placeholder="Resize: both (default)" resize="both" />
    <Textarea placeholder="Resize: horizontal" resize="horizontal" />
    <Textarea placeholder="Resize: vertical" resize="vertical" />
    <Textarea placeholder="Resize: none" resize="none" />
  </Stack>
);

export const ColorPalettes = () => (
  <Stack gap={4}>
    <Textarea
      label="Primary (default)"
      placeholder="Hover and focus to see primary color"
      colorPalette="primary"
    />
    <Textarea label="Blue" placeholder="Hover and focus to see blue color" colorPalette="blue" />
    <Textarea label="Green" placeholder="Hover and focus to see green color" colorPalette="green" />
    <Textarea
      label="Purple"
      placeholder="Hover and focus to see purple color"
      colorPalette="purple"
    />
    <Textarea label="Teal" placeholder="Hover and focus to see teal color" colorPalette="teal" />
  </Stack>
);

export const Animations = () => {
  const [showError, setShowError] = useState(false);

  return (
    <Stack gap={6}>
      <Box>
        <Text fontWeight="bold" mb={2}>
          Focus Animation
        </Text>
        <Text fontSize="sm" color="gray.600" mb={3}>
          Click on the textarea to see the glow effect
        </Text>
        <Textarea
          label="Focus me"
          placeholder="Click to see focus animation"
          helperText="Textarea shows a subtle glow effect on focus"
        />
      </Box>

      <Box>
        <Text fontWeight="bold" mb={2}>
          Hover Animation
        </Text>
        <Text fontSize="sm" color="gray.600" mb={3}>
          Hover over the textarea to see border color change
        </Text>
        <Textarea label="Hover me" placeholder="Hover to see border animation" />
      </Box>

      <Box>
        <Text fontWeight="bold" mb={2}>
          Error Shake Animation
        </Text>
        <Text fontSize="sm" color="gray.600" mb={3}>
          Toggle error state to see shake animation
        </Text>
        <Stack gap={3}>
          <Textarea
            label="Comments"
            placeholder="Enter comments"
            invalid={showError}
            errorText={showError ? "Comments are required" : undefined}
          />
          <Button onClick={() => setShowError(!showError)} size="sm" width="fit-content">
            {showError ? "Clear Error" : "Show Error"}
          </Button>
        </Stack>
      </Box>

      <Box>
        <Text fontWeight="bold" mb={2}>
          Disabled State
        </Text>
        <Textarea label="Disabled textarea" placeholder="This textarea is disabled" disabled />
      </Box>
    </Stack>
  );
};

export const WithoutAnimations = () => (
  <Stack gap={4}>
    <Text fontSize="sm" color="gray.600">
      Set animated=false to disable all animations
    </Text>
    <Textarea
      label="No animations"
      placeholder="Animations disabled"
      helperText="This textarea has animations disabled"
      animated={false}
    />
  </Stack>
);
